<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
    "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <meta http-equiv="content-type" content="text/html;charset=UTF-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1">
  
  <meta name="google-site-verification" content="3hq0Cfhw5_-EaaNMum0ETX847sTl7jeQ3ZNAla4a6UQ" />
  
  <title>myExperiment &#45; Workflows &#45; Mapping OligoNucleotides to an assembly (Wassinki) [Taverna 1 Workflow]</title>
  <link rel="alternate" href="http://www.myexperiment.org/workflows/603.rdf" type="application/rdf+xml" title="RDF+XML" />
  <link rel="alternate" href="http://www.myexperiment.org/workflows/603.xml" type="application/xml" title="REST XML" />
  <link rel="shortcut icon" href="/favicon.ico" type="image/x-icon"/>
  <link href="/stylesheets/acts_as_taggable_stylesheet.css?1680607579" media="screen" rel="stylesheet" type="text/css" />
<link href="/stylesheets/star_rating.css?1680607579" media="screen" rel="stylesheet" type="text/css" />
<link href="/stylesheets/gadgets.css?1680607579" media="screen" rel="stylesheet" type="text/css" />
<link href="/stylesheets/misc_skinning.css?1680607579" media="screen" rel="stylesheet" type="text/css" />
<link href="/stylesheets/pagination.css?1680607579" media="screen" rel="stylesheet" type="text/css" />
<link href="/stylesheets/bootstrap.css?1680607579" media="screen" rel="stylesheet" type="text/css" />
<link href="/stylesheets/navbar.css?1680607579" media="screen" rel="stylesheet" type="text/css" />
<link href="/stylesheets/styles.css?1680607579" media="screen" rel="stylesheet" type="text/css" />
  <script src="/javascripts/jquery-1.11.1.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/jquery-noconflict.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/prototype-1.7.2-patched.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/effects.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/dragdrop.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/controls.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/boxover.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/bootstrap.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/tabs.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/folds.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/application.js?1680607579" type="text/javascript"></script>

  
  






  
  
    <script src='//www.google.com/recaptcha/api.js'></script>
  
</head>
<body>



<div class="navbar navbar-myexp" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      
          <a id="logo" href="/">
            <img alt="Logo" class="hidden-sm" src="/images/logo.png?1680607579" />
            <img alt="Logo_cropped" class="visible-sm" src="/images/logo_cropped.png?1680607579" />
          </a>
      
    </div>
    <div class="collapse navbar-collapse">
      <ul class="nav navbar-nav">
        
          <ul class="nav navbar-nav">
  
    <li >
      <a href="/home">Home</a>    </li>
  
    <li >
      <a href="/users">Users</a>    </li>
  
    <li >
      <a href="/groups">Groups</a>    </li>
  
    <li class="active">
      <a href="/workflows">Workflows</a>    </li>
  
    <li >
      <a href="/files">Files</a>    </li>
  
    <li >
      <a href="/packs">Packs</a>    </li>
  
</ul>

        
        
            <form action="/search" class="navbar-form navbar-left" id="search" method="get">
    <div class="input-group">
      <input class="form-control" id="query" name="query" placeholder="Search" type="text" />
      <span class="input-group-btn">
        
        <input class="btn btn-default" name="commit" type="submit" value="Go" />
        </span>
    </div>
</form>

        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
            <li><a href="/session/new?return_to=%2Fworkflows%2F603.html">Log in</a></li>
    <li><a href="/users/new">Register</a></li>


      </ul>
    </div><!--/.nav-collapse -->
  </div>
  
</div>




    <div class="container" id="main">
      <div id="myexp_breadcrumbs_bar">
        <ol class="breadcrumb">
  <li><a href="/home">Home</a></li>
  


  <li><a href="/workflows">Workflows</a></li>



  
    <li>Mapping OligoNucleotides to an assembly</li>
  


  <div style="float: right"></div>
</ol>

      </div>
      <div id="myexp_content">
        


        



<h1 class="contribution_title">
  Mapping OligoNucleotides to an assembly
</h1>

<div class="contribution_aftertitle">
	<b>Created:</b> 2009-02-13 09:05:35
	
		&nbsp;&nbsp;&nbsp;&nbsp;
		<b>Last updated: </b>2009-02-13 09:08:20
	
</div>
<ul class="sectionIcons">
  
      <li><span class="icon"><a href="http://www.myexperiment.org/workflows/603/download/Mapping_OligoNucleotides_to_an_assembly-v7.xml?version=7"><img alt="Download Workflow (for version 7)" src="/images/redmond_studio/arrow-down_16.png?1680607579" /> Download Workflow</a></span></li>
      
  
      
</ul>


<div class="contribution_left_box">

  <div class="contribution_version_box">



    <div class="contribution_version_inner_box">

      
    <div class="contribution_description">
      <div class="contribution_description">
<div class="contribution_description">
<p><b>Version info</b></p>
<p>The former version of the workflow expected that results from BioMART only report transcripts when the query (the probe in our<br />
case) are entirely encapsulated in an exon of that transcript. However, the BioMart service also returns transcripts when the query is not or only partially overlapping with an exon in the stretch on the assembly on which a transcript is defined. This resulted in too many oligos classified as having multiple transcripts or having multiple genes.<br />
<br />
<b>Workflow description</b></p>
<p>We used RShell in the design process of a Zebrafish microarray<br />
(supp. info Figure S1 and Figure S2). A microarray with 15k probes<br />
of 60-mer oligonucleotides was designed on gene sequences from<br />
Vega (<a href="http://vega.sanger.ac.uk/Danio_rerio">http://vega.sanger.ac.uk/Danio_rerio</a>) and Ensembl<br />
(<a href="http://www.ensembl.org/Danio_rerio/">http://www.ensembl.org/Danio_rerio/</a>) that are also known<br />
in the Zebrafish Information Network (<a href="http://zfin.org/">http://zfin.org</a>) (for zebra<br />
fish, the <span class="caps">VEGA</span> set is not a subset of the Ensembl set) of the genome<br />
<span class="caps">DNA</span>-sequence assemblies and to judge the agreement that exists between<br />
the different assembly annotations, we mapped the Vega-designed probes<br />
onto the Ensembl assembly</p>
<p>It first performs an alignment using the BioMoby Blat and Blast service provided by <span class="caps"><span class="caps"><span class="caps">WUR</span></span></span> (<a href="http://www.bioinformatics.nl/">www.bioinformatics.nl</a>). Next, for each hit, tries to find the corresponding transcripts and genes using a biomart webservice. The final task is an analysis task using RShell. It calculates for each oligo to which class it belongs:</p>
<p>0 no hit<br />
1 single hit, single transcript, single gene<br />
2 multiple hits, single transcript, single gene, intron spanning<br />
3 multiple hits, single transcript, single gene, possible intron spanning *<br />
4 multiple hits, single transcript, single gene, no intron spanning<br />
5 multiple hits, multiple transcripts, single gene, intron spanning<br />
6 multiple hits, multiple transcripts, single gene, possible intron spanning *<br />
7 multiple hits, multiple transcripts, single gene, no intron spanning<br />
8 single hit, does not meet additional criteria <strong>**<br />
</strong> 9 multiple hits, single transcript, do not meet additional criteria **<br />
10 multiple hits, multiple transcripts, do not meet additional criteria <strong>**<br />
</strong>11 multiple hits, multiple genes<br />
12 no transcript found but hit(s) meet additional criteria **<br />
13 no transcript found and hit(s) do not meet additional criteria <strong>**<br />
</strong>14 multiple hits, single transcript, single gene plus hit without transcript found and hits<br />
meet additional criteria **<br />
* Oligo below e-value cut-off 1e-12, but also intron spanning criteria met.<br />
** Additional criteria: either e-value below 1e-12 or intron spanning.</p>
<p>To run this workflow, a <b>certificate </b>to access <a href="http://www.bioinformatics.nl/">www.bioinformatics.nl</a> needs to installed (Some services use an <span class="caps"><span class="caps"><span class="caps">SSL</span></span></span> connection). Look at the link below how to install this certificate.<br />
<br />
<a href="../../../files/148">http://www.myexperiment.org/files/148</a></p>
<p>The myExperiment pack <a href="../../../packs/45">http://www.myexperiment.org/packs/45</a> contains the workflow, the input and a test input. The whole input set is large. It takes about 6 hours on a 3 GHz Linux pc with 24 Gig <span class="caps"><span class="caps">RAM</span></span>. The test input set can be run on almost any computer with Taverna and R installed. This set takes approximately 10 minutes.</p>
</div>
</div>
    </div>


      <h3>
        <img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[This section shows the preview of this version of the Workflow (click on the image to expand)] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
        Preview
      </h3>

      

          <div class="contribution_preview" style="width: 100%;">
            <p style="text-align: center">
              <a href="/workflows/603/versions/7/previews/full" onclick="window.open(this.href);return false;"><img alt="Medium" src="/workflows/603/versions/7/previews/medium" /></a>
            </p>
            
                <p style="text-align: right; color: #666666;">
                  <a href="/workflows/603/versions/7/previews/svg">Download as scalable diagram (SVG)</a>
                </p>
            

            
          </div>
      



      <h3>
        <img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[This section provides options for running this version of the Workflow] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
        Run
      </h3>

      
          
              <div style="margin: 0 0.5em;">
	<h4>
		Run this Workflow in the Taverna Workbench...
	</h4>
	
	<div class="option_box">
		
		<p style="text-align: center;">
			<b>Option 1:</b>
		</p>
		
		<p style="font-size: 93%; text-align: center;">
			Copy and paste this link into File > 'Open workflow location...'<br/>
			<a href="http://www.myexperiment.org/workflows/603/download?version=7">http://www.myexperiment.org/workflows/603/download?version=7</a><br/>
			<small>[ <a href="#" onclick="Effect.toggle(&quot;run_taverna_more_box&quot;,'blind',{duration:0.3});; return false;">More Info<img alt="Expand" src="/images/folds/unfold.png?1680607579" style="margin-left: 0.3em; vertical-align: middle;" /></a> ]</small>
		</p>
		
		<div id="run_taverna_more_box" style="display: none;">
			<p style="font-size: 85%; margin-top: 0.5em; text-align: center;">
				Taverna is available from <a href="http://taverna.sourceforge.net/" onclick="window.open(this.href);return false;">http://taverna.sourceforge.net/</a>
			</p>
			
			<p style="text-align: center; margin-top: 0.5em; font-size: 85%;">
				If you are having problems downloading it in Taverna, you may need to provide your username and password in the URL so that Taverna can access the Workflow: <br/>
				
					<b>Replace <font color="#990000">http://</font> in the link above with <font color="#990000">http://yourusername:yourpassword@</font></b>
				
			</p>
		</div>
	
	</div>
	
	
</div>
          
      

      <br/>

      <h3>
        <img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[This section shows the internal components of this workflow version] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
        Workflow Components
      </h3>

      
          
              

	
		
		<!-- Sources -->
		
		<div class="fold">
			<div class="foldTitle">
		    Inputs (3)
			</div>
		  <div class="foldContent" style="display: none;">
				
					<table class="simple">
					  <tr>
					    <th>Name</th>
					    <th>Description</th>
					  </tr>
					
						
						  <tr>
						    <td><b>DataBaseName</b></td>
						    <td>Database name (for example Danio_rerio_Genome)</td>
						  </tr>
						
						  <tr>
						    <td><b>Sequences</b></td>
						    <td>Sequences in fasta format, for example

&gt;ENSDART00000061775
TTGTTTCCTCATCAACACAGCAGATCGAATCATTCGAGTTTACGACGGTCGAGAGATCCT
&gt;ENSDART00000100022
TGCTGTTCAGTGGTTATGTTGTTGTTTGAATAAATGTTAAGAGCCAGTGGATGGCACAAA
&gt;OTTDART00000006800
ATCTCTTAGCACTCTGCTGACTCACAACTTCTTCAGAAATGACTTTTTGGATATCATGAA
&gt;OTTDART00000002447
AAGACTGTACGACAAGACAGTGCAAATGGCACCATAGTAAATTCAACCGCTCACCAGGAA
&gt;ENSDART00000047499
CTCTATGACGTATATTGCTATGTGGAGAACATTCATGGGGAGGTTTTTCATGGCTCAACC
&gt;ENSDART00000093312
CAGAGGGTTGCAACCTCTTCATCTATCATCTACCACAAGAGTTTGGTGACAATGAGCTTA
&gt;OTTDART00000002445
AATGTTGCTGGTATCAGTGACCCCTTTCTGCAGGTGCGCATTCTTAGATTGCTAAGGATT
&gt;ENSDART00000093311
CGGGCCTTTCTGGAGAAACGCAAACCTGTGTGGAGCAACACAGACGACTGCATTCACTGA
&gt;ENSDART00000085701
AGAATGACAATGACTGTGGAGCTTTTGTTTTGGAGTACTGTAAGTGCCTGGCCTTCATGA
&gt;OTTDART00000002443
AAACCATCACGCTTTAATTAGTTTCCCCTGTTAACCATTGTCCCACAAGTCTTATGTGGA
&gt;OTTDART00000002442
CTGAAAGGCACTTGAGTTAATCAAATCCGCTTCTATGTAAGTGTTTTGTAAGAGCAGGCT</td>
						  </tr>
						
						  <tr>
						    <td><b>Restart</b></td>
						    <td>Whether the workflow should start from scratch. 
If false, the workflow will continue. This is useful if a crash has occured
in Taverna (and it does sometimes when the data set is large)</td>
						  </tr>
						
					</table>
				
			</div>
		</div>
		
		<!-- Processors -->
		
		<div class="fold">
			<div class="foldTitle">
		    Processors (20)
			</div>
		  <div class="foldContent" style="display: none;">
				
					<table class="simple">
					  <tr>
					    <th>Name</th>
							<th>Type</th>
					    <th>Description</th>
					  </tr>
					
						
						  <tr>
						    <td><b>BlastHitsFinished_Filename</b></td>
								<td>stringconstant</td>
						    <td></td>
						  </tr>
						
						  <tr>
						    <td><b>BioMartReport_Filename</b></td>
								<td>stringconstant</td>
						    <td></td>
						  </tr>
						
						  <tr>
						    <td><b>OligosNotFound_Filename</b></td>
								<td>stringconstant</td>
						    <td></td>
						  </tr>
						
						  <tr>
						    <td><b>BlastReport_Filename</b></td>
								<td>stringconstant</td>
						    <td></td>
						  </tr>
						
						  <tr>
						    <td><b>GeneratePlots</b></td>
								<td>rshell</td>
						    <td></td>
						  </tr>
						
						  <tr>
						    <td><b>Read_BlastReport</b></td>
								<td>local</td>
						    <td></td>
						  </tr>
						
						  <tr>
						    <td><b>Create_Header_BlastReport</b></td>
								<td>beanshell</td>
						    <td>Processor to add content to a (existing)  file. The content is added to the end of the file.

The inputs:
Filename: the file name of a file, if the file does not exists, a new file is added
Content: the string to append
NewLine [default = true]: if true, a newline is added to the end of the line (useful if you want to add a record each time)</td>
						  </tr>
						
						  <tr>
						    <td><b>Create_Semaphore</b></td>
								<td>beanshell</td>
						    <td></td>
						  </tr>
						
						  <tr>
						    <td><b>Read_OligosNotFound</b></td>
								<td>local</td>
						    <td></td>
						  </tr>
						
						  <tr>
						    <td><b>Chunk_Size</b></td>
								<td>stringconstant</td>
						    <td></td>
						  </tr>
						
						  <tr>
						    <td><b>Create_Header_BioMartReport</b></td>
								<td>beanshell</td>
						    <td>Processor to add content to a (existing)  file. The content is added to the end of the file.

The inputs:
Filename: the file name of a file, if the file does not exists, a new file is added
Content: the string to append
NewLine [default = true]: if true, a newline is added to the end of the line (useful if you want to add a record each time)</td>
						  </tr>
						
						  <tr>
						    <td><b>Create_Sequence_Chunks</b></td>
								<td>beanshell</td>
						    <td></td>
						  </tr>
						
						  <tr>
						    <td><b>Touch_BlastHitsFinished_File</b></td>
								<td>beanshell</td>
						    <td></td>
						  </tr>
						
						  <tr>
						    <td><b>Split_Blast_Report</b></td>
								<td>beanshell</td>
						    <td></td>
						  </tr>
						
						  <tr>
						    <td><b>Read_BioMartReport</b></td>
								<td>local</td>
						    <td></td>
						  </tr>
						
						  <tr>
						    <td><b>Touch_OligosNotFound_File</b></td>
								<td>beanshell</td>
						    <td>Processor to add content to a (existing)  file. The content is added to the end of the file.

The inputs:
Filename: the file name of a file, if the file does not exists, a new file is added
Content: the string to append
NewLine [default = true]: if true, a newline is added to the end of the line (useful if you want to add a record each time)</td>
						  </tr>
						
						  <tr>
						    <td><b>Filter_Sequences_For_Blast</b></td>
								<td>beanshell</td>
						    <td></td>
						  </tr>
						
						  <tr>
						    <td><b>Split_sequences</b></td>
								<td>beanshell</td>
						    <td></td>
						  </tr>
						
						  <tr>
						    <td><b>DoBioMart</b></td>
								<td>workflow</td>
						    <td></td>
						  </tr>
						
						  <tr>
						    <td><b>BlatOrBlast</b></td>
								<td>workflow</td>
						    <td>This workflow combines the blat and blast workflows. It takes as input a database name (Danio_rerio_Genome for Zebra Fish for example) and and a set of Fasta sequences. It first tries to perform a blat (at www.bioinformatics.nl). When this service returns nothing, a blast is done (also at www.bioinformatics.nl). The resulting reports are combined.</td>
						  </tr>
						
					</table>
				
			</div>
		</div>
		
		<!-- Beanshells -->
    
    <div class="fold">
      <div class="foldTitle">
        Beanshells (28)
      </div>
      <div class="foldContent" style="display: none;">
        
          <table class="simple">
            <tr>
              <th>Name</th>
              <th>Description</th>
              <th>Inputs</th>
              <th>Outputs</th>
            </tr>
          
            
              <tr>
                <td><b>Create_BioMart_Record</b></td>
                <td></td>
                <td>
                  
                    
                      geneId
                      <hr />
                    
                      transcriptId
                      <hr />
                    
                      transcriptStart
                      <hr />
                    
                      transcriptEnd
                      
                    
                  
                </td>
                <td>
                  
                    
                      record
                      
                    
                  
                </td>
              </tr>
            
              <tr>
                <td><b>EmptyList</b></td>
                <td></td>
                <td>
                  
                </td>
                <td>
                  
                    
                      list
                      
                    
                  
                </td>
              </tr>
            
              <tr>
                <td><b>checkIsInExon</b></td>
                <td></td>
                <td>
                  
                    
                      DStart
                      <hr />
                    
                      DStop
                      <hr />
                    
                      ExonStart
                      <hr />
                    
                      ExonStop
                      
                    
                  
                </td>
                <td>
                  
                    
                      isInExon
                      
                    
                  
                </td>
              </tr>
            
              <tr>
                <td><b>Append_To_BioMartReport</b></td>
                <td>Processor to add content to a (existing)  file. The content is added to the end of the file.

The inputs:
Filename: the file name of a file, if the file does not exists, a new file is added
Content: the string to append
NewLine [default = true]: if true, a newline is added to the end of the line (useful if you want to add a record each time)</td>
                <td>
                  
                    
                      Filename
                      <hr />
                    
                      Content
                      <hr />
                    
                      NewLine
                      
                    
                  
                </td>
                <td>
                  
                </td>
              </tr>
            
              <tr>
                <td><b>Split_Blast_Record</b></td>
                <td></td>
                <td>
                  
                    
                      blastRecord
                      
                    
                  
                </td>
                <td>
                  
                    
                      oligoId
                      <hr />
                    
                      blastIndex
                      <hr />
                    
                      chromosomeRegion
                      <hr />
                    
                      dstart
                      <hr />
                    
                      dstop
                      
                    
                  
                </td>
              </tr>
            
              <tr>
                <td><b>Add_OligoID_BlastIndex_Prefix</b></td>
                <td></td>
                <td>
                  
                    
                      BioMartRecord
                      <hr />
                    
                      OligoID
                      <hr />
                    
                      BlastIndex
                      
                    
                  
                </td>
                <td>
                  
                    
                      FullRecord
                      
                    
                  
                </td>
              </tr>
            
              <tr>
                <td><b>Append_To_BlastHitsFinished_File</b></td>
                <td></td>
                <td>
                  
                    
                      Filename
                      <hr />
                    
                      Content
                      <hr />
                    
                      NewLine
                      
                    
                  
                </td>
                <td>
                  
                </td>
              </tr>
            
              <tr>
                <td><b>Correct_Moby_Object</b></td>
                <td></td>
                <td>
                  
                    
                      inputXML
                      
                    
                  
                </td>
                <td>
                  
                    
                      outputXML
                      
                    
                  
                </td>
              </tr>
            
              <tr>
                <td><b>isRunning</b></td>
                <td></td>
                <td>
                  
                    
                      status
                      
                    
                  
                </td>
                <td>
                  
                    
                      isRunning
                      
                    
                  
                </td>
              </tr>
            
              <tr>
                <td><b>DownloadURLWithBasicAuth</b></td>
                <td>This Beanshell downloads a file to disk. The standard download local Java widgets don&#39;t handle URLs with HTTP(S) Basic Authentication, but this Beanshell can. When a webserver uses BasicAuth, a login and password can be coded as part of the URL using the following syntax: http(s)://login:password@www.some.website/my/great/tool/result.xml. This beanshel extracts the login and password from the URL and supplies them automatically to the webserver. This prevents Taverna from showing popup dialogs requesting the login and password from the user as this will be problematic for large workflows.

Please note that the path where the downloaded file will be stored must be an absolute path to a folder ended with a slash. (Slash backward on Windows or a slash forward on Linux/Unix/Mac OS X.) The filename for the result is automatically extracted from the URL.</td>
                <td>
                  
                    
                      theURL
                      
                    
                  
                </td>
                <td>
                  
                    
                      blastResults
                      
                    
                  
                </td>
              </tr>
            
              <tr>
                <td><b>Download_Report_and_Filter</b></td>
                <td>This Beanshell downloads a file to disk. The standard download local Java widgets don&#39;t handle URLs with HTTP(S) Basic Authentication, but this Beanshell can. When a webserver uses BasicAuth, a login and password can be coded as part of the URL using the following syntax: http(s)://login:password@www.some.website/my/great/tool/result.xml. This beanshel extracts the login and password from the URL and supplies them automatically to the webserver. This prevents Taverna from showing popup dialogs requesting the login and password from the user as this will be problematic for large workflows.

Please note that the path where the downloaded file will be stored must be an absolute path to a folder ended with a slash. (Slash backward on Windows or a slash forward on Linux/Unix/Mac OS X.) The filename for the result is automatically extracted from the URL.</td>
                <td>
                  
                    
                      URL
                      <hr />
                    
                      eValue
                      
                    
                  
                </td>
                <td>
                  
                    
                      blatResults
                      
                    
                  
                </td>
              </tr>
            
              <tr>
                <td><b>isEmpty</b></td>
                <td></td>
                <td>
                  
                    
                      string
                      
                    
                  
                </td>
                <td>
                  
                    
                      isEmpty
                      
                    
                  
                </td>
              </tr>
            
              <tr>
                <td><b>Filter_Sequences_Not_Found</b></td>
                <td></td>
                <td>
                  
                    
                      Report
                      <hr />
                    
                      Sequences
                      
                    
                  
                </td>
                <td>
                  
                    
                      SequencesNotFound
                      
                    
                  
                </td>
              </tr>
            
              <tr>
                <td><b>Filter_Sequences_For_Blast</b></td>
                <td></td>
                <td>
                  
                    
                      sequences
                      <hr />
                    
                      blatResult
                      
                    
                  
                </td>
                <td>
                  
                    
                      sequencesForBlast
                      
                    
                  
                </td>
              </tr>
            
              <tr>
                <td><b>Append_To_BlastReport</b></td>
                <td>Processor to add content to a (existing)  file. The content is added to the end of the file.

The inputs:
Filename: the file name of a file, if the file does not exists, a new file is added
Content: the string to append
NewLine [default = true]: if true, a newline is added to the end of the line (useful if you want to add a record each time)</td>
                <td>
                  
                    
                      Filename
                      <hr />
                    
                      Content
                      <hr />
                    
                      NewLine
                      
                    
                  
                </td>
                <td>
                  
                </td>
              </tr>
            
              <tr>
                <td><b>Append_To_OligosNotFound</b></td>
                <td>Processor to add content to a (existing)  file. The content is added to the end of the file.

The inputs:
Filename: the file name of a file, if the file does not exists, a new file is added
Content: the string to append
NewLine [default = true]: if true, a newline is added to the end of the line (useful if you want to add a record each time)</td>
                <td>
                  
                    
                      Filename
                      <hr />
                    
                      Content
                      <hr />
                    
                      NewLine
                      
                    
                  
                </td>
                <td>
                  
                </td>
              </tr>
            
              <tr>
                <td><b>Join_Blat_Blast_Results</b></td>
                <td></td>
                <td>
                  
                    
                      list1
                      <hr />
                    
                      list2
                      
                    
                  
                </td>
                <td>
                  
                    
                      outputList
                      
                    
                  
                </td>
              </tr>
            
              <tr>
                <td><b>EmptyList</b></td>
                <td></td>
                <td>
                  
                </td>
                <td>
                  
                    
                      list
                      
                    
                  
                </td>
              </tr>
            
              <tr>
                <td><b>Add_Indices_To_BlastReport</b></td>
                <td></td>
                <td>
                  
                    
                      record
                      
                    
                  
                </td>
                <td>
                  
                    
                      record_with_index
                      
                    
                  
                </td>
              </tr>
            
              <tr>
                <td><b>Create_Header_BlastReport</b></td>
                <td>Processor to add content to a (existing)  file. The content is added to the end of the file.

The inputs:
Filename: the file name of a file, if the file does not exists, a new file is added
Content: the string to append
NewLine [default = true]: if true, a newline is added to the end of the line (useful if you want to add a record each time)</td>
                <td>
                  
                    
                      Filename
                      <hr />
                    
                      Content
                      <hr />
                    
                      NewLine
                      <hr />
                    
                      Restart
                      
                    
                  
                </td>
                <td>
                  
                </td>
              </tr>
            
              <tr>
                <td><b>Create_Semaphore</b></td>
                <td></td>
                <td>
                  
                </td>
                <td>
                  
                </td>
              </tr>
            
              <tr>
                <td><b>Create_Header_BioMartReport</b></td>
                <td>Processor to add content to a (existing)  file. The content is added to the end of the file.

The inputs:
Filename: the file name of a file, if the file does not exists, a new file is added
Content: the string to append
NewLine [default = true]: if true, a newline is added to the end of the line (useful if you want to add a record each time)</td>
                <td>
                  
                    
                      Filename
                      <hr />
                    
                      Content
                      <hr />
                    
                      NewLine
                      <hr />
                    
                      Restart
                      
                    
                  
                </td>
                <td>
                  
                </td>
              </tr>
            
              <tr>
                <td><b>Create_Sequence_Chunks</b></td>
                <td></td>
                <td>
                  
                    
                      sequences
                      <hr />
                    
                      chunkSize
                      
                    
                  
                </td>
                <td>
                  
                    
                      chunks
                      
                    
                  
                </td>
              </tr>
            
              <tr>
                <td><b>Touch_BlastHitsFinished_File</b></td>
                <td></td>
                <td>
                  
                    
                      Filename
                      <hr />
                    
                      Content
                      <hr />
                    
                      NewLine
                      <hr />
                    
                      Restart
                      
                    
                  
                </td>
                <td>
                  
                </td>
              </tr>
            
              <tr>
                <td><b>Split_Blast_Report</b></td>
                <td></td>
                <td>
                  
                    
                      BlastReport
                      <hr />
                    
                      filename
                      
                    
                  
                </td>
                <td>
                  
                    
                      Records
                      
                    
                  
                </td>
              </tr>
            
              <tr>
                <td><b>Touch_OligosNotFound_File</b></td>
                <td>Processor to add content to a (existing)  file. The content is added to the end of the file.

The inputs:
Filename: the file name of a file, if the file does not exists, a new file is added
Content: the string to append
NewLine [default = true]: if true, a newline is added to the end of the line (useful if you want to add a record each time)</td>
                <td>
                  
                    
                      Filename
                      <hr />
                    
                      Content
                      <hr />
                    
                      NewLine
                      
                    
                  
                </td>
                <td>
                  
                </td>
              </tr>
            
              <tr>
                <td><b>Filter_Sequences_For_Blast</b></td>
                <td></td>
                <td>
                  
                    
                      sequences
                      <hr />
                    
                      filename
                      
                    
                  
                </td>
                <td>
                  
                    
                      sequencesToDo
                      
                    
                  
                </td>
              </tr>
            
              <tr>
                <td><b>Split_sequences</b></td>
                <td></td>
                <td>
                  
                    
                      sequenceText
                      
                    
                  
                </td>
                <td>
                  
                    
                      sequences
                      
                    
                  
                </td>
              </tr>
            
          </table>
        
      </div>
    </div>
    
    <!-- Sinks -->
		
		<div class="fold">
			<div class="foldTitle">
		    Outputs (6)
			</div>
		  <div class="foldContent" style="display: none;">
				
					<table class="simple">
					  <tr>
					    <th>Name</th>
					    <th>Description</th>
					  </tr>
					
						
						  <tr>
						    <td><b>BlastReport</b></td>
						    <td>The blast hits</td>
						  </tr>
						
						  <tr>
						    <td><b>BioMartReport</b></td>
						    <td>The biomart genes and transcripts</td>
						  </tr>
						
						  <tr>
						    <td><b>SequencesNotFound</b></td>
						    <td>The sequences not found by blast and blat</td>
						  </tr>
						
						  <tr>
						    <td><b>BarPlot</b></td>
						    <td>A bar plot of the oligos per class</td>
						  </tr>
						
						  <tr>
						    <td><b>Classes</b></td>
						    <td>The classes, same as bar plot</td>
						  </tr>
						
						  <tr>
						    <td><b>Report</b></td>
						    <td>The total R report</td>
						  </tr>
						
					</table>
				
			</div>
		</div>
		
		<!-- Links -->
		
		<div class="fold">
			<div class="foldTitle">
		    Links (33)
			</div>
		  <div class="foldContent" style="display: none;">
				
					<table class="simple">
					  <tr>
					    <th>Source</th>
					    <th>Sink</th>
					  </tr>
					
						
						  <tr>
						    <td>DataBaseName</td>
						    <td>BlatOrBlast:DataBaseName</td>
						  </tr>
						
						  <tr>
						    <td>BioMartReport_Filename:value</td>
						    <td>Create_Header_BioMartReport:Filename</td>
						  </tr>
						
						  <tr>
						    <td>BioMartReport_Filename:value</td>
						    <td>DoBioMart:BioMartReport_Filename</td>
						  </tr>
						
						  <tr>
						    <td>BioMartReport_Filename:value</td>
						    <td>Read_BioMartReport:fileurl</td>
						  </tr>
						
						  <tr>
						    <td>BlastReport_Filename:value</td>
						    <td>BlatOrBlast:BlastReport_Filename</td>
						  </tr>
						
						  <tr>
						    <td>BlastReport_Filename:value</td>
						    <td>Create_Header_BlastReport:Filename</td>
						  </tr>
						
						  <tr>
						    <td>BlastReport_Filename:value</td>
						    <td>Read_BlastReport:fileurl</td>
						  </tr>
						
						  <tr>
						    <td>Chunk_Size:value</td>
						    <td>Create_Sequence_Chunks:chunkSize</td>
						  </tr>
						
						  <tr>
						    <td>Create_Sequence_Chunks:chunks</td>
						    <td>BlatOrBlast:Sequences</td>
						  </tr>
						
						  <tr>
						    <td>OligosNotFound_Filename:value</td>
						    <td>BlatOrBlast:OligosNotFound_Filename</td>
						  </tr>
						
						  <tr>
						    <td>OligosNotFound_Filename:value</td>
						    <td>Read_OligosNotFound:fileurl</td>
						  </tr>
						
						  <tr>
						    <td>OligosNotFound_Filename:value</td>
						    <td>Touch_OligosNotFound_File:Filename</td>
						  </tr>
						
						  <tr>
						    <td>Read_BioMartReport:filecontents</td>
						    <td>GeneratePlots:biomartfile</td>
						  </tr>
						
						  <tr>
						    <td>Read_BlastReport:filecontents</td>
						    <td>GeneratePlots:blastresultfile</td>
						  </tr>
						
						  <tr>
						    <td>Read_BlastReport:filecontents</td>
						    <td>Split_Blast_Report:BlastReport</td>
						  </tr>
						
						  <tr>
						    <td>Read_OligosNotFound:filecontents</td>
						    <td>GeneratePlots:oligosNotFound</td>
						  </tr>
						
						  <tr>
						    <td>Restart</td>
						    <td>Create_Header_BioMartReport:Restart</td>
						  </tr>
						
						  <tr>
						    <td>Restart</td>
						    <td>Create_Header_BlastReport:Restart</td>
						  </tr>
						
						  <tr>
						    <td>Sequences</td>
						    <td>Filter_Sequences_For_Blast:sequences</td>
						  </tr>
						
						  <tr>
						    <td>BlastHitsFinished_Filename:value</td>
						    <td>Touch_BlastHitsFinished_File:Filename</td>
						  </tr>
						
						  <tr>
						    <td>BlastReport_Filename:value</td>
						    <td>Filter_Sequences_For_Blast:filename</td>
						  </tr>
						
						  <tr>
						    <td>Filter_Sequences_For_Blast:sequencesToDo</td>
						    <td>Split_sequences:sequenceText</td>
						  </tr>
						
						  <tr>
						    <td>Restart</td>
						    <td>Touch_BlastHitsFinished_File:Restart</td>
						  </tr>
						
						  <tr>
						    <td>BlastHitsFinished_Filename:value</td>
						    <td>DoBioMart:BlastHitsFinished_FileName</td>
						  </tr>
						
						  <tr>
						    <td>BlastHitsFinished_Filename:value</td>
						    <td>Split_Blast_Report:filename</td>
						  </tr>
						
						  <tr>
						    <td>GeneratePlots:BarPlot</td>
						    <td>BarPlot</td>
						  </tr>
						
						  <tr>
						    <td>GeneratePlots:Classes</td>
						    <td>Classes</td>
						  </tr>
						
						  <tr>
						    <td>GeneratePlots:Report</td>
						    <td>Report</td>
						  </tr>
						
						  <tr>
						    <td>Read_BioMartReport:filecontents</td>
						    <td>BioMartReport</td>
						  </tr>
						
						  <tr>
						    <td>Read_BlastReport:filecontents</td>
						    <td>BlastReport</td>
						  </tr>
						
						  <tr>
						    <td>Read_OligosNotFound:filecontents</td>
						    <td>SequencesNotFound</td>
						  </tr>
						
						  <tr>
						    <td>Split_Blast_Report:Records</td>
						    <td>DoBioMart:blastRecord</td>
						  </tr>
						
						  <tr>
						    <td>Split_sequences:sequences</td>
						    <td>Create_Sequence_Chunks:sequences</td>
						  </tr>
						
					</table>
				
			</div>
		</div>
		
		<!-- Coordinations -->
		
		<div class="fold">
			<div class="foldTitle">
		    Coordinations (11)
			</div>
		  <div class="foldContent" style="display: none;">
				
					<table class="simple">
					  <tr>
					    <th>Controller</th>
					    <th>Target</th>
					  </tr>
					
						
						  <tr>
						    <td>Create_Header_BlastReport</td>
						    <td>BlatOrBlast</td>
						  </tr>
						
						  <tr>
						    <td>BlatOrBlast</td>
						    <td>Read_BlastReport</td>
						  </tr>
						
						  <tr>
						    <td>BlatOrBlast</td>
						    <td>Read_OligosNotFound</td>
						  </tr>
						
						  <tr>
						    <td>DoBioMart</td>
						    <td>Read_BioMartReport</td>
						  </tr>
						
						  <tr>
						    <td>Create_Header_BioMartReport</td>
						    <td>DoBioMart</td>
						  </tr>
						
						  <tr>
						    <td>Create_Header_BlastReport</td>
						    <td>Filter_Sequences_For_Blast</td>
						  </tr>
						
						  <tr>
						    <td>Touch_OligosNotFound_File</td>
						    <td>BlatOrBlast</td>
						  </tr>
						
						  <tr>
						    <td>Create_Semaphore</td>
						    <td>Create_Header_BlastReport</td>
						  </tr>
						
						  <tr>
						    <td>Create_Semaphore</td>
						    <td>Touch_OligosNotFound_File</td>
						  </tr>
						
						  <tr>
						    <td>Create_Semaphore</td>
						    <td>Create_Header_BioMartReport</td>
						  </tr>
						
						  <tr>
						    <td>Create_Semaphore</td>
						    <td>Touch_BlastHitsFinished_File</td>
						  </tr>
						
					</table>
				
			</div>
		</div>
	
	


          
      
    </div>

  </div>

</div>

<div class="contribution_right_box">
  <div class="contribution_section_box" style= "font-size: 100%; padding: 0.7em 0.9em; font-weight: bold;">
    <p><img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[The type of workflow system that this Workflow is designed for.] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" /> Workflow Type</p>
    <p><a href="/content_types/1">Taverna 1</a></p>
  </div>

  <div class="contribution_section_box">
	<p class="heading">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[The Uploader is the person who uploaded/created this Workflow on myExperiment, and who manages it on myExperiment.] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Uploader
	</p>
	<p>
		<center>
			<div class="avatar" style="width: 66px">
  <a href="/users/622"><img alt="345?size=60x60" class="framed" src="/pictures/show/345?size=60x60" title="Wassinki" /></a>
  
	<span style="display:block;margin-top:2px;text-align:center;">
		
			<img alt="Nl" src="/images/famfamfam_flags/nl.png?1680607579" style="vertical-align:middle; margin-right: 0.3em;" title="header=[] body=[&lt;b&gt;Location: &lt;/b&gt;Enschede, Netherlands] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		

    
		  <a href="/users/622" title="header=[] body=[Wassinki] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Wassinki</a>
		

    
	</span>
  
</div>

		</center>
	</p>
</div>


  

  

<div class="contribution_section_box">
	<p class="heading">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[This specifies what rights users have when they download and use this Workflow.] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		License
		<a name="license"></a>
	</p>

  

    
    
    
    
    


    

    <!--

    <rdf:RDF xmlns="http://creativecommons.org/ns#"
        xmlns:dc="http://purl.org/dc/elements/1.1/"
        xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#">
    <Work rdf:about="">
       <dc:title>Mapping OligoNucleotides to an assembly</dc:title>
       <dc:date>2009-02-13 09:05:35 UTC</dc:date>
       <dc:description>&lt;div class=&quot;contribution_description&quot;&gt;
&lt;div class=&quot;contribution_description&quot;&gt;
&lt;p&gt;&lt;b&gt;Version info&lt;/b&gt;&lt;/p&gt;
&lt;p&gt;The former version of the workflow expected that results from BioMART only report transcripts when the query (the probe in our&lt;br /&gt;
case) are entirely encapsulated in an exon of that transcript. However, the BioMart service also returns transcripts when the query is not or only partially overlapping with an exon in the stretch on the assembly on which a transcript is defined. This resulted in too many oligos classified as having multiple transcripts or having multiple genes.&lt;br /&gt;
&lt;br /&gt;
&lt;b&gt;Workflow description&lt;/b&gt;&lt;/p&gt;
&lt;p&gt;We used RShell in the design process of a Zebrafish microarray&lt;br /&gt;
(supp. info Figure S1 and Figure S2). A microarray with 15k probes&lt;br /&gt;
of 60-mer oligonucleotides was designed on gene sequences from&lt;br /&gt;
Vega (&lt;a href=&quot;http://vega.sanger.ac.uk/Danio_rerio&quot;&gt;http://vega.sanger.ac.uk/Danio_rerio&lt;/a&gt;) and Ensembl&lt;br /&gt;
(&lt;a href=&quot;http://www.ensembl.org/Danio_rerio/&quot;&gt;http://www.ensembl.org/Danio_rerio/&lt;/a&gt;) that are also known&lt;br /&gt;
in the Zebrafish Information Network (&lt;a href=&quot;http://zfin.org/&quot;&gt;http://zfin.org&lt;/a&gt;) (for zebra&lt;br /&gt;
fish, the &lt;span class=&quot;caps&quot;&gt;VEGA&lt;/span&gt; set is not a subset of the Ensembl set) of the genome&lt;br /&gt;
&lt;span class=&quot;caps&quot;&gt;DNA&lt;/span&gt;-sequence assemblies and to judge the agreement that exists between&lt;br /&gt;
the different assembly annotations, we mapped the Vega-designed probes&lt;br /&gt;
onto the Ensembl assembly&lt;/p&gt;
&lt;p&gt;It first performs an alignment using the BioMoby Blat and Blast service provided by &lt;span class=&quot;caps&quot;&gt;&lt;span class=&quot;caps&quot;&gt;&lt;span class=&quot;caps&quot;&gt;WUR&lt;/span&gt;&lt;/span&gt;&lt;/span&gt; (&lt;a href=&quot;http://www.bioinformatics.nl/&quot;&gt;www.bioinformatics.nl&lt;/a&gt;). Next, for each hit, tries to find the corresponding transcripts and genes using a biomart webservice. The final task is an analysis task using RShell. It calculates for each oligo to which class it belongs:&lt;/p&gt;
&lt;p&gt;0 no hit&lt;br /&gt;
1 single hit, single transcript, single gene&lt;br /&gt;
2 multiple hits, single transcript, single gene, intron spanning&lt;br /&gt;
3 multiple hits, single transcript, single gene, possible intron spanning *&lt;br /&gt;
4 multiple hits, single transcript, single gene, no intron spanning&lt;br /&gt;
5 multiple hits, multiple transcripts, single gene, intron spanning&lt;br /&gt;
6 multiple hits, multiple transcripts, single gene, possible intron spanning *&lt;br /&gt;
7 multiple hits, multiple transcripts, single gene, no intron spanning&lt;br /&gt;
8 single hit, does not meet additional criteria &lt;strong&gt;**&lt;br /&gt;
&lt;/strong&gt; 9 multiple hits, single transcript, do not meet additional criteria **&lt;br /&gt;
10 multiple hits, multiple transcripts, do not meet additional criteria &lt;strong&gt;**&lt;br /&gt;
&lt;/strong&gt;11 multiple hits, multiple genes&lt;br /&gt;
12 no transcript found but hit(s) meet additional criteria **&lt;br /&gt;
13 no transcript found and hit(s) do not meet additional criteria &lt;strong&gt;**&lt;br /&gt;
&lt;/strong&gt;14 multiple hits, single transcript, single gene plus hit without transcript found and hits&lt;br /&gt;
meet additional criteria **&lt;br /&gt;
* Oligo below e-value cut-off 1e-12, but also intron spanning criteria met.&lt;br /&gt;
** Additional criteria: either e-value below 1e-12 or intron spanning.&lt;/p&gt;
&lt;p&gt;To run this workflow, a &lt;b&gt;certificate &lt;/b&gt;to access &lt;a href=&quot;http://www.bioinformatics.nl/&quot;&gt;www.bioinformatics.nl&lt;/a&gt; needs to installed (Some services use an &lt;span class=&quot;caps&quot;&gt;&lt;span class=&quot;caps&quot;&gt;&lt;span class=&quot;caps&quot;&gt;SSL&lt;/span&gt;&lt;/span&gt;&lt;/span&gt; connection). Look at the link below how to install this certificate.&lt;br /&gt;
&lt;br /&gt;
&lt;a href=&quot;../../../files/148&quot;&gt;http://www.myexperiment.org/files/148&lt;/a&gt;&lt;/p&gt;
&lt;p&gt;The myExperiment pack &lt;a href=&quot;../../../packs/45&quot;&gt;http://www.myexperiment.org/packs/45&lt;/a&gt; contains the workflow, the input and a test input. The whole input set is large. It takes about 6 hours on a 3 GHz Linux pc with 24 Gig &lt;span class=&quot;caps&quot;&gt;&lt;span class=&quot;caps&quot;&gt;RAM&lt;/span&gt;&lt;/span&gt;. The test input set can be run on almost any computer with Taverna and R installed. This set takes approximately 10 minutes.&lt;/p&gt;
&lt;/div&gt;
&lt;/div&gt;</dc:description>
       <dc:creator><Agent>
          <dc:title><a href="/users/622" title="header=[] body=[Wassinki] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Wassinki</a></dc:title>
       </Agent></dc:creator>
       <dc:rights><Agent>
          <dc:title><a href="/users/622" title="header=[] body=[Wassinki] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Wassinki</a></dc:title>
       </Agent></dc:rights>
       <dc:type rdf:resource="http://purl.org/dc/dcmitype/Data" />
       <dc:source rdf:resource="http://www.myexperiment.org/workflows/603.html"/>
       <license rdf:resource="http://creativecommons.org/licenses/by-sa/3.0/" />
    </Work>

    <License rdf:about="http://creativecommons.org/licenses/by-sa/3.0/">
      <permits rdf:resource="http://creativecommons.org/ns#Reproduction" />  <permits rdf:resource="http://creativecommons.org/ns#Distribution" />  <requires rdf:resource="http://creativecommons.org/ns#Notice" />  <requires rdf:resource="http://creativecommons.org/ns#Attribution" />  <permits rdf:resource="http://creativecommons.org/ns#DerivativeWorks" />  <requires rdf:resource="http://creativecommons.org/ns#ShareAlike" />
    </License>

    </rdf:RDF>

    -->

  

  <div class="contribution_currentlicense">
    <p>
      
        All versions of this Workflow are
      
      
        licensed under: <br/><br/><a rel="Copyright" href="/licenses/2" title="Creative Commons Attribution-Share Alike 3.0 Unported License"><img src="/images/by-sa.png" /></a>
      
    </p>
  </div>
	
</div>


  <script type="text/javascript">
    function showVersion(form) {
        var url = $("resource_versions").value;
        location.href = url;
        form.submit
    }
</script>
<style>
  .version_selection {
      font-size:smaller;
      text-align: center;
  }
</style>

<div class="contribution_section_box">
  <p class="heading">
    <img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[This box shows version 7 for this entry] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />    <span>Version 7 (latest)</span>
    <span class="count_text">(of 7)</span>
  </p>
  
      <form onsubmit="showVersion(this); return false;" style="text-align: right;">
        <p class="version_selection">
        View version:
        <select id="resource_versions" onchange="showVersion(this.form)">
          
              <option value="/workflows/603/versions/7" selected>
                7 (latest)
              </option>
          
              <option value="/workflows/603/versions/6" >
                6 
              </option>
          
              <option value="/workflows/603/versions/5" >
                5 
              </option>
          
              <option value="/workflows/603/versions/4" >
                4 
              </option>
          
              <option value="/workflows/603/versions/3" >
                3 
              </option>
          
              <option value="/workflows/603/versions/2" >
                2 
              </option>
          
              <option value="/workflows/603/versions/1" >
                1 (earliest)
              </option>
          
        </select>
        </p>
      </form>
  
</div>

  <div class="contribution_section_box">
	
	<p class="heading" style="margin-bottom: 0; padding-bottom: 0;">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[Which people/groups get the credit for this Workflow?] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Credits <span class="count_text">(2)</span>
		<a name="credits"></a>
	</p>
	<span style="font-size: 85%; color: #666666; display: block; padding: 0.3em 0.5em 0 0.5em; margin: 0; line-height: 1em;">(People/Groups)</span>
	
		<div style="font-size: 85%;">
			<ul class="items">
				
					<li>
						
							
								<span class="icon"><img alt="User" src="/images/famfamfam_silk/user.png?1680607579" /> </span> 
							
							<a href="/users/622" title="header=[] body=[Wassinki] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Wassinki</a>
						
					</li>
				
					<li>
						
							
								<span class="icon"><img alt="User" src="/images/famfamfam_silk/user.png?1680607579" /> </span> 
							
							<a href="/users/621" title="header=[] body=[Pieter Neerincx] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Pieter Neerincx</a>
						
					</li>
				
			</ul>
		</div>
	
	
	
	<p class="heading" style="margin-bottom: 0; padding-bottom: 0;">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[What other Workflows/Files is this one based on?] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Attributions <span class="count_text">(6)</span>
		<a name="attributions"></a>
	</p>
	<span style="font-size: 85%; color: #666666; display: block; padding: 0.3em 0.5em 0 0.5em; margin: 0; line-height: 1em;">(Workflows/Files)</span>
	
		<div style="font-size: 85%;">
			<p>
				<ul class="items">
					
						<li>
							
								
									
										<span class="icon"><img alt="Workflow" src="/images/redmond_studio/applications_16.png?1680607579" /> </span> 
									 
									<a href="/workflows/532">Blat against ENSEMBLE Danio_rerio_Genome</a>
								
						  
						</li>
					
						<li>
							
								
									
										<span class="icon"><img alt="Workflow" src="/images/redmond_studio/applications_16.png?1680607579" /> </span> 
									 
									<a href="/workflows/531">BlatBlastCombi</a>
								
						  
						</li>
					
						<li>
							
								
									
										<span class="icon"><img alt="Workflow" src="/images/redmond_studio/applications_16.png?1680607579" /> </span> 
									 
									<a href="/workflows/530">Blast against ENSEMBLE Danio_rerio_Genome</a>
								
						  
						</li>
					
						<li>
							
								
									
										<span class="icon"><img alt="Workflow" src="/images/redmond_studio/applications_16.png?1680607579" /> </span> 
									 
									<a href="/workflows/560">AppendToFile</a>
								
						  
						</li>
					
						<li>
							
								
									
										<span class="icon"><img alt="Blob" src="/images/redmond_studio/documents_16.png?1680607579" /> </span>
									 
									<a href="/files/164">Test Input for Mapping oligonucleotides to an assembly</a>
								
						  
						</li>
					
						<li>
							
								
									
										<span class="icon"><img alt="Blob" src="/images/redmond_studio/documents_16.png?1680607579" /> </span>
									 
									<a href="/files/163">Input for Mapping oligonucleotides to an assembly</a>
								
						  
						</li>
					
				</ul>
			</p>
		</div>
	
	
	
	
</div>


  


  <div class="contribution_section_box">
	<p class="heading">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[The tags for this Workflow entry.&lt;br/&gt;Note(1): if this Workflow has many versions, the tags apply to ALL versions in this entry.&lt;br/&gt;Note(2): the size of the tags show how popular they are on myExperiment.] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Tags <span id="tags_box_header_tag_count_span" class="count_text">(7)</span>
		<a name="tags"></a>
	</p>

	<script type="text/javascript">
  function showAllTags() {
    Element.show('all_tags');
	  Element.hide('user_tags');
	  return false;
  }
  
  function showUsersTags() {
  	Element.hide('all_tags');
	  Element.show('user_tags');
	  return false;
  }
	
	function flip_tags_mode()
	{
		if( $('toggle_tags_mode').checked )
		  showUsersTags();
	  else
		  showAllTags();
	}
</script>

<div id="tags_inner_box">
	
		<div class="box_currentuser_specific" style="padding: 0.3em 0.2em; margin: 0.6em 0.4em;">
		  <label for="toggle_tags_mode" style="font-size: 85%; font-weight: bold;"><input id="toggle_tags_mode" name="toggle_tags_mode" onclick="flip_tags_mode();" style="margin-right: 0.3em;" type="checkbox" value="1" />Uploader tags</label>
		</div>
		<div id="all_tags">
			
				<div class="hTagcloud">
  <ul class="popularity">
    <li><a href='/tags/535' class='v-popular'>biomart</a></li>
<li> | </li>
    <li><a href='/tags/49' class='vvvv-popular'>BLAST</a></li>
<li> | </li>
    <li><a href='/tags/966' class='popular'>blat</a></li>
<li> | </li>
    <li><a href='/tags/144' class='vv-popular'>ensembl</a></li>
<li> | </li>
    <li><a href='/tags/305' class='vv-popular'>microarray</a></li>
<li> | </li>
    <li><a href='/tags/750' class='vv-popular'>r</a></li>
<li> | </li>
    <li><a href='/tags/1083' class='popular'>rshell</a></li>
  </ul>
</div>

			
		</div>
		<div id="user_tags"  style="display: none;">
			
				<div class="hTagcloud">
  <ul class="popularity">
    <li><a href='/tags/535' class='v-popular'>biomart</a></li>
<li> | </li>
    <li><a href='/tags/49' class='vvvv-popular'>BLAST</a></li>
<li> | </li>
    <li><a href='/tags/966' class='popular'>blat</a></li>
<li> | </li>
    <li><a href='/tags/144' class='vv-popular'>ensembl</a></li>
<li> | </li>
    <li><a href='/tags/305' class='vv-popular'>microarray</a></li>
<li> | </li>
    <li><a href='/tags/750' class='vv-popular'>r</a></li>
<li> | </li>
    <li><a href='/tags/1083' class='popular'>rshell</a></li>
  </ul>
</div>

			
		</div>
	
</div>
	
	
	
	
		<p><small><b><a href="/session/new">Log in</a></b> to add Tags</small></p>
	
</div>


  
<div class="contribution_section_box">
	<p class="heading">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[All the Groups that this item has been shared with] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Shared with Groups <span class="count_text">(0)</span>
	</p>
	
		<p class="none_text">None</p>
	
</div>

  <div class="contribution_section_box" id="in_packs_box">
  
	<p class="heading">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[All the Packs that this item (or versions of this item [[if available]]) is referred to in] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Featured In Packs  <span class="count_text">(2)</span>
		<a name="featured_in_packs"></a>
	</p>
	
		<div style="font-size: 85%;">
			<ul class="items">
				
					<li>
						
							<span class="icon"><img alt="Pack" src="/images/manhattan_studio/folder-closed_16.png?1680607579" /> </span>
							<a href="/packs/45">Package: mapping oligonucleotides to an assembly</a>
						
					</li>
				
					<li>
						
							<span class="icon"><img alt="Pack" src="/images/manhattan_studio/folder-closed_16.png?1680607579" /> </span>
							<a href="/packs/47">useful_for_sysmo</a>
						
					</li>
				
			</ul>
		</div>
	
	
	
		<p><small><b><a href="/session/new">Log in</a></b> to add to one of your Packs</small></p>
	
</div>


  <div class="contribution_section_box">
	
	<p class="heading" style="margin-bottom: 0; padding-bottom: 0;">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[Shows files and workflows that have attributed this Workflow.] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Attributed By <span class="count_text">(0)</span>
	  <a name="attributed_by"></a>
	</p>
	<span style="font-size: 85%; color: #666666; display: block; padding: 0.3em 0.5em 0 0.5em; margin: 0; line-height: 1em;">(Workflows/Files)</span>
	
		
		<p class="none_text">None</p>
	
	
</div>


  
<div class="contribution_section_box">
	<p class="heading">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[The users who have added this Workflow to their favourites] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Favourited By <font class="count_text">(1)</font>
		<a name="favourited_by"></a>
	</p>
	
	
		<div style="font-size: 85%;">
			<ul class="items">
				
					<li>
						<span class="icon"><img alt="User" src="/images/famfamfam_silk/user.png?1680607579" /> </span>
						<a href="/users/13" title="header=[] body=[Katy Wolstencroft] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Katy Wolstencroft</a>
						
					</li>
				
			</ul>
		</div>
	
	
	
</div>

  

<div class="contribution_section_box">
	<p class="heading" style="margin-bottom: 0.3em;">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[Viewing / download counts are grand totals for this Workflow.&lt;br/&gt;Click &#39;see breakdown&#39; to view details.] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Statistics
	</p>
	
	<div class="stats_box"> 
		<p style="font-size: 85%;">
  10021 viewings
</p>

	<p style="font-size: 85%;">
	  3129 downloads
	</p>

<p style="margin-top: 0.2em;">
	<small>[ <a href="/workflows/603/statistics">see breakdown</a> ]</small>
</p>
		
		<div class="fold" style="padding-top: 0.6em;">
			<div class="foldTitle">More</div>
			<div class="foldContent" style="display: none; text-align: left;">
				
					<p>
						<a href="#credits">2 credits</a> <!-- won't work for type=='group'; needs anchor called '#group_creditations' -->
					</p>
					
					<p>
						<a href="#attributions">6 attributions</a>
					</p>
				
				
				
					<p>
			      <a href="#tags">7 tags</a>
			    </p>
				
				
				
					<p>
			      <a href="#favourited_by">1 favourite</a>
			    </p>
				

				
					<p>
						<a href="#citations">0 citations</a>
			    </p>
				
				
				
					<p>
			      <a href="#reviews">0 reviews</a>
			    </p>
				
				
				
			    <p>
			    	<a href="#comments">0 comments</a>
			    </p>
				
			</div>
		</div>
		
		
		
		
	</div>
</div>

</div>

<div class="clearer">&nbsp;</div>

<!-- BEGIN tabs -->

<br/>

<div id="tabsContainer" class="tabsContainer"></div>

<div class="tabContainer">
  <div class="tabTitle">Citations (0)</div>
  <div class="tabContent">

    <a name="citations"></a>
    
      <p><i>None</i></p>
    
		
		<br/>

    

  </div>
</div>


<div class="tabContainer">
  <div class="tabTitle">Version History</div>
  <div class="tabContent">
    <a name="version_history"></a>
    <p>In chronological order:</p>

<ol class="version_history">
  
    <li>
      <div class="title">
        <a href="/workflows/603/versions/1">Mapping oligonucleotides to an assembly</a>
      </div>
      <div class="metadata">

        <p>
          Created
          
            by <a href="/users/622" title="header=[] body=[Wassinki] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Wassinki</a>
          
          on Thursday 11 December 2008 12:11:59 (UTC)
        </p>

        
          <p>
            Last edited
            
              by <a href="/users/622" title="header=[] body=[Wassinki] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Wassinki</a>
            
            on Thursday 11 December 2008 12:25:55 (UTC)
          </p>
        

        
      </div>
    </li>
  
    <li>
      <div class="title">
        <a href="/workflows/603/versions/2">Mapping OligoNucleotides to an assembly</a>
      </div>
      <div class="metadata">

        <p>
          Created
          
            by <a href="/users/622" title="header=[] body=[Wassinki] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Wassinki</a>
          
          on Thursday 11 December 2008 12:26:50 (UTC)
        </p>

        
          <p>
            Last edited
            
              by <a href="/users/622" title="header=[] body=[Wassinki] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Wassinki</a>
            
            on Thursday 11 December 2008 12:32:04 (UTC)
          </p>
        

        
      </div>
    </li>
  
    <li>
      <div class="title">
        <a href="/workflows/603/versions/3">Mapping OligoNucleotides to an assembly</a>
      </div>
      <div class="metadata">

        <p>
          Created
          
            by <a href="/users/622" title="header=[] body=[Wassinki] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Wassinki</a>
          
          on Friday 19 December 2008 09:41:13 (UTC)
        </p>

        

        
          <p>
            <div>Revision comment:</div>
            <div class="comment">
              <p>
&lt;meta />
&lt;meta />
&lt;meta />
&lt;meta />
&lt;link />&lt;!--[if gte mso 9]>&lt;xml>
&lt;w:WordDocument>
&lt;w:View>Normal&lt;/w:View>
&lt;w:Zoom>0&lt;/w:Zoom>
&lt;w:PunctuationKerning />
&lt;w:ValidateAgainstSchemas />
&lt;w:SaveIfXMLInvalid>false&lt;/w:SaveIfXMLInvalid>
&lt;w:IgnoreMixedContent>false&lt;/w:IgnoreMixedContent>
&lt;w:AlwaysShowPlaceholderText>false&lt;/w:AlwaysShowPlaceholderText>
&lt;w:Compatibility>
&lt;w:BreakWrappedTables />
&lt;w:SnapToGridInCell />
&lt;w:WrapTextWithPunct />
&lt;w:UseAsianBreakRules />
&lt;w:DontGrowAutofit />
&lt;/w:Compatibility>
&lt;w:BrowserLevel>MicrosoftInternetExplorer4&lt;/w:BrowserLevel>
&lt;/w:WordDocument>
&lt;/xml>&lt;![endif]-->&lt;!--[if gte mso 9]>&lt;xml>
&lt;w:LatentStyles DefLockedState="false" LatentStyleCount="156">
&lt;/w:LatentStyles>
&lt;/xml>&lt;![endif]--><p>----------------------------------------</p>
<p>The newest version only takes into account the probes that have blast hits that map on exons. The BioMart sub workflow has been modified to do this by adding an extra BioMart processor and a beanshell processor to filter those blast hits that map on exons.</p>
<p>----------------------------------------</p>
<p>This workflow maps the input oligo set to an assembly.&lt;o:p>&lt;/o:p></p>
<p>It first performs an alignment using the BioMoby Blat and Blast service provided by <span class="caps">WUR</span> (<a href="http://www.bioinformatics.nl/" rel="nofollow">www.bioinformatics.nl</a>). Next, for each hit, tries to find the corresponding transcripts and genes using a biomart webservice. The final task is an analysis task using RShell. It calculates for each oligo to which class it belongs:&lt;o:p>&lt;/o:p></p>
<p>1 single hit<br />
2-4 multiple hits single transcript<br />
5-7 mulitple hits multiple transcripts<br />
8 single hit, discarded<br />
9 multiple hits single transcript, discarded<br />
10 multiple transcripts, discarded* <br />
11 multi gene, discarded <br />
12 no transcript <br />
13 no transcript, discarded<br />
* classified on the criteria intron spanning only, possible intron spanning and no intron spanning.<br />
* hit(s) do not meet high stringency threshold<br />
* no transcript found but hit(s) meet high stringency threshold.&lt;o:p>&lt;/o:p></p>
<p>To run this workflow, a <b>certificate </b>to access <a href="http://www.bioinformatics.nl/" rel="nofollow">www.bioinformatics.nl</a> needs to installed (Some services use an <span class="caps">SSL</span> connection). Look at the link below how to install this certificate.<br />
<br />
<a href="../../../files/148" rel="nofollow">http://www.myexperiment.org/files/148</a>&lt;o:p>&lt;/o:p></p>
<p>The myExperiment pack http://www.myexperiment.org/packs/45 contains the workflow, the input and a test input. The whole input set is large. It takes about 6 hours on a 3 GHz Linux pc with 24 Gig RAM. The test input set can be run on almost any computer with Taverna and R installed. This set takes approximately 10 minutes.&lt;o:p>&lt;/o:p></p>
<p>&lt;o:p>&nbsp;&lt;/o:p></p>
<p class="MsoNormal">&lt;o:p>&nbsp;&lt;/o:p></p>
</p>
            </div>
          </p>
        
      </div>
    </li>
  
    <li>
      <div class="title">
        <a href="/workflows/603/versions/4">Mapping OligoNucleotides to an assembly</a>
      </div>
      <div class="metadata">

        <p>
          Created
          
            by <a href="/users/622" title="header=[] body=[Wassinki] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Wassinki</a>
          
          on Tuesday 03 February 2009 15:27:18 (UTC)
        </p>

        
          <p>
            Last edited
            
              by <a href="/users/622" title="header=[] body=[Wassinki] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Wassinki</a>
            
            on Tuesday 03 February 2009 15:33:01 (UTC)
          </p>
        

        
          <p>
            <div>Revision comment:</div>
            <div class="comment">
              <p>The former version of the workflow expected that results from BioMART only report transcripts when the query (the probe in our case) are entirely encapsulated in an exon of that transcript. However, the BioMart service also returns transcripts when the query is not or only partially overlapping with an exon in the stretch on the assembly on which a transcript is defined. This resulted in too many oligos classified as having multiple transcripts or having multiple genes.<br />
&nbsp;</p>
            </div>
          </p>
        
      </div>
    </li>
  
    <li>
      <div class="title">
        <a href="/workflows/603/versions/5">Mapping OligoNucleotides to an assembly</a>
      </div>
      <div class="metadata">

        <p>
          Created
          
            by <a href="/users/622" title="header=[] body=[Wassinki] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Wassinki</a>
          
          on Wednesday 04 February 2009 08:25:36 (UTC)
        </p>

        
          <p>
            Last edited
            
              by <a href="/users/622" title="header=[] body=[Wassinki] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Wassinki</a>
            
            on Wednesday 04 February 2009 08:26:04 (UTC)
          </p>
        

        
      </div>
    </li>
  
    <li>
      <div class="title">
        <a href="/workflows/603/versions/6">Analysing workflows</a>
      </div>
      <div class="metadata">

        <p>
          Created
          
            by <a href="/users/622" title="header=[] body=[Wassinki] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Wassinki</a>
          
          on Friday 13 February 2009 09:03:37 (UTC)
        </p>

        

        
      </div>
    </li>
  
    <li>
      <div class="title">
        <a href="/workflows/603/versions/7">Mapping OligoNucleotides to an assembly</a>
      </div>
      <div class="metadata">

        <p>
          Created
          
            by <a href="/users/622" title="header=[] body=[Wassinki] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Wassinki</a>
          
          on Friday 13 February 2009 09:05:35 (UTC)
        </p>

        
          <p>
            Last edited
            
              by <a href="/users/622" title="header=[] body=[Wassinki] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Wassinki</a>
            
            on Friday 13 February 2009 09:08:23 (UTC)
          </p>
        

        
      </div>
    </li>
  
</ol>


  </div>
</div>



<!-- END tabs -->

<br/>
<br/>

<div id="ratingsBox">
  <h2>
	<span class="icon"><img alt="Reviews" src="/images/famfamfam_silk/report.png?1680607579" /> </span>	<span style="vertical-align: middle;"><a href="/workflows/603/reviews">Reviews</a></span>
	<span class="count_text" style="vertical-align: middle;">(0)</span>
</h2>

<div class="reviewsBox">

	
		<p><i>No reviews yet</i></p>
		<a name="reviews"></a>
		<p style="margin-top: 0.6em;">
			<strong><a href="/workflows/603/reviews/new">Be the first to review!</a></strong>
		</p>
	
	
</div>

</div>

<br/>
<br/>

<div id="commentsBox">
  <script src="/javascripts/ckeditor/ckeditor.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/widgets/tooltip.js?1680607579" type="text/javascript"></script>

<h2>
	<span class="icon"><img alt="Comments" src="/images/famfamfam_silk/comments.png?1680607579" /> </span>	<span style="vertical-align: middle;">Comments</span>
	<span class="count_text" style="vertical-align: middle;">(0)</span>
</h2>

<ul class="sectionIcons">
  <li><span class="icon"><a href="http://www.myexperiment.org/workflows/603/comments/timeline"><img alt="Timeline" src="/images/famfamfam_silk/timeline_marker.png?1680607579" /> View Timeline</a></span></li>
</ul>

<div class="commentsBox">
	
	
		<p><i>No comments yet</i></p>
		<a name="comments"></a>
		
			<p style="margin-top: 0.6em;"><strong><a href="/session/new">Log in</a> to make a comment </strong></p>
		
	
	
	<br/>
	
	

</div>


</div>

<!-- Other workflows that use a service this workflow users -->

<br />
<br />

<h2>
  <span class="icon"><img alt="Workflow" src="/images/redmond_studio/applications_16.png?1680607579" /> </span>  <span style="vertical-align: middle;">Other workflows that use similar services</span>
  <span class="count_text" style="vertical-align: middle;">(0)</span>
</h2>




    <p>There are no workflows in myExperiment that use similar services to this Workflow.</p>



<!-- OpenURL context object -->

<span class="Z3988" title="url_ver=Z39.88-2004&amp;url_ctx_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Actx&amp;ctx_ver=Z39.88-2004&amp;ctx_tim=2023-08-16T01%3A46%3A19%2B00%3A00&amp;ctx_id=&amp;ctx_enc=info%3Aofi%2Fenc%3AUTF-8&amp;rft.title=Mapping+OligoNucleotides+to+an+assembly&amp;rft.date=2009-02-13&amp;rft.au=Wassinki&amp;rft.genre=unknown&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Adc"></span>
      </div>
    </div>




            <div id="ft">
          <div class="container">
  <div class="row">
    <div class="col-md-4 text-left">
      Funded by:
      <a href="http://biovel.eu">BioVeL</a>,
      <a href="http://www.epsrc.ac.uk/">EPSRC</a>,
      <a href="http://jisc.ac.uk/">JISC</a>,
      <a href="http://www.microsoft.com">Microsoft</a>

      <br/><br/>

      Affiliates:
      <a href="http://www.mygrid.org.uk">myGrid</a>,
      <a href="http://www.taverna.org.uk/">Taverna</a>,
      <a href="http://www.biocatalogue.org">BioCatalogue</a>
    </div>

    <div class="col-md-4 text-center">
      <a href="/about">About</a> |
      <a href="/privacy">Privacy</a> |
      <a href="http://wiki.myexperiment.org/index.php/Papers">Publications</a> |
      <a href="/feedback">Contact / Feedback</a> |
      <a href="http://lists.nongnu.org/mailman/listinfo/myexperiment-discuss">Mailing List</a>
    </div>

    <div class="col-md-4 text-right">
      Icons: <a href="http://www.famfamfam.com/lab/icons/silk/">Silk icon set 1.3</a><br/><br/>
      
    </div>

  </div>
  <div class="row">
    <div class="col-md-12 text-center">
      Copyright &#169; 2007 &ndash; 2018
      <a href="http://www.manchester.ac.uk/">The University of Manchester</a>
      and
      <a href="http://www.soton.ac.uk/">University of Southampton</a>
    </div>
  </div>
</div>

        </div>
    



<script type="text/javascript">
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
        (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
        m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-1498417-4', 'auto');
    ga('set', 'anonymizeIp', true);
    ga('send', 'pageview');
</script>




</body>
</html>

