<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
    "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <meta http-equiv="content-type" content="text/html;charset=UTF-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1">
  
  <meta name="google-site-verification" content="3hq0Cfhw5_-EaaNMum0ETX847sTl7jeQ3ZNAla4a6UQ" />
  
  <title>myExperiment &#45; Workflows &#45; get promoter region + operate on genomic intervals (Eleni) [Taverna 2 Workflow]</title>
  <link rel="alternate" href="http://www.myexperiment.org/workflows/3886.rdf" type="application/rdf+xml" title="RDF+XML" />
  <link rel="alternate" href="http://www.myexperiment.org/workflows/3886.xml" type="application/xml" title="REST XML" />
  <link rel="shortcut icon" href="/favicon.ico" type="image/x-icon"/>
  <link href="/stylesheets/acts_as_taggable_stylesheet.css?1680607579" media="screen" rel="stylesheet" type="text/css" />
<link href="/stylesheets/star_rating.css?1680607579" media="screen" rel="stylesheet" type="text/css" />
<link href="/stylesheets/gadgets.css?1680607579" media="screen" rel="stylesheet" type="text/css" />
<link href="/stylesheets/misc_skinning.css?1680607579" media="screen" rel="stylesheet" type="text/css" />
<link href="/stylesheets/pagination.css?1680607579" media="screen" rel="stylesheet" type="text/css" />
<link href="/stylesheets/bootstrap.css?1680607579" media="screen" rel="stylesheet" type="text/css" />
<link href="/stylesheets/navbar.css?1680607579" media="screen" rel="stylesheet" type="text/css" />
<link href="/stylesheets/styles.css?1680607579" media="screen" rel="stylesheet" type="text/css" />
  <script src="/javascripts/jquery-1.11.1.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/jquery-noconflict.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/prototype-1.7.2-patched.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/effects.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/dragdrop.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/controls.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/boxover.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/bootstrap.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/tabs.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/folds.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/application.js?1680607579" type="text/javascript"></script>

  
  






  
  
    <script src='//www.google.com/recaptcha/api.js'></script>
  
</head>
<body>



<div class="navbar navbar-myexp" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      
          <a id="logo" href="/">
            <img alt="Logo" class="hidden-sm" src="/images/logo.png?1680607579" />
            <img alt="Logo_cropped" class="visible-sm" src="/images/logo_cropped.png?1680607579" />
          </a>
      
    </div>
    <div class="collapse navbar-collapse">
      <ul class="nav navbar-nav">
        
          <ul class="nav navbar-nav">
  
    <li >
      <a href="/home">Home</a>    </li>
  
    <li >
      <a href="/users">Users</a>    </li>
  
    <li >
      <a href="/groups">Groups</a>    </li>
  
    <li class="active">
      <a href="/workflows">Workflows</a>    </li>
  
    <li >
      <a href="/files">Files</a>    </li>
  
    <li >
      <a href="/packs">Packs</a>    </li>
  
</ul>

        
        
            <form action="/search" class="navbar-form navbar-left" id="search" method="get">
    <div class="input-group">
      <input class="form-control" id="query" name="query" placeholder="Search" type="text" />
      <span class="input-group-btn">
        
        <input class="btn btn-default" name="commit" type="submit" value="Go" />
        </span>
    </div>
</form>

        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
            <li><a href="/session/new?return_to=%2Fworkflows%2F3886.html">Log in</a></li>
    <li><a href="/users/new">Register</a></li>


      </ul>
    </div><!--/.nav-collapse -->
  </div>
  
</div>




    <div class="container" id="main">
      <div id="myexp_breadcrumbs_bar">
        <ol class="breadcrumb">
  <li><a href="/home">Home</a></li>
  


  <li><a href="/workflows">Workflows</a></li>



  
    <li>get promoter region + operate on genomic intervals</li>
  


  <div style="float: right"></div>
</ol>

      </div>
      <div id="myexp_content">
        


        



<h1 class="contribution_title">
  get promoter region + operate on genomic intervals
</h1>

<div class="contribution_aftertitle">
	<b>Created:</b> 2013-11-04 19:28:07
	
</div>
<ul class="sectionIcons">
  
      <li><span class="icon"><a href="http://www.myexperiment.org/workflows/3886/download/get_promoter_region___operate_on_genomic_intervals-v1.t2flow?version=1"><img alt="Download Workflow (for version 1)" src="/images/redmond_studio/arrow-down_16.png?1680607579" /> Download Workflow</a></span></li>
      
  
      
</ul>


<div class="contribution_left_box">

  <div class="contribution_version_box">



    <div class="contribution_version_inner_box">

      
    <div class="contribution_description">
      <p>
Finds the overlap between two datasets which contain genomic information (e.g. [gene id], chromosome name, gene start, gene end), plus some statistics. Returns rows of file_1 which overlap with the second file. A kolmogorov smirnov test is applied between the list that overlaps and the one that does not. 
NOTE: The library(GenomicRanges) is a prerequisite for this workflow
</p>


    </div>


      <h3>
        <img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[This section shows the preview of this version of the Workflow (click on the image to expand)] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
        Preview
      </h3>

      

          <div class="contribution_preview" style="width: 100%;">
            <p style="text-align: center">
              <a href="/workflows/3886/versions/1/previews/full" onclick="window.open(this.href);return false;"><img alt="Medium" src="/workflows/3886/versions/1/previews/medium" /></a>
            </p>
            
                <p style="text-align: right; color: #666666;">
                  <a href="/workflows/3886/versions/1/previews/svg">Download as scalable diagram (SVG)</a>
                </p>
            

            
          </div>
      



      <h3>
        <img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[This section provides options for running this version of the Workflow] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
        Run
      </h3>

      
          
              <div style="margin: 0 0.5em;">
	<h4>
		Run this Workflow in the Taverna Workbench...
	</h4>
	
	<div class="option_box">
		
		<p style="text-align: center;">
			<b>Option 1:</b>
		</p>
		
		<p style="font-size: 93%; text-align: center;">
			Copy and paste this link into File > 'Open workflow location...'<br/>
			<a href="http://www.myexperiment.org/workflows/3886/download?version=1">http://www.myexperiment.org/workflows/3886/download?version=1</a><br/>
			<small>[ <a href="#" onclick="Effect.toggle(&quot;run_taverna_more_box&quot;,'blind',{duration:0.3});; return false;">More Info<img alt="Expand" src="/images/folds/unfold.png?1680607579" style="margin-left: 0.3em; vertical-align: middle;" /></a> ]</small>
		</p>
		
		<div id="run_taverna_more_box" style="display: none;">
			<p style="font-size: 85%; margin-top: 0.5em; text-align: center;">
				Taverna is available from <a href="http://taverna.sourceforge.net/" onclick="window.open(this.href);return false;">http://taverna.sourceforge.net/</a>
			</p>
			
			<p style="text-align: center; margin-top: 0.5em; font-size: 85%;">
				If you are having problems downloading it in Taverna, you may need to provide your username and password in the URL so that Taverna can access the Workflow: <br/>
				
					<b>Replace <font color="#990000">http://</font> in the link above with <font color="#990000">http://yourusername:yourpassword@</font></b>
				
			</p>
		</div>
	
	</div>
</div>


          
      

      <br/>

      <h3>
        <img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[This section shows the internal components of this workflow version] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
        Workflow Components
      </h3>

      
          
              
	    <!-- Authors -->
        <div class="fold">
      <div class="foldTitle">
        <img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[This is the author information extracted from the workflow version] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
        Authors (1)
      </div>
      <div class="foldContent" style="display: none;">
                  <table class="simple">
                          <tr>
                <td>Eleni Mina</td>
              </tr>
            
          </table>
        
      </div>
    </div>
    
    <!-- Titles -->
        <div class="fold">
      <div class="foldTitle">
        <img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[These are the descriptive titles embedded within the workflow version] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
        Titles (1)
      </div>
      <div class="foldContent" style="display: none;">
                  <table class="simple">
                          <tr>
                <td>get promoter region + operate on genomic intervals</td>
              </tr>
            
          </table>
        
      </div>
    </div>

    <!-- Descriptions -->
        <div class="fold">
      <div class="foldTitle">
        <img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[These are the descriptions embedded within the workflow version] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
        Descriptions (1)
      </div>
      <div class="foldContent" style="display: none;">
                  <table class="simple">
                          <tr>
                <td>
Finds the overlap between two datasets which contain genomic information (e.g. [gene id], chromosome name, gene start, gene end), plus some statistics. Returns rows of file_1 which overlap with the second file. A kolmogorov smirnov test is applied between the list that overlaps and the one that does not. 
NOTE: The library(GenomicRanges) is a prerequisite for this workflow
</td>
              </tr>
            
          </table>
        
      </div>
    </div>
        
    <!-- Dependencies -->
        <div class="fold">
      <div class="foldTitle">
        <img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[These are the listed dependencies of the workflow] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
        Dependencies (0)
      </div>
      <div class="foldContent" style="display: none;">
        
          <p class="none_text">None</p>
        
      </div>
    </div>
        
    <!-- Sources -->
        <div class="fold">
      <div class="foldTitle">
        Inputs (11)
      </div>
      <div class="foldContent" style="display: none;">
                  <table class="simple">
            <tr>
              <th>Name</th>
              <th>Description</th>
            </tr>
          
                          <tr>
                <td><b>workingdir</b></td>
                <td>
                  
                    
                      working directory                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>biomartfile</b></td>
                <td>
                  
                    
                      genomic information for the gene file.                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>epigen_file</b></td>
                <td>
                  
                    
                      File with epigenetic information                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>epigen_file_end</b></td>
                <td>
                  
                    
                      column for the end coordinate for each row 
in the epigenetic file info                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>epigen_file_start</b></td>
                <td>
                  
                    
                      column for the start coordinate for each row 
in the epigenetic file info                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>epigen_file_chr</b></td>
                <td>
                  
                    
                      chromosome column at the epigenetic file                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>gene_chr</b></td>
                <td>
                  
                    
                      column for the gene chromosome coordinate

                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>upstream</b></td>
                <td>
                  
                    
                      number of base pairs for computing a promoter region 
upstream the transcriptional start site (TSS)                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>downstream</b></td>
                <td>
                  
                    
                      number of base pairs for computing a promoter region 
downstream the transcriptional start site (TSS)                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>overlap</b></td>
                <td>
                  
                    
                      overlap parameter (in base pairs) for computing overlapping regions	                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>gene_expr_file</b></td>
                <td>
                  
                    
                      file with the gene expression values                      
                    
                  
                </td>
              </tr>
            
          </table>
        
      </div>
    </div>
      
    <!-- Processors -->
        <div class="fold">
      <div class="foldTitle">
        Processors (6)
      </div>
      <div class="foldContent" style="display: none;">
                  <table class="simple">
            <tr>
              <th>Name</th>
              <th>Type</th>
              <th>Description</th>
            </tr>
          
                          <tr>
                <td><b>read_files</b></td>
                <td>rshell</td>
                <td>
                  component that reads the epigenetic file and the file that was mapped on the genome                  <div><h3>Script</h3><pre>setwd(dir)

epigen_file<-read.table(epigenetic_file, header=TRUE, sep="\t")
epigen_file[,2]<-gsub("chr","",epigen_file[,2])

genes<-read.table(biomart,header=TRUE, sep="\t")

#genes_pvalues<-read.table(gene_pvals,header=TRUE)
#sep="\t"
print("filesread")</pre></div><div><h3>R Server</h3>localhost:6311</div>                </td>
              </tr>
                          <tr>
                <td><b>transform_files_to_gen_ranges</b></td>
                <td>rshell</td>
                <td>
                  transform the files to genomic ranges using the R package GenomicRanges to compute overlapping regions                  <div><h3>Script</h3><pre>library(GenomicRanges)
#transform each file in GRanges format

#t1 <- GRanges(seqnames = genes[,gene_chr], 
	#	ranges = IRanges(genes[,gene_start], genes[,gene_end] , names = genes$entrezgene))

t1 <- GRanges(seqnames = genes$chromosome_name, ranges = IRanges(genes$promoter_start, genes$promoter_end , names = genes$entrezgene))

t2 <- GRanges(seqnames = epigen_file[,inter_chr],ranges = IRanges(epigen_file[,inter_start], epigen_file[,inter_end]))


print("t1_t2_done")
#t2 <- GRanges(seqnames = cpg$chrom, 
	#	ranges = IRanges(cpg$chromStart, cpg$chromEnd))
</pre></div><div><h3>R Server</h3>localhost:6311</div>                </td>
              </tr>
                          <tr>
                <td><b>calculate_overlaps</b></td>
                <td>rshell</td>
                <td>
                  compute overlaps between the two genomic files                  <div><h3>Script</h3><pre>ov1<-findOverlaps(t1,t2,minoverlap=overlap)
print("ovdone")

genes_pvalues<-read.table(gene_expr_file,header=TRUE, sep="\t")

idx<-match(genes$entrezgene[as.matrix(ov1)[,1]],genes_pvalues$gene_id )

overlapping_genes<-genes$entrezgene[as.matrix(ov1)[,1]]
non_overlapping_genes<-genes$entrezgene[-(as.matrix(ov1)[,1])]
list_non_overlapping<-unique(non_overlapping_genes)
list_overlapping<-unique(overlapping_genes)

idx_com<-match(list_non_overlapping,list_overlapping)

idx_non_overlapping_without_overlapping_idx<-which(is.na(idx_com)==TRUE)
list_non_overlapping_new<-list_non_overlapping[idx_non_overlapping_without_overlapping_idx]

idx_overlapping<- match(list_overlapping, genes_pvalues$gene_id)
overlapping_genes_pvals<-genes_pvalues[idx_overlapping,]
overlapping_genes_pvals_out<- paste("overlapping_genes",Sys.time(),".txt",sep="")

idx_non_overlapping<- match(list_non_overlapping_new, genes_pvalues$gene_id)
non_overlapping_genes_pvals<-genes_pvalues[idx_non_overlapping,]
non_overlapping_genes_pvals_out<-paste("non_overlapping_genes",Sys.time(),".txt",sep="")

write.table(overlapping_genes_pvals,overlapping_genes_pvals_out,col.names=TRUE, row.names=FALSE, sep="\t")
write.table(non_overlapping_genes_pvals,non_overlapping_genes_pvals_out ,col.names=TRUE,row.names=FALSE,sep="\t")

</pre></div><div><h3>R Server</h3>localhost:6311</div>                </td>
              </tr>
                          <tr>
                <td><b>calculate_promoter_region</b></td>
                <td>rshell</td>
                <td>
                  For each one of the genes we compute a promoter region according to the prespecified values of the variables upstream_bp and downstream_bp the transcription start site. In this component we take into account the direction that a gene is transcribed. The variable &quot;strand&quot; is responsible for that.                   <div><h3>Script</h3><pre>#a<-unique(biomart$entrezgene)
setwd(workdir)
biomart<-read.table(genes,header=TRUE, sep="\t")
mat<-matrix(NA, nrow=dim(biomart)[1], ncol=10)
colnames(mat)<-c(colnames(biomart),"promoter_start","promoter_end")
for (i in 1:dim(biomart)[1] ) {
	if(biomart[i,3] > 0 ) {
	promoter_start<-as.numeric(biomart$transcript_start[i])-upstream
	promoter_end<-as.numeric(biomart$transcript_start[i])+downstream
   	mat[i,]<-c(biomart[i,1],as.character(biomart[i,2]),biomart[i,3],as.character(biomart[i,4]), biomart[i,5],biomart[i,6],biomart[i,7],biomart[i,8],promoter_start,promoter_end)	
	}	
	if(biomart[i,3] < 0 ) {
	promoter_start<-as.numeric(biomart$transcript_end[i])+upstream
	promoter_end<-as.numeric(biomart$transcript_end[i])-downstream
   	mat[i,]<-c(biomart[i,1],as.character(biomart[i,2]),biomart[i,3],as.character(biomart[i,4]), biomart[i,5],biomart[i,6],biomart[i,7],biomart[i,8],promoter_end,promoter_start)	
	}
}
output_file<- "mapped_file_promoter_region.txt"
write.table(mat,output_file, col.names=TRUE, row.names=FALSE, sep="\t", quote=FALSE)


</pre></div><div><h3>R Server</h3>localhost:6311</div>                </td>
              </tr>
                          <tr>
                <td><b>ecdf_plot</b></td>
                <td>rshell</td>
                <td>
                  this component plots the empirical cumulative distribution function of the P-values
of the two groups of genes (the ones that overlap and the ones that do not). 
in black we indicate overlapping genes and in red non overlapping genes                  <div><h3>Script</h3><pre>png(filename=cn, height=400, width=400, bg="white");
plot(ecdf(as.numeric((overlapping_genes_pvals[,4]))),xlab="adjusted p-value", main="",ylab= "proportion of genes <= x",  col.main="black");
lines(ecdf(as.numeric((non_overlapping_genes_pvals[,4]))), col="red")
legend("topleft", inset=.05, c("overlapping genes","nonOverlapping genes"), lwd=2, lty=c(1, 1, 1, 2), col=c("black","red"))
dev.off()

</pre></div><div><h3>R Server</h3>localhost:6311</div>                </td>
              </tr>
                          <tr>
                <td><b>ks.test</b></td>
                <td>rshell</td>
                <td>
                  Using again the adjusted pvalues of the two groups of genes (overlapping genes
and non overlapping genes), we perform a kolmogorov smirnov test in order
to test for differences between the two distributions. We export the p value of 
the statistical test, and the distance D between the two distributions.                   <div><h3>Script</h3><pre>#ks.test(cpg_genes_pvals[,4],no_cpg_genes_pvals[,4])

ks_test<-ks.test(overlapping_genes_pvals[,4],non_overlapping_genes_pvals[,4])
#cn_ks_p<-ks.test(as.numeric(brain_cpgs_cn[,4]), as.numeric(brain_no_cpgs_cn[,4]))$D
ks_list<-unlist(ks_test)
ks_D<-ks_list[[1]]
ks_p<-ks_list[[2]]

</pre></div><div><h3>R Server</h3>localhost:6311</div>                </td>
              </tr>
            
          </table>
        
      </div>
    </div>
    
    <!-- Beanshells -->
        <div class="fold">
      <div class="foldTitle">
        Beanshells (0)
      </div>
      <div class="foldContent" style="display: none;">
        
          <p class="none_text">None</p>
        
      </div>
    </div>
    
    <!-- Sinks -->
        <div class="fold">
      <div class="foldTitle">
        Outputs (6)
      </div>
      <div class="foldContent" style="display: none;">
                  <table class="simple">
            <tr>
              <th>Name</th>
              <th>Description</th>
            </tr>
          
                          <tr>
                <td><b>genes_overlap</b></td>
                <td>
                  
                    
                      The file with the genes that overlap with the epigenetic file                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>genes_non_overlap</b></td>
                <td>
                  
                    
                      The file with the genes that do not overlap with the epigenetic file                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>ecdf_plot</b></td>
                <td>
                  
                    
                      empirical cumulative distribution plot between the two distributions of genes that overlap and the ones that do not overlap with the epigenetic file under study.                       
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>distance</b></td>
                <td>
                  
                    
                      the distance D between the two distributions of genes that overlap and the ones that do not overlap with the epigenetic file. The greatest the difference the bigger the effect of the epigenetic file on the list of genes                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>pvalue</b></td>
                <td>
                  
                    
                      the pvalue indicating the significance of the kolmogorov smirnov test, between the two distributions of genes that overlap and the ones that do not overlap with the epigenetic file.                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>promoter_file</b></td>
                <td>
                  
                    
                      The file we provided as input with genomic information
with a promoter region for each gene included.                      
                    
                  
                </td>
              </tr>
            
          </table>
        
      </div>
    </div>
    
    <!-- Datalinks -->
        <div class="fold">
      <div class="foldTitle">
        Datalinks (19)
      </div>
      <div class="foldContent" style="display: none;">
                  <table class="simple">
            <tr>
              <th>Source</th>
              <th>Sink</th>
            </tr>
          
                          <tr>
                <td>workingdir</td>
                <td>read_files:dir</td>
              </tr>
                          <tr>
                <td>epigen_file</td>
                <td>read_files:epigenetic_file</td>
              </tr>
                          <tr>
                <td>calculate_promoter_region:output_file</td>
                <td>read_files:biomart</td>
              </tr>
                          <tr>
                <td>gene_chr</td>
                <td>transform_files_to_gen_ranges:gene_chr</td>
              </tr>
                          <tr>
                <td>epigen_file_end</td>
                <td>transform_files_to_gen_ranges:inter_end</td>
              </tr>
                          <tr>
                <td>epigen_file_start</td>
                <td>transform_files_to_gen_ranges:inter_start</td>
              </tr>
                          <tr>
                <td>epigen_file_chr</td>
                <td>transform_files_to_gen_ranges:inter_chr</td>
              </tr>
                          <tr>
                <td>overlap</td>
                <td>calculate_overlaps:overlap</td>
              </tr>
                          <tr>
                <td>gene_expr_file</td>
                <td>calculate_overlaps:gene_expr_file</td>
              </tr>
                          <tr>
                <td>upstream</td>
                <td>calculate_promoter_region:upstream</td>
              </tr>
                          <tr>
                <td>downstream</td>
                <td>calculate_promoter_region:downstream</td>
              </tr>
                          <tr>
                <td>biomartfile</td>
                <td>calculate_promoter_region:genes</td>
              </tr>
                          <tr>
                <td>workingdir</td>
                <td>calculate_promoter_region:workdir</td>
              </tr>
                          <tr>
                <td>calculate_overlaps:overlapping_genes_pvals_out</td>
                <td>genes_overlap</td>
              </tr>
                          <tr>
                <td>calculate_overlaps:non_overlapping_genes_pvals_out</td>
                <td>genes_non_overlap</td>
              </tr>
                          <tr>
                <td>ecdf_plot:cn</td>
                <td>ecdf_plot</td>
              </tr>
                          <tr>
                <td>ks.test:ks_D</td>
                <td>distance</td>
              </tr>
                          <tr>
                <td>ks.test:ks_p</td>
                <td>pvalue</td>
              </tr>
                          <tr>
                <td>calculate_promoter_region:output_file</td>
                <td>promoter_file</td>
              </tr>
            
          </table>
        
      </div>
    </div>
    
    <!-- Coordinations -->
        <div class="fold">
      <div class="foldTitle">
        Coordinations (4)
      </div>
      <div class="foldContent" style="display: none;">
                  <table class="simple">
            <tr>
              <th>Controller</th>
              <th>Target</th>
            </tr>
          
                          <tr>
                <td>calculate_overlaps</td>
                <td>ks.test</td>
              </tr>
                          <tr>
                <td>calculate_overlaps</td>
                <td>ecdf_plot</td>
              </tr>
                          <tr>
                <td>read_files</td>
                <td>transform_files_to_gen_ranges</td>
              </tr>
                          <tr>
                <td>transform_files_to_gen_ranges</td>
                <td>calculate_overlaps</td>
              </tr>
            
          </table>
        
      </div>
    </div>
  
    

          
      
    </div>

  </div>

</div>

<div class="contribution_right_box">
  <div class="contribution_section_box" style= "font-size: 100%; padding: 0.7em 0.9em; font-weight: bold;">
    <p><img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[The type of workflow system that this Workflow is designed for.] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" /> Workflow Type</p>
    <p><a href="/content_types/2">Taverna 2</a></p>
  </div>

  <div class="contribution_section_box">
	<p class="heading">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[The Uploader is the person who uploaded/created this Workflow on myExperiment, and who manages it on myExperiment.] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Uploader
	</p>
	<p>
		<center>
			<div class="avatar" style="width: 66px">
  <a href="/users/17730"><img alt="4185?size=60x60" class="framed" src="/pictures/show/4185?size=60x60" title="Eleni" /></a>
  
	<span style="display:block;margin-top:2px;text-align:center;">
		
			<img alt="Nl" src="/images/famfamfam_flags/nl.png?1680607579" style="vertical-align:middle; margin-right: 0.3em;" title="header=[] body=[&lt;b&gt;Location: &lt;/b&gt;Leiden, Netherlands] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		

    
		  <a href="/users/17730" title="header=[] body=[Eleni] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Eleni</a>
		

    
	</span>
  
</div>

		</center>
	</p>
</div>


  

  

<div class="contribution_section_box">
	<p class="heading">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[This specifies what rights users have when they download and use this Workflow.] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		License
		<a name="license"></a>
	</p>

  

    
    
    
    
    


    

    <!--

    <rdf:RDF xmlns="http://creativecommons.org/ns#"
        xmlns:dc="http://purl.org/dc/elements/1.1/"
        xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#">
    <Work rdf:about="">
       <dc:title>get promoter region + operate on genomic intervals</dc:title>
       <dc:date>2013-11-04 19:28:07 UTC</dc:date>
       <dc:description>&lt;p&gt;
Finds the overlap between two datasets which contain genomic information (e.g. [gene id], chromosome name, gene start, gene end), plus some statistics. Returns rows of file_1 which overlap with the second file. A kolmogorov smirnov test is applied between the list that overlaps and the one that does not. 
NOTE: The library(GenomicRanges) is a prerequisite for this workflow
&lt;/p&gt;

</dc:description>
       <dc:creator><Agent>
          <dc:title><a href="/users/17730" title="header=[] body=[Eleni] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Eleni</a></dc:title>
       </Agent></dc:creator>
       <dc:rights><Agent>
          <dc:title><a href="/users/17730" title="header=[] body=[Eleni] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Eleni</a></dc:title>
       </Agent></dc:rights>
       <dc:type rdf:resource="http://purl.org/dc/dcmitype/Data" />
       <dc:source rdf:resource="http://www.myexperiment.org/workflows/3886.html"/>
       <license rdf:resource="http://creativecommons.org/licenses/by-sa/3.0/" />
    </Work>

    <License rdf:about="http://creativecommons.org/licenses/by-sa/3.0/">
      <permits rdf:resource="http://creativecommons.org/ns#Reproduction" />  <permits rdf:resource="http://creativecommons.org/ns#Distribution" />  <requires rdf:resource="http://creativecommons.org/ns#Notice" />  <requires rdf:resource="http://creativecommons.org/ns#Attribution" />  <permits rdf:resource="http://creativecommons.org/ns#DerivativeWorks" />  <requires rdf:resource="http://creativecommons.org/ns#ShareAlike" />
    </License>

    </rdf:RDF>

    -->

  

  <div class="contribution_currentlicense">
    <p>
      
        All versions of this Workflow are
      
      
        licensed under: <br/><br/><a rel="Copyright" href="/licenses/2" title="Creative Commons Attribution-Share Alike 3.0 Unported License"><img src="/images/by-sa.png" /></a>
      
    </p>
  </div>
	
</div>


  <script type="text/javascript">
    function showVersion(form) {
        var url = $("resource_versions").value;
        location.href = url;
        form.submit
    }
</script>
<style>
  .version_selection {
      font-size:smaller;
      text-align: center;
  }
</style>

<div class="contribution_section_box">
  <p class="heading">
    <img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[This box shows version 1 for this entry] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />    <span>Version 1 </span>
    <span class="count_text">(of 1)</span>
  </p>
  
</div>

  <div class="contribution_section_box">
	
	<p class="heading" style="margin-bottom: 0; padding-bottom: 0;">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[Which people/groups get the credit for this Workflow?] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Credits <span class="count_text">(1)</span>
		<a name="credits"></a>
	</p>
	<span style="font-size: 85%; color: #666666; display: block; padding: 0.3em 0.5em 0 0.5em; margin: 0; line-height: 1em;">(People/Groups)</span>
	
		<div style="font-size: 85%;">
			<ul class="items">
				
					<li>
						
							
								<span class="icon"><img alt="User" src="/images/famfamfam_silk/user.png?1680607579" /> </span> 
							
							<a href="/users/17730" title="header=[] body=[Eleni] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Eleni</a>
						
					</li>
				
			</ul>
		</div>
	
	
	
	<p class="heading" style="margin-bottom: 0; padding-bottom: 0;">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[What other Workflows/Files is this one based on?] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Attributions <span class="count_text">(0)</span>
		<a name="attributions"></a>
	</p>
	<span style="font-size: 85%; color: #666666; display: block; padding: 0.3em 0.5em 0 0.5em; margin: 0; line-height: 1em;">(Workflows/Files)</span>
	
		<p class="none_text">None</p>
	
	
	
	
</div>


  


  <div class="contribution_section_box">
	<p class="heading">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[The tags for this Workflow entry.&lt;br/&gt;Note(1): if this Workflow has many versions, the tags apply to ALL versions in this entry.&lt;br/&gt;Note(2): the size of the tags show how popular they are on myExperiment.] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Tags <span id="tags_box_header_tag_count_span" class="count_text">(9)</span>
		<a name="tags"></a>
	</p>

	<script type="text/javascript">
  function showAllTags() {
    Element.show('all_tags');
	  Element.hide('user_tags');
	  return false;
  }
  
  function showUsersTags() {
  	Element.hide('all_tags');
	  Element.show('user_tags');
	  return false;
  }
	
	function flip_tags_mode()
	{
		if( $('toggle_tags_mode').checked )
		  showUsersTags();
	  else
		  showAllTags();
	}
</script>

<div id="tags_inner_box">
	
		<div class="box_currentuser_specific" style="padding: 0.3em 0.2em; margin: 0.6em 0.4em;">
		  <label for="toggle_tags_mode" style="font-size: 85%; font-weight: bold;"><input id="toggle_tags_mode" name="toggle_tags_mode" onclick="flip_tags_mode();" style="margin-right: 0.3em;" type="checkbox" value="1" />Uploader tags</label>
		</div>
		<div id="all_tags">
			
				<div class="hTagcloud">
  <ul class="popularity">
    <li><a href='/tags/862' class='popular'>bioconductor</a></li>
<li> | </li>
    <li><a href='/tags/3878' class='popular'>chromatin states</a></li>
<li> | </li>
    <li><a href='/tags/3980' class='popular'>compute promoter region</a></li>
<li> | </li>
    <li><a href='/tags/3886' class='popular'>entrez gene ids</a></li>
<li> | </li>
    <li><a href='/tags/3875' class='popular'>epigenetics</a></li>
<li> | </li>
    <li><a href='/tags/178' class='vvvv-popular'>gene</a></li>
<li> | </li>
    <li><a href='/tags/3879' class='popular'>genomic overlaps</a></li>
<li> | </li>
    <li><a href='/tags/3982' class='popular'>genomicranges</a></li>
<li> | </li>
    <li><a href='/tags/4000' class='popular'>huntington disease</a></li>
  </ul>
</div>

			
		</div>
		<div id="user_tags"  style="display: none;">
			
				<div class="hTagcloud">
  <ul class="popularity">
    <li><a href='/tags/862' class='popular'>bioconductor</a></li>
<li> | </li>
    <li><a href='/tags/3878' class='popular'>chromatin states</a></li>
<li> | </li>
    <li><a href='/tags/3980' class='popular'>compute promoter region</a></li>
<li> | </li>
    <li><a href='/tags/3886' class='popular'>entrez gene ids</a></li>
<li> | </li>
    <li><a href='/tags/3875' class='popular'>epigenetics</a></li>
<li> | </li>
    <li><a href='/tags/178' class='vvvv-popular'>gene</a></li>
<li> | </li>
    <li><a href='/tags/3879' class='popular'>genomic overlaps</a></li>
<li> | </li>
    <li><a href='/tags/3982' class='popular'>genomicranges</a></li>
<li> | </li>
    <li><a href='/tags/4000' class='popular'>huntington disease</a></li>
  </ul>
</div>

			
		</div>
	
</div>
	
	
	
	
		<p><small><b><a href="/session/new">Log in</a></b> to add Tags</small></p>
	
</div>


  
<div class="contribution_section_box">
	<p class="heading">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[All the Groups that this item has been shared with] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Shared with Groups <span class="count_text">(0)</span>
	</p>
	
		<p class="none_text">None</p>
	
</div>

  <div class="contribution_section_box" id="in_packs_box">
  
	<p class="heading">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[All the Packs that this item (or versions of this item [[if available]]) is referred to in] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Featured In Packs  <span class="count_text">(1)</span>
		<a name="featured_in_packs"></a>
	</p>
	
		<div style="font-size: 85%;">
			<ul class="items">
				
					<li>
						
							<span class="icon"><img alt="Pack" src="/images/manhattan_studio/folder-closed_16.png?1680607579" /> </span>
							<a href="/packs/485">HD chromatin analysis</a>
						
					</li>
				
			</ul>
		</div>
	
	
	
		<p><small><b><a href="/session/new">Log in</a></b> to add to one of your Packs</small></p>
	
</div>


  <div class="contribution_section_box">
	
	<p class="heading" style="margin-bottom: 0; padding-bottom: 0;">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[Shows files and workflows that have attributed this Workflow.] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Attributed By <span class="count_text">(0)</span>
	  <a name="attributed_by"></a>
	</p>
	<span style="font-size: 85%; color: #666666; display: block; padding: 0.3em 0.5em 0 0.5em; margin: 0; line-height: 1em;">(Workflows/Files)</span>
	
		
		<p class="none_text">None</p>
	
	
</div>


  
<div class="contribution_section_box">
	<p class="heading">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[The users who have added this Workflow to their favourites] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Favourited By <font class="count_text">(0)</font>
		<a name="favourited_by"></a>
	</p>
	
	
		<p class="none_text">No one</p>
	
	
	
</div>

  

<div class="contribution_section_box">
	<p class="heading" style="margin-bottom: 0.3em;">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[Viewing / download counts are grand totals for this Workflow.&lt;br/&gt;Click &#39;see breakdown&#39; to view details.] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Statistics
	</p>
	
	<div class="stats_box"> 
		<p style="font-size: 85%;">
  1069 viewings
</p>

	<p style="font-size: 85%;">
	  856 downloads
	</p>

<p style="margin-top: 0.2em;">
	<small>[ <a href="/workflows/3886/statistics">see breakdown</a> ]</small>
</p>
		
		<div class="fold" style="padding-top: 0.6em;">
			<div class="foldTitle">More</div>
			<div class="foldContent" style="display: none; text-align: left;">
				
					<p>
						<a href="#credits">1 credit</a> <!-- won't work for type=='group'; needs anchor called '#group_creditations' -->
					</p>
					
					<p>
						<a href="#attributions">0 attributions</a>
					</p>
				
				
				
					<p>
			      <a href="#tags">9 tags</a>
			    </p>
				
				
				
					<p>
			      <a href="#favourited_by">0 favourites</a>
			    </p>
				

				
					<p>
						<a href="#citations">0 citations</a>
			    </p>
				
				
				
					<p>
			      <a href="#reviews">0 reviews</a>
			    </p>
				
				
				
			    <p>
			    	<a href="#comments">0 comments</a>
			    </p>
				
			</div>
		</div>
		
		
		
		
	</div>
</div>

</div>

<div class="clearer">&nbsp;</div>

<!-- BEGIN tabs -->

<br/>

<div id="tabsContainer" class="tabsContainer"></div>

<div class="tabContainer">
  <div class="tabTitle">Citations (0)</div>
  <div class="tabContent">

    <a name="citations"></a>
    
      <p><i>None</i></p>
    
		
		<br/>

    

  </div>
</div>


<div class="tabContainer">
  <div class="tabTitle">Version History</div>
  <div class="tabContent">
    <a name="version_history"></a>
    <p>In chronological order:</p>

<ol class="version_history">
  
    <li>
      <div class="title">
        <a href="/workflows/3886/versions/1">get promoter region + operate on genomic intervals</a>
      </div>
      <div class="metadata">

        <p>
          Created
          
            by <a href="/users/17730" title="header=[] body=[Eleni] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Eleni</a>
          
          on Monday 04 November 2013 19:28:07 (UTC)
        </p>

        

        
      </div>
    </li>
  
</ol>


  </div>
</div>



<!-- END tabs -->

<br/>
<br/>

<div id="ratingsBox">
  <h2>
	<span class="icon"><img alt="Reviews" src="/images/famfamfam_silk/report.png?1680607579" /> </span>	<span style="vertical-align: middle;"><a href="/workflows/3886/reviews">Reviews</a></span>
	<span class="count_text" style="vertical-align: middle;">(0)</span>
</h2>

<div class="reviewsBox">

	
		<p><i>No reviews yet</i></p>
		<a name="reviews"></a>
		<p style="margin-top: 0.6em;">
			<strong><a href="/workflows/3886/reviews/new">Be the first to review!</a></strong>
		</p>
	
	
</div>

</div>

<br/>
<br/>

<div id="commentsBox">
  <script src="/javascripts/ckeditor/ckeditor.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/widgets/tooltip.js?1680607579" type="text/javascript"></script>

<h2>
	<span class="icon"><img alt="Comments" src="/images/famfamfam_silk/comments.png?1680607579" /> </span>	<span style="vertical-align: middle;">Comments</span>
	<span class="count_text" style="vertical-align: middle;">(0)</span>
</h2>

<ul class="sectionIcons">
  <li><span class="icon"><a href="http://www.myexperiment.org/workflows/3886/comments/timeline"><img alt="Timeline" src="/images/famfamfam_silk/timeline_marker.png?1680607579" /> View Timeline</a></span></li>
</ul>

<div class="commentsBox">
	
	
		<p><i>No comments yet</i></p>
		<a name="comments"></a>
		
			<p style="margin-top: 0.6em;"><strong><a href="/session/new">Log in</a> to make a comment </strong></p>
		
	
	
	<br/>
	
	

</div>


</div>

<!-- Other workflows that use a service this workflow users -->

<br />
<br />

<h2>
  <span class="icon"><img alt="Workflow" src="/images/redmond_studio/applications_16.png?1680607579" /> </span>  <span style="vertical-align: middle;">Other workflows that use similar services</span>
  <span class="count_text" style="vertical-align: middle;">(0)</span>
</h2>




    <p>There are no workflows in myExperiment that use similar services to this Workflow.</p>



<!-- OpenURL context object -->

<span class="Z3988" title="url_ver=Z39.88-2004&amp;url_ctx_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Actx&amp;ctx_ver=Z39.88-2004&amp;ctx_tim=2023-08-16T17%3A13%3A23%2B00%3A00&amp;ctx_id=&amp;ctx_enc=info%3Aofi%2Fenc%3AUTF-8&amp;rft.title=get+promoter+region+%2B+operate+on+genomic+intervals&amp;rft.date=2013-11-04&amp;rft.au=Eleni&amp;rft.genre=unknown&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Adc"></span>
      </div>
    </div>




            <div id="ft">
          <div class="container">
  <div class="row">
    <div class="col-md-4 text-left">
      Funded by:
      <a href="http://biovel.eu">BioVeL</a>,
      <a href="http://www.epsrc.ac.uk/">EPSRC</a>,
      <a href="http://jisc.ac.uk/">JISC</a>,
      <a href="http://www.microsoft.com">Microsoft</a>

      <br/><br/>

      Affiliates:
      <a href="http://www.mygrid.org.uk">myGrid</a>,
      <a href="http://www.taverna.org.uk/">Taverna</a>,
      <a href="http://www.biocatalogue.org">BioCatalogue</a>
    </div>

    <div class="col-md-4 text-center">
      <a href="/about">About</a> |
      <a href="/privacy">Privacy</a> |
      <a href="http://wiki.myexperiment.org/index.php/Papers">Publications</a> |
      <a href="/feedback">Contact / Feedback</a> |
      <a href="http://lists.nongnu.org/mailman/listinfo/myexperiment-discuss">Mailing List</a>
    </div>

    <div class="col-md-4 text-right">
      Icons: <a href="http://www.famfamfam.com/lab/icons/silk/">Silk icon set 1.3</a><br/><br/>
      
    </div>

  </div>
  <div class="row">
    <div class="col-md-12 text-center">
      Copyright &#169; 2007 &ndash; 2018
      <a href="http://www.manchester.ac.uk/">The University of Manchester</a>
      and
      <a href="http://www.soton.ac.uk/">University of Southampton</a>
    </div>
  </div>
</div>

        </div>
    



<script type="text/javascript">
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
        (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
        m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-1498417-4', 'auto');
    ga('set', 'anonymizeIp', true);
    ga('send', 'pageview');
</script>




</body>
</html>

