<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
    "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <meta http-equiv="content-type" content="text/html;charset=UTF-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1">
  
  <meta name="google-site-verification" content="3hq0Cfhw5_-EaaNMum0ETX847sTl7jeQ3ZNAla4a6UQ" />
  
  <title>myExperiment &#45; Workflows &#45; Retention Time Prediction with X!Tandem (Magnus Palmblad) [Taverna 2 Workflow]</title>
  <link rel="alternate" href="http://www.myexperiment.org/workflows/3134.rdf" type="application/rdf+xml" title="RDF+XML" />
  <link rel="alternate" href="http://www.myexperiment.org/workflows/3134.xml" type="application/xml" title="REST XML" />
  <link rel="shortcut icon" href="/favicon.ico" type="image/x-icon"/>
  <link href="/stylesheets/acts_as_taggable_stylesheet.css?1680607579" media="screen" rel="stylesheet" type="text/css" />
<link href="/stylesheets/star_rating.css?1680607579" media="screen" rel="stylesheet" type="text/css" />
<link href="/stylesheets/gadgets.css?1680607579" media="screen" rel="stylesheet" type="text/css" />
<link href="/stylesheets/misc_skinning.css?1680607579" media="screen" rel="stylesheet" type="text/css" />
<link href="/stylesheets/pagination.css?1680607579" media="screen" rel="stylesheet" type="text/css" />
<link href="/stylesheets/bootstrap.css?1680607579" media="screen" rel="stylesheet" type="text/css" />
<link href="/stylesheets/navbar.css?1680607579" media="screen" rel="stylesheet" type="text/css" />
<link href="/stylesheets/styles.css?1680607579" media="screen" rel="stylesheet" type="text/css" />
  <script src="/javascripts/jquery-1.11.1.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/jquery-noconflict.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/prototype-1.7.2-patched.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/effects.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/dragdrop.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/controls.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/boxover.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/bootstrap.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/tabs.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/folds.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/application.js?1680607579" type="text/javascript"></script>

  
  






  
  
    <script src='//www.google.com/recaptcha/api.js'></script>
  
</head>
<body>



<div class="navbar navbar-myexp" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      
          <a id="logo" href="/">
            <img alt="Logo" class="hidden-sm" src="/images/logo.png?1680607579" />
            <img alt="Logo_cropped" class="visible-sm" src="/images/logo_cropped.png?1680607579" />
          </a>
      
    </div>
    <div class="collapse navbar-collapse">
      <ul class="nav navbar-nav">
        
          <ul class="nav navbar-nav">
  
    <li >
      <a href="/home">Home</a>    </li>
  
    <li >
      <a href="/users">Users</a>    </li>
  
    <li >
      <a href="/groups">Groups</a>    </li>
  
    <li class="active">
      <a href="/workflows">Workflows</a>    </li>
  
    <li >
      <a href="/files">Files</a>    </li>
  
    <li >
      <a href="/packs">Packs</a>    </li>
  
</ul>

        
        
            <form action="/search" class="navbar-form navbar-left" id="search" method="get">
    <div class="input-group">
      <input class="form-control" id="query" name="query" placeholder="Search" type="text" />
      <span class="input-group-btn">
        
        <input class="btn btn-default" name="commit" type="submit" value="Go" />
        </span>
    </div>
</form>

        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
            <li><a href="/session/new?return_to=%2Fworkflows%2F3134.html">Log in</a></li>
    <li><a href="/users/new">Register</a></li>


      </ul>
    </div><!--/.nav-collapse -->
  </div>
  
</div>




    <div class="container" id="main">
      <div id="myexp_breadcrumbs_bar">
        <ol class="breadcrumb">
  <li><a href="/home">Home</a></li>
  


  <li><a href="/workflows">Workflows</a></li>



  
    <li>Retention Time Prediction with X!Tandem</li>
  


  <div style="float: right"></div>
</ol>

      </div>
      <div id="myexp_content">
        


        



<h1 class="contribution_title">
  Retention Time Prediction with X!Tandem
</h1>

<div class="contribution_aftertitle">
	<b>Created:</b> 2012-09-03 10:43:31
	
		&nbsp;&nbsp;&nbsp;&nbsp;
		<b>Last updated: </b>2012-09-03 10:44:33
	
</div>
<ul class="sectionIcons">
  
      <li><span class="icon"><a href="http://www.myexperiment.org/workflows/3134/download/Retention_Time_Prediction_with_X_Tandem-v1.t2flow?version=1"><img alt="Download Workflow (for version 1)" src="/images/redmond_studio/arrow-down_16.png?1680607579" /> Download Workflow</a></span></li>
      
  
      
</ul>


<div class="contribution_left_box">

  <div class="contribution_version_box">



    <div class="contribution_version_inner_box">

      
    <div class="contribution_description">
      <p>This workflow identifies peptides from tandem mass spectra using X!Tandem as in a standard installation of the Trans-Proteomic Pipeline (TPP, version 4.6, but earlier versions should also work). The peptide-spectrum matches are validates using PeptideProphet (also from a standard installation of TPP) and peptides with at least a probability 0.95 are used to train a linear retention time predictor (Palmblad et al., 2002), whereby retention coefficients are also derived. These indirectly provide information on the chromatographic stationary and mobile phase (composition and pH) and the gradient.  The two Rshell scrips visualize the prediction versus measured retention times and amino acid coefficients colored by physico-chemical properties (basic/acidic/polar etc.).</p>
    </div>


      <h3>
        <img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[This section shows the preview of this version of the Workflow (click on the image to expand)] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
        Preview
      </h3>

      

          <div class="contribution_preview" style="width: 100%;">
            <p style="text-align: center">
              <a href="/workflows/3134/versions/1/previews/full" onclick="window.open(this.href);return false;"><img alt="Medium" src="/workflows/3134/versions/1/previews/medium" /></a>
            </p>
            
                <p style="text-align: right; color: #666666;">
                  <a href="/workflows/3134/versions/1/previews/svg">Download as scalable diagram (SVG)</a>
                </p>
            

            
          </div>
      



      <h3>
        <img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[This section provides options for running this version of the Workflow] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
        Run
      </h3>

      
          
              <div style="margin: 0 0.5em;">
	<h4>
		Run this Workflow in the Taverna Workbench...
	</h4>
	
	<div class="option_box">
		
		<p style="text-align: center;">
			<b>Option 1:</b>
		</p>
		
		<p style="font-size: 93%; text-align: center;">
			Copy and paste this link into File > 'Open workflow location...'<br/>
			<a href="http://www.myexperiment.org/workflows/3134/download?version=1">http://www.myexperiment.org/workflows/3134/download?version=1</a><br/>
			<small>[ <a href="#" onclick="Effect.toggle(&quot;run_taverna_more_box&quot;,'blind',{duration:0.3});; return false;">More Info<img alt="Expand" src="/images/folds/unfold.png?1680607579" style="margin-left: 0.3em; vertical-align: middle;" /></a> ]</small>
		</p>
		
		<div id="run_taverna_more_box" style="display: none;">
			<p style="font-size: 85%; margin-top: 0.5em; text-align: center;">
				Taverna is available from <a href="http://taverna.sourceforge.net/" onclick="window.open(this.href);return false;">http://taverna.sourceforge.net/</a>
			</p>
			
			<p style="text-align: center; margin-top: 0.5em; font-size: 85%;">
				If you are having problems downloading it in Taverna, you may need to provide your username and password in the URL so that Taverna can access the Workflow: <br/>
				
					<b>Replace <font color="#990000">http://</font> in the link above with <font color="#990000">http://yourusername:yourpassword@</font></b>
				
			</p>
		</div>
	
	</div>
</div>


          
      

      <br/>

      <h3>
        <img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[This section shows the internal components of this workflow version] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
        Workflow Components
      </h3>

      
          
              
	    <!-- Authors -->
        <div class="fold">
      <div class="foldTitle">
        <img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[This is the author information extracted from the workflow version] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
        Authors (1)
      </div>
      <div class="foldContent" style="display: none;">
                  <table class="simple">
                          <tr>
                <td>Jeroen de Bruin
Yassene Mohammed
Wendy Plugge
Alex Henneman
Magnus Palmblad (n.m.palmblad@lumc.nl)</td>
              </tr>
            
          </table>
        
      </div>
    </div>
    
    <!-- Titles -->
        <div class="fold">
      <div class="foldTitle">
        <img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[These are the descriptive titles embedded within the workflow version] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
        Titles (1)
      </div>
      <div class="foldContent" style="display: none;">
                  <table class="simple">
                          <tr>
                <td>Retention Time Prediction with X!Tandem</td>
              </tr>
            
          </table>
        
      </div>
    </div>

    <!-- Descriptions -->
        <div class="fold">
      <div class="foldTitle">
        <img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[These are the descriptions embedded within the workflow version] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
        Descriptions (1)
      </div>
      <div class="foldContent" style="display: none;">
                  <table class="simple">
                          <tr>
                <td>This workflow identifies peptides from tandem mass spectra using X!Tandem as in a standard installation of the Trans-Proteomic Pipeline (TPP, version 4.6, but earlier versions should also work). The peptide-spectrum matches are validates using PeptideProphet (also from a standard installation of TPP) and peptides with at least a probability 0.95 are used to train a linear retention time predictor (Palmblad et al., 2002), whereby retention coefficients are also derived. These indirectly provide information on the chromatographic stationary and mobile phase (composition and pH) and the gradient.

The two Rshell scrips visualize the prediction versus measured retention times and amino acid coefficients colored by physico-chemical properties (basic/acidic/polar etc.).</td>
              </tr>
            
          </table>
        
      </div>
    </div>
        
    <!-- Dependencies -->
        <div class="fold">
      <div class="foldTitle">
        <img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[These are the listed dependencies of the workflow] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
        Dependencies (0)
      </div>
      <div class="foldContent" style="display: none;">
        
          <p class="none_text">None</p>
        
      </div>
    </div>
        
    <!-- Sources -->
        <div class="fold">
      <div class="foldTitle">
        Inputs (4)
      </div>
      <div class="foldContent" style="display: none;">
                  <table class="simple">
            <tr>
              <th>Name</th>
              <th>Description</th>
            </tr>
          
                          <tr>
                <td><b>Tandem_Param_File</b></td>
                <td>
                  
                    
                      Parameters for X!Tandem                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>FASTA_File</b></td>
                <td>
                  
                    
                      Sequence database in FASTA format                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>Log_File</b></td>
                <td>
                  
                    
                      Log file to which X!Tandem Beanshell output will be written (can be omitted if the corresponding line in the &quot;Tandem&quot; component is removed).                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>mzXML_File</b></td>
                <td>
                  
                    
                      LC-MS/MS data in mzXML format                      
                    
                  
                </td>
              </tr>
            
          </table>
        
      </div>
    </div>
      
    <!-- Processors -->
        <div class="fold">
      <div class="foldTitle">
        Processors (10)
      </div>
      <div class="foldContent" style="display: none;">
                  <table class="simple">
            <tr>
              <th>Name</th>
              <th>Type</th>
              <th>Description</th>
            </tr>
          
                          <tr>
                <td><b>Read_Text_File</b></td>
                <td>localworker</td>
                <td>
                  This shell reads the Tandem results to feed the XPath extractors below.                  <div><h3>Script</h3><pre>BufferedReader getReader (String fileUrl, String encoding) throws IOException {
		InputStreamReader reader;
		try {
			if (encoding == null) {
				reader = new FileReader(fileUrl);
			} else {
				reader = new InputStreamReader(new FileInputStream(fileUrl),encoding); 
			}
		}
		catch (FileNotFoundException e) {
			// try a real URL instead
			URL url = new URL(fileUrl);
			if (encoding == null) {
				reader = new InputStreamReader (url.openStream());
			} else {
				reader = new InputStreamReader (url.openStream(), encoding);
			}
		}
		return new BufferedReader(reader);
	}



StringBuffer sb = new StringBuffer(4000);

if (encoding == void) {
	encoding = null;
}

BufferedReader in = getReader(fileurl, encoding);
String str;
String lineEnding = System.getProperty("line.separator");

while ((str = in.readLine()) != null) {
	sb.append(str);
	sb.append(lineEnding);
}
in.close();
filecontents = sb.toString();
  
</pre></div>                </td>
              </tr>
                          <tr>
                <td><b>Extract_Peptides</b></td>
                <td>xpath</td>
                <td>
                  This XPath extracts the amino acid sequences of the identified peptides.                  <div><h3>Xpath Expression</h3><pre>/default:msms_pipeline_analysis/default:msms_run_summary/default:spectrum_query/default:search_result/default:search_hit/@peptide</pre></div>                </td>
              </tr>
                          <tr>
                <td><b>Extract_Scan_Numbers</b></td>
                <td>xpath</td>
                <td>
                  This XPath extracts the corresponding scan numbers for the identified peptides.                  <div><h3>Xpath Expression</h3><pre>/default:msms_pipeline_analysis/default:msms_run_summary/default:spectrum_query/@start_scan</pre></div>                </td>
              </tr>
                          <tr>
                <td><b>Join_and_Insert_Tabs</b></td>
                <td>beanshell</td>
                <td>
                  This Beanshell formats the data for rt.                  <div><h3>Script</h3><pre>
String out = new String();

for (i=0; i<in1.size();i++){
	out = out + in1.get(i) + "\t" + in2.get(i) + "\n";
}

out1 = out;</pre></div>                </td>
              </tr>
                          <tr>
                <td><b>rt</b></td>
                <td>externaltool</td>
                <td>
                  This tool calls a local installation of rt - here rt 3.0.                                  </td>
              </tr>
                          <tr>
                <td><b>Plot_Predicted_vs_Measured_RT</b></td>
                <td>rshell</td>
                <td>
                  This Rshell script plots predicted versus measured retention time for the identified peptides.                  <div><h3>Script</h3><pre>#
# Plot_Predicted_vs_Measured_RT
# By Wendy Plugge and Magnus Palmblad 2012
#

# Load and convert text data from RT.exe to dataframe

data_AA_rt=in2
file_split=unlist(strsplit(data_AA_rt, split="\\n")) # split data into lines
extract_AA_RT=file_split[1:25]

AminoAcids=as.character(lapply(strsplit(extract_AA_RT,"\\t"), function(x) x[1])) # create string vector using every first element of extract_AA_RT  
RetTime=as.numeric(lapply(strsplit(extract_AA_RT,"\\t"), function(x) x[2]))      # create string vector using every second element of extract_AA_RT  
frame_AA_RT=data.frame(AminoAcids, RetTime, stringsAsFactors=FALSE)            # create data frame with headers


# Load peptide data from PeptideProphet and convert this to a data frame

data_pep=in1
file_split_pep=unlist(strsplit(data_pep, split="\\n"))                          # split data into lines
peptide=as.character(lapply(strsplit(file_split_pep,"\\t"), function(x) x[1]))  # create string vector using every first element of file_split_pep
pep_RetTime=as.numeric(lapply(strsplit(file_split_pep,"\\t"), function(x) x[2]))# create string vector using every second element of file_split_pep
data_pep_frame=data.frame(peptide, pep_RetTime, stringsAsFactors=FALSE)         # create data frame with headers
 
offset=frame_AA_RT$RetTime[25] # For offset ("void time")


# Create filename for output

file_name=in3
extension=".RT_prediction.pdf"               # new extension of PDF image
txt_name=paste(file_name, extension, sep="") # join input name with new extension
output_file_name=txt_name


# Calculate predicted retention times for peptides

list_value=list()
for(x in data_pep_frame$peptide){
      split_pep=unlist(strsplit(x, split=NULL))
      value=offset
      for(x in split_pep){
         place=grep(x,frame_AA_RT$AminoAcids)
         value=value+frame_AA_RT$RetTime[place]
}
list_value<-c(list_value,value)
}
Predicted_RT=as.numeric(list_value ) # convert character strings to numeric values
res=lm(Predicted_RT~data_pep_frame$pep_RetTime)
pdf(output_file_name)
plot(data_pep_frame$pep_RetTime, Predicted_RT,col="black", pch=19, xlab="measured retention time (scan number)", ylab="predicted retention time (scan number)")
abline(res,col="red",lwd=1.5)
dev.off()</pre></div><div><h3>R Server</h3>localhost:6311</div>                </td>
              </tr>
                          <tr>
                <td><b>Plot_Retention_Coefficients</b></td>
                <td>rshell</td>
                <td>
                  This Rshell script plots the retention coefficients for the different amino acids.                  <div><h3>Script</h3><pre>#
# Plot_Retention_Coefficients
# By Wendy Plugge and Magnus Palmblad 2012
#

# Load files

read_file=in1
file_name=in2


# Create filename for output file

extension=".R_coefficient_AA.pdf"
pdf_name=paste(file_name, extension, sep="")
output_file_name=pdf_name


# Convert input of read_file to data frame

file_split=unlist(strsplit(read_file, split="\\n")) # split data into lines
extract_AA_RT=file_split[1:25]

AminoAcids=as.character(lapply(strsplit(extract_AA_RT,"\\t"), function(x) x[1])) # create string vector using every first element of extract_AA_RT 
RetTime=as.numeric(lapply(strsplit(extract_AA_RT,"\\t"), function(x) x[2]))      # create string vector using every second element of extract_AA_RT  
frame_AA_RT=data.frame(AminoAcids, RetTime, stringsAsFactors=FALSE)            # create data frame


# Add column to data frame

frame_AA_RT["color_AA"]<-c("orange","blue","magenta","red","green","red","magenta","orange","blue","green","green","blue","green","green","green","orange","orange","green","green","green","yellow","yellow","yellow","yellow","black")

sortData<-frame_AA_RT[order(frame_AA_RT$RetTime),]                           # sort data frame according to retention time
sortData["number"]<-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25) # add numbers for x-axis

color <- as.character(sortData$color_AA)


# Open PDF file and create graphics

pdf(pdf_name)
plot(sortData$number,sortData$RetTime, xaxt="n", xlab="", pch=19, cex=3.5, col=color, ylab="best fit to experimental retention data (scans/residue)", main="predicted retention coefficients of amino acids")
grid(NA, NULL, lty=2, col="black")
points(sortData$number,sortData$RetTime,pch=19, cex=3.5, col=color)
text(sortData$number,sortData$RetTime,sortData$AminoAcids, col="white",font=2)
legend("topleft",inset=.02, c("small nonpolar","hydrophobic", "polar", "acidic", "basic", "modified residue", "offset"), pch=19, pt.cex=1.5, col=c("orange", "green", "magenta", "red", "blue", "yellow", "black"),bg="white")

dev.off() # close PDF file




 
</pre></div><div><h3>R Server</h3>localhost:6311</div>                </td>
              </tr>
                          <tr>
                <td><b>Tandem</b></td>
                <td>beanshell</td>
                <td>
                  This Beanshell sets up and runs X!Tandem with the provided FASTA file and LC-MS/MS data in mzXML.                  <div><h3>Script</h3><pre>/****************** Input descriptions ****************/
// input port: String mzxml_file (Depth 0)
// input port: String fasta_file (Depth 0)
// input port: String parameter_file (Depth 0)
// input port: String log_file (Depth 0)
// output port: String tandem_file (Depth 0)
// output port: String pepxml_file (Depth 0)
// output port: String interact_file (Depth 0)

/****************** Class Declarations ****************/
class InProcStreamReader extends Thread
{
    private StringBuffer sb;
    public  InputStream is;
    
    InProcStreamReader(InputStream is) {
        this.is = is;
    }// InProcStreamReader(InputStream is) 
    
    public void run() {
        try {
            InputStreamReader isr = new InputStreamReader(is);
            sb = new StringBuffer();
            while ((c = isr.read()) != -1)
                sb.append((char) c);
        } catch (IOException ioe)
        {
          ioe.printStackTrace();  
        }/// catch
    }// run

    public String getOutput() { 
      return sb.toString();
    }// getOutput
}// InProcStreamReader


/****************** Global variables ******************/
tppDir = "C:\\Inetpub\\tpp-bin\\";
installDir = "C:\\Taverna\\programs\\";

/****************** Global functions ******************/
// Replaces substrings in a string with a replacement string
String replace(String str, String pattern, String replace) { 
  int s = 0; 
  int e = 0; 
  StringBuffer result = new StringBuffer(); 
  while ((e = str.indexOf(pattern, s)) >= 0) { 
    result.append(str.substring(s, e)); 
    result.append(replace); 
    s = e+pattern.length(); 
  } // while
  result.append(str.substring(s)); 
  return result.toString(); 
}//  String replace(String str, String pattern, String replace) 

// Generates a file based on a name and a new extension (without the .)
String genNewFileName(String inputFile, String extension) {
  int dotPos = inputFile.lastIndexOf(".");
  String strFilename = inputFile.substring(0, dotPos);
  return strFilename + "." + extension;
}// String genNewFileName(String inputFile, String extension) 

// Generates tandem output file name
String genOutputFileName(String inputFile) {
  return genNewFileName(inputFile, "tandem");
}// String genOutputFileName(String inputFile)

// Generates taxonomy file name
String genTaxonomyFileName(String inputFile) {
  return genNewFileName(inputFile, "taxonomy.xml");
}// String genTaxonomyFileName((String inputFile)

// Modifies the taxonomy predicate file and returns the path of the new file */
String modTaxonomyPredicate(String fasta_location, String inputFile) {   
  String taxFileName = genTaxonomyFileName( inputFile );
  BufferedReader input = new BufferedReader( new FileReader( installDir + "Tools\\Tandem\\taxonomy_pred.xml" ));
  BufferedWriter output = new BufferedWriter( new FileWriter( taxFileName ));
  try {
    String line = null; 
    while (( line = input.readLine()) != null){      
      if (line.indexOf("$FASTA_LOCATION$") >= 0) {  
        line = replace( line, "$FASTA_LOCATION$", fasta_location );
        output.write( line + '\n');
      } else {
        output.write( line + '\n');
      }// else
    }// while
  }// try
  finally {
    input.close();
    output.close();
  }// finally

  return taxFileName;

}// String modTaxonomyPredicate(String fasta_location) 

// Generates parameter file name
String genParameterFileName(String inputFile) {
  return genNewFileName(inputFile, "param.xml");
}// String genParameterFileName((String inputFile)

// Modifies the param predicate file and returns the path of the new file */
String modParamPredicate(String mzxmlFile, String taxonFile, String paramFile, String outputFile) {   
  BufferedReader input = new BufferedReader(new FileReader(installDir + "Tools\\Tandem\\tandem_pred.params"));
  BufferedReader parinput = new BufferedReader(new FileReader(paramFile));
  String paramFileName = genParameterFileName( mzxmlFile );
  BufferedWriter output = new BufferedWriter( new FileWriter( paramFileName ));
  String line = null; //not declared within while loop
  try {
    while (( line = input.readLine()) != null){
      if (line.indexOf("$PARAMS$") >= 0) {
        while (( line = parinput.readLine()) != null){
          output.write( line + '\n'); 
        }
      } else if (line.indexOf("$TAXONOMY_LOCATION$") >= 0) {
        line = replace( line, "$TAXONOMY_LOCATION$", taxonFile );
        output.write( line + '\n'); 
      } else if (line.indexOf("$SPECTRUM_LOCATION$") >= 0) {
        line = replace( line, "$SPECTRUM_LOCATION$", mzxmlFile );
        output.write( line + '\n');
      } else if (line.indexOf("$OUTPUT_LOCATION$") >= 0) {
        line = replace( line, "$OUTPUT_LOCATION$", outputFile );
        output.write( line + '\n');
      } else {
        output.write( line + '\n');
      }// else
    }// while
  }// try
  finally {
    input.close();
    parinput.close();
    output.close();
  }// finally

  return paramFileName;

}// String modParamPredicate(String mzxmlFile, String taxonFile, String outputFile) 


/************** Main matter, input checking ************/
if (mzxml_file == void || mzxml_file.equals("")) {
	throw new RuntimeException("The 'mzxml_file' port cannot be null.");
} else if (fasta_file == void || fasta_file.equals("")) {
	throw new RuntimeException("The 'fasta_file' port cannot be null.");
} else if (parameter_file == void || parameter_file.equals("")) {
	throw new RuntimeException("The 'parameter_file' port cannot be null.");
}// else

/************ Main matter, Running executable **********/
// Setting up the command line 
appArray = new ArrayList();

// Adding the command itself
appArray.add(tppDir + "Tandem.exe");

// Creating tandem configuration files and setting them as input
taxonFile = modTaxonomyPredicate(fasta_file, mzxml_file);
tmpout = genOutputFileName(mzxml_file);
appArray.add(modParamPredicate(mzxml_file, taxonFile, parameter_file, tmpout));

// Preparing execution of tandem
workDir = new File(tppDir);
launcher = new ProcessBuilder(appArray);
launcher.directory = workDir;
proc = launcher.start();

outputReader = new InProcStreamReader(proc.getInputStream());
outputReader.start();

// any error???
exitVal = proc.waitFor();
outputReader.join();


/************** Main matter, generating log ************/
fstream = new FileWriter(log_file, true);
out = new BufferedWriter(fstream);

if (!(log_file == void || log_file.equals(""))) {
  out.write("=============== X!Tandem ==============="); 
  out.newLine();
  out.write("Parameters: ");
  for (i=0; i<appArray.size(); i++)
    out.write(appArray.get(i).toString());
  out.newLine();
  out.write("Output:");
  out.newLine();
  out.write(outputReader.getOutput()); 
  out.newLine();
  out.close();
}// if

// final step, assigning the output
if (exitVal == 0)
  {
   tandem_file = tmpout;
   pepxml_file = tandem_file + ".pep.xml";
   interact_file = tandem_file + ".interact.pep.xml";
}
else
{
  tandem_file = NULL;
  pepxml_file = NULL;
  interact_file = NULL;
}
 

</pre></div>                </td>
              </tr>
                          <tr>
                <td><b>Tandem2XML</b></td>
                <td>externaltool</td>
                <td>
                  This tool calls the Tandem2XML converter for transforming X!Tandem results to pepXML.                                  </td>
              </tr>
                          <tr>
                <td><b>PeptideProphet_Tandem</b></td>
                <td>externaltool</td>
                <td>
                  The PeptideProphet_Tandem tool runs PeptideProphet (in its default location after a standard installation of the Trans-Proteomic Pipeline).                                  </td>
              </tr>
            
          </table>
        
      </div>
    </div>
    
    <!-- Beanshells -->
        <div class="fold">
      <div class="foldTitle">
        Beanshells (2)
      </div>
      <div class="foldContent" style="display: none;">
                  <table class="simple">
            <tr>
              <th>Name</th>
              <th>Description</th>
              <th>Inputs</th>
              <th>Outputs</th>
            </tr>
          
                          <tr>
                <td><b>Join_and_Insert_Tabs</b></td>
                <td>This Beanshell formats the data for rt.</td>
                <td>
                  
                    
                      in1                      <hr />
                    
                      in2                      
                    
                  
                </td>
                <td>
                  
                    
                      out1                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>Tandem</b></td>
                <td>This Beanshell sets up and runs X!Tandem with the provided FASTA file and LC-MS/MS data in mzXML.</td>
                <td>
                  
                    
                      mzxml_file                      <hr />
                    
                      fasta_file                      <hr />
                    
                      parameter_file                      <hr />
                    
                      log_file                      
                    
                  
                </td>
                <td>
                  
                    
                      tandem_file                      <hr />
                    
                      pepxml_file                      <hr />
                    
                      interact_file                      
                    
                  
                </td>
              </tr>
            
          </table>
        
      </div>
    </div>
    
    <!-- Sinks -->
        <div class="fold">
      <div class="foldTitle">
        Outputs (2)
      </div>
      <div class="foldContent" style="display: none;">
                  <table class="simple">
            <tr>
              <th>Name</th>
              <th>Description</th>
            </tr>
          
                          <tr>
                <td><b>Predicted_RT_Peptides</b></td>
                <td>
                  
                </td>
              </tr>
                          <tr>
                <td><b>Graphic_RetentionCoefficient_AA</b></td>
                <td>
                  
                </td>
              </tr>
            
          </table>
        
      </div>
    </div>
    
    <!-- Datalinks -->
        <div class="fold">
      <div class="foldTitle">
        Datalinks (21)
      </div>
      <div class="foldContent" style="display: none;">
                  <table class="simple">
            <tr>
              <th>Source</th>
              <th>Sink</th>
            </tr>
          
                          <tr>
                <td>Tandem:interact_file</td>
                <td>Read_Text_File:fileurl</td>
              </tr>
                          <tr>
                <td>Read_Text_File:filecontents</td>
                <td>Extract_Peptides:xml_text</td>
              </tr>
                          <tr>
                <td>Read_Text_File:filecontents</td>
                <td>Extract_Scan_Numbers:xml_text</td>
              </tr>
                          <tr>
                <td>Extract_Peptides:nodelist</td>
                <td>Join_and_Insert_Tabs:in1</td>
              </tr>
                          <tr>
                <td>Extract_Scan_Numbers:nodelist</td>
                <td>Join_and_Insert_Tabs:in2</td>
              </tr>
                          <tr>
                <td>Join_and_Insert_Tabs:out1</td>
                <td>rt:in1</td>
              </tr>
                          <tr>
                <td>rt:STDOUT</td>
                <td>Plot_Predicted_vs_Measured_RT:in2</td>
              </tr>
                          <tr>
                <td>Join_and_Insert_Tabs:out1</td>
                <td>Plot_Predicted_vs_Measured_RT:in1</td>
              </tr>
                          <tr>
                <td>Tandem:interact_file</td>
                <td>Plot_Predicted_vs_Measured_RT:in3</td>
              </tr>
                          <tr>
                <td>rt:STDOUT</td>
                <td>Plot_Retention_Coefficients:in1</td>
              </tr>
                          <tr>
                <td>Tandem:interact_file</td>
                <td>Plot_Retention_Coefficients:in2</td>
              </tr>
                          <tr>
                <td>FASTA_File</td>
                <td>Tandem:fasta_file</td>
              </tr>
                          <tr>
                <td>mzXML_File</td>
                <td>Tandem:mzxml_file</td>
              </tr>
                          <tr>
                <td>Log_File</td>
                <td>Tandem:log_file</td>
              </tr>
                          <tr>
                <td>Tandem_Param_File</td>
                <td>Tandem:parameter_file</td>
              </tr>
                          <tr>
                <td>Tandem:tandem_file</td>
                <td>Tandem2XML:in1</td>
              </tr>
                          <tr>
                <td>Tandem:pepxml_file</td>
                <td>Tandem2XML:in2</td>
              </tr>
                          <tr>
                <td>Tandem:pepxml_file</td>
                <td>PeptideProphet_Tandem:in1</td>
              </tr>
                          <tr>
                <td>Tandem:interact_file</td>
                <td>PeptideProphet_Tandem:in2</td>
              </tr>
                          <tr>
                <td>Plot_Predicted_vs_Measured_RT:output_file_name</td>
                <td>Predicted_RT_Peptides</td>
              </tr>
                          <tr>
                <td>Plot_Retention_Coefficients:output_file_name</td>
                <td>Graphic_RetentionCoefficient_AA</td>
              </tr>
            
          </table>
        
      </div>
    </div>
    
    <!-- Coordinations -->
        <div class="fold">
      <div class="foldTitle">
        Coordinations (2)
      </div>
      <div class="foldContent" style="display: none;">
                  <table class="simple">
            <tr>
              <th>Controller</th>
              <th>Target</th>
            </tr>
          
                          <tr>
                <td>PeptideProphet_Tandem</td>
                <td>Read_Text_File</td>
              </tr>
                          <tr>
                <td>Tandem2XML</td>
                <td>PeptideProphet_Tandem</td>
              </tr>
            
          </table>
        
      </div>
    </div>
  
    

          
      
    </div>

  </div>

</div>

<div class="contribution_right_box">
  <div class="contribution_section_box" style= "font-size: 100%; padding: 0.7em 0.9em; font-weight: bold;">
    <p><img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[The type of workflow system that this Workflow is designed for.] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" /> Workflow Type</p>
    <p><a href="/content_types/2">Taverna 2</a></p>
  </div>

  <div class="contribution_section_box">
	<p class="heading">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[The Uploader is the person who uploaded/created this Workflow on myExperiment, and who manages it on myExperiment.] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Uploader
	</p>
	<p>
		<center>
			<div class="avatar" style="width: 66px">
  <a href="/users/23068"><img alt="3655?size=60x60" class="framed" src="/pictures/show/3655?size=60x60" title="Magnus Palmblad" /></a>
  
	<span style="display:block;margin-top:2px;text-align:center;">
		
			<img alt="Nl" src="/images/famfamfam_flags/nl.png?1680607579" style="vertical-align:middle; margin-right: 0.3em;" title="header=[] body=[&lt;b&gt;Location: &lt;/b&gt;Leiden, Netherlands] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		

    
			<a href="/users/23068" title="header=[] body=[Magnus Palmblad] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Magnus Palmblad</a>
		

    
	</span>
  
</div>

		</center>
	</p>
</div>


  

  

<div class="contribution_section_box">
	<p class="heading">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[This specifies what rights users have when they download and use this Workflow.] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		License
		<a name="license"></a>
	</p>

  

    
    
    
    
    


    

    <!--

    <rdf:RDF xmlns="http://creativecommons.org/ns#"
        xmlns:dc="http://purl.org/dc/elements/1.1/"
        xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#">
    <Work rdf:about="">
       <dc:title>Retention Time Prediction with X!Tandem</dc:title>
       <dc:date>2012-09-03 10:43:30 UTC</dc:date>
       <dc:description>&lt;p&gt;This workflow identifies peptides from tandem mass spectra using X!Tandem as in a standard installation of the Trans-Proteomic Pipeline (TPP, version 4.6, but earlier versions should also work). The peptide-spectrum matches are validates using PeptideProphet (also from a standard installation of TPP) and peptides with at least a probability 0.95 are used to train a linear retention time predictor (Palmblad et al., 2002), whereby retention coefficients are also derived. These indirectly provide information on the chromatographic stationary and mobile phase (composition and pH) and the gradient.  The two Rshell scrips visualize the prediction versus measured retention times and amino acid coefficients colored by physico-chemical properties (basic/acidic/polar etc.).&lt;/p&gt;</dc:description>
       <dc:creator><Agent>
          <dc:title><a href="/users/23068" title="header=[] body=[Magnus Palmblad] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Magnus Palmblad</a></dc:title>
       </Agent></dc:creator>
       <dc:rights><Agent>
          <dc:title><a href="/users/23068" title="header=[] body=[Magnus Palmblad] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Magnus Palmblad</a></dc:title>
       </Agent></dc:rights>
       <dc:type rdf:resource="http://purl.org/dc/dcmitype/Data" />
       <dc:source rdf:resource="http://www.myexperiment.org/workflows/3134.html"/>
       <license rdf:resource="http://creativecommons.org/licenses/by-sa/3.0/" />
    </Work>

    <License rdf:about="http://creativecommons.org/licenses/by-sa/3.0/">
      <permits rdf:resource="http://creativecommons.org/ns#Reproduction" />  <permits rdf:resource="http://creativecommons.org/ns#Distribution" />  <requires rdf:resource="http://creativecommons.org/ns#Notice" />  <requires rdf:resource="http://creativecommons.org/ns#Attribution" />  <permits rdf:resource="http://creativecommons.org/ns#DerivativeWorks" />  <requires rdf:resource="http://creativecommons.org/ns#ShareAlike" />
    </License>

    </rdf:RDF>

    -->

  

  <div class="contribution_currentlicense">
    <p>
      
        All versions of this Workflow are
      
      
        licensed under: <br/><br/><a rel="Copyright" href="/licenses/2" title="Creative Commons Attribution-Share Alike 3.0 Unported License"><img src="/images/by-sa.png" /></a>
      
    </p>
  </div>
	
</div>


  <script type="text/javascript">
    function showVersion(form) {
        var url = $("resource_versions").value;
        location.href = url;
        form.submit
    }
</script>
<style>
  .version_selection {
      font-size:smaller;
      text-align: center;
  }
</style>

<div class="contribution_section_box">
  <p class="heading">
    <img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[This box shows version 1 for this entry] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />    <span>Version 1 </span>
    <span class="count_text">(of 1)</span>
  </p>
  
</div>

  <div class="contribution_section_box">
	
	<p class="heading" style="margin-bottom: 0; padding-bottom: 0;">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[Which people/groups get the credit for this Workflow?] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Credits <span class="count_text">(3)</span>
		<a name="credits"></a>
	</p>
	<span style="font-size: 85%; color: #666666; display: block; padding: 0.3em 0.5em 0 0.5em; margin: 0; line-height: 1em;">(People/Groups)</span>
	
		<div style="font-size: 85%;">
			<ul class="items">
				
					<li>
						
							
								<span class="icon"><img alt="User" src="/images/famfamfam_silk/user.png?1680607579" /> </span> 
							
							<a href="/users/23068" title="header=[] body=[Magnus Palmblad] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Magnus Palmblad</a>
						
					</li>
				
					<li>
						
							
								<span class="icon"><img alt="User" src="/images/famfamfam_silk/user.png?1680607579" /> </span> 
							
							<a href="/users/17146" title="header=[] body=[Yassene] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Yassene</a>
						
					</li>
				
					<li>
						
							
								<span class="icon"><img alt="User" src="/images/famfamfam_silk/user.png?1680607579" /> </span> 
							
							<a href="/users/17730" title="header=[] body=[Eleni] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Eleni</a>
						
					</li>
				
			</ul>
		</div>
	
	
	
	<p class="heading" style="margin-bottom: 0; padding-bottom: 0;">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[What other Workflows/Files is this one based on?] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Attributions <span class="count_text">(1)</span>
		<a name="attributions"></a>
	</p>
	<span style="font-size: 85%; color: #666666; display: block; padding: 0.3em 0.5em 0 0.5em; margin: 0; line-height: 1em;">(Workflows/Files)</span>
	
		<div style="font-size: 85%;">
			<p>
				<ul class="items">
					
						<li>
							
								
									
										<span class="icon"><img alt="Blob" src="/images/redmond_studio/documents_16.png?1680607579" /> </span>
									 
									<a href="/files/830">rt3.c</a>
								
						  
						</li>
					
				</ul>
			</p>
		</div>
	
	
	
	
</div>


  


  <div class="contribution_section_box">
	<p class="heading">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[The tags for this Workflow entry.&lt;br/&gt;Note(1): if this Workflow has many versions, the tags apply to ALL versions in this entry.&lt;br/&gt;Note(2): the size of the tags show how popular they are on myExperiment.] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Tags <span id="tags_box_header_tag_count_span" class="count_text">(5)</span>
		<a name="tags"></a>
	</p>

	<script type="text/javascript">
  function showAllTags() {
    Element.show('all_tags');
	  Element.hide('user_tags');
	  return false;
  }
  
  function showUsersTags() {
  	Element.hide('all_tags');
	  Element.show('user_tags');
	  return false;
  }
	
	function flip_tags_mode()
	{
		if( $('toggle_tags_mode').checked )
		  showUsersTags();
	  else
		  showAllTags();
	}
</script>

<div id="tags_inner_box">
	
		<div class="box_currentuser_specific" style="padding: 0.3em 0.2em; margin: 0.6em 0.4em;">
		  <label for="toggle_tags_mode" style="font-size: 85%; font-weight: bold;"><input id="toggle_tags_mode" name="toggle_tags_mode" onclick="flip_tags_mode();" style="margin-right: 0.3em;" type="checkbox" value="1" />Uploader tags</label>
		</div>
		<div id="all_tags">
			
				<div class="hTagcloud">
  <ul class="popularity">
    <li><a href='/tags/3631' class='popular'>liquid chromatography</a></li>
<li> | </li>
    <li><a href='/tags/298' class='vvvv-popular'>mass spectrometry</a></li>
<li> | </li>
    <li><a href='/tags/373' class='v-popular'>peptide</a></li>
<li> | </li>
    <li><a href='/tags/405' class='vvv-popular'>proteomics</a></li>
<li> | </li>
    <li><a href='/tags/3630' class='popular'>retention time prediction</a></li>
  </ul>
</div>

			
		</div>
		<div id="user_tags"  style="display: none;">
			
				<div class="hTagcloud">
  <ul class="popularity">
    <li><a href='/tags/3631' class='popular'>liquid chromatography</a></li>
<li> | </li>
    <li><a href='/tags/298' class='vvvv-popular'>mass spectrometry</a></li>
<li> | </li>
    <li><a href='/tags/373' class='v-popular'>peptide</a></li>
<li> | </li>
    <li><a href='/tags/405' class='vvv-popular'>proteomics</a></li>
<li> | </li>
    <li><a href='/tags/3630' class='popular'>retention time prediction</a></li>
  </ul>
</div>

			
		</div>
	
</div>
	
	
	
	
		<p><small><b><a href="/session/new">Log in</a></b> to add Tags</small></p>
	
</div>


  
<div class="contribution_section_box">
	<p class="heading">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[All the Groups that this item has been shared with] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Shared with Groups <span class="count_text">(0)</span>
	</p>
	
		<p class="none_text">None</p>
	
</div>

  <div class="contribution_section_box" id="in_packs_box">
  
	<p class="heading">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[All the Packs that this item (or versions of this item [[if available]]) is referred to in] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Featured In Packs  <span class="count_text">(0)</span>
		<a name="featured_in_packs"></a>
	</p>
	
		<p class="none_text">None</p>
	
	
	
		<p><small><b><a href="/session/new">Log in</a></b> to add to one of your Packs</small></p>
	
</div>


  <div class="contribution_section_box">
	
	<p class="heading" style="margin-bottom: 0; padding-bottom: 0;">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[Shows files and workflows that have attributed this Workflow.] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Attributed By <span class="count_text">(3)</span>
	  <a name="attributed_by"></a>
	</p>
	<span style="font-size: 85%; color: #666666; display: block; padding: 0.3em 0.5em 0 0.5em; margin: 0; line-height: 1em;">(Workflows/Files)</span>
	
		
		<div style="font-size: 85%;">
			<p>
				<ul class="items">
					
						<li>
							
								
									
										<span class="icon"><img alt="Workflow" src="/images/redmond_studio/applications_16.png?1680607579" /> </span> 
									 
									<a href="/workflows/3691">Optimization of retention time prediction</a>
								
							
						</li>
					
						<li>
							
							  <span class="none_text">Item doesn't exist anymore</span>
							
						</li>
					
						<li>
							
							  <span class="none_text">Item doesn't exist anymore</span>
							
						</li>
					
				</ul>
			</p>
		</div>
	
	
</div>


  
<div class="contribution_section_box">
	<p class="heading">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[The users who have added this Workflow to their favourites] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Favourited By <font class="count_text">(0)</font>
		<a name="favourited_by"></a>
	</p>
	
	
		<p class="none_text">No one</p>
	
	
	
</div>

  

<div class="contribution_section_box">
	<p class="heading" style="margin-bottom: 0.3em;">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[Viewing / download counts are grand totals for this Workflow.&lt;br/&gt;Click &#39;see breakdown&#39; to view details.] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Statistics
	</p>
	
	<div class="stats_box"> 
		<p style="font-size: 85%;">
  1675 viewings
</p>

	<p style="font-size: 85%;">
	  954 downloads
	</p>

<p style="margin-top: 0.2em;">
	<small>[ <a href="/workflows/3134/statistics">see breakdown</a> ]</small>
</p>
		
		<div class="fold" style="padding-top: 0.6em;">
			<div class="foldTitle">More</div>
			<div class="foldContent" style="display: none; text-align: left;">
				
					<p>
						<a href="#credits">3 credits</a> <!-- won't work for type=='group'; needs anchor called '#group_creditations' -->
					</p>
					
					<p>
						<a href="#attributions">1 attribution</a>
					</p>
				
				
				
					<p>
			      <a href="#tags">5 tags</a>
			    </p>
				
				
				
					<p>
			      <a href="#favourited_by">0 favourites</a>
			    </p>
				

				
					<p>
						<a href="#citations">0 citations</a>
			    </p>
				
				
				
					<p>
			      <a href="#reviews">0 reviews</a>
			    </p>
				
				
				
			    <p>
			    	<a href="#comments">1 comment</a>
			    </p>
				
			</div>
		</div>
		
		
		
		
	</div>
</div>

</div>

<div class="clearer">&nbsp;</div>

<!-- BEGIN tabs -->

<br/>

<div id="tabsContainer" class="tabsContainer"></div>

<div class="tabContainer">
  <div class="tabTitle">Citations (0)</div>
  <div class="tabContent">

    <a name="citations"></a>
    
      <p><i>None</i></p>
    
		
		<br/>

    

  </div>
</div>


<div class="tabContainer">
  <div class="tabTitle">Version History</div>
  <div class="tabContent">
    <a name="version_history"></a>
    <p>In chronological order:</p>

<ol class="version_history">
  
    <li>
      <div class="title">
        <a href="/workflows/3134/versions/1">Retention Time Prediction with X!Tandem</a>
      </div>
      <div class="metadata">

        <p>
          Created
          
            by <a href="/users/23068" title="header=[] body=[Magnus Palmblad] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Magnus Palmblad</a>
          
          on Monday 03 September 2012 10:43:30 (UTC)
        </p>

        
          <p>
            Last edited
            
              by <a href="/users/23068" title="header=[] body=[Magnus Palmblad] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Magnus Palmblad</a>
            
            on Monday 03 September 2012 10:44:32 (UTC)
          </p>
        

        
      </div>
    </li>
  
</ol>


  </div>
</div>



<!-- END tabs -->

<br/>
<br/>

<div id="ratingsBox">
  <h2>
	<span class="icon"><img alt="Reviews" src="/images/famfamfam_silk/report.png?1680607579" /> </span>	<span style="vertical-align: middle;"><a href="/workflows/3134/reviews">Reviews</a></span>
	<span class="count_text" style="vertical-align: middle;">(0)</span>
</h2>

<div class="reviewsBox">

	
		<p><i>No reviews yet</i></p>
		<a name="reviews"></a>
		<p style="margin-top: 0.6em;">
			<strong><a href="/workflows/3134/reviews/new">Be the first to review!</a></strong>
		</p>
	
	
</div>

</div>

<br/>
<br/>

<div id="commentsBox">
  <script src="/javascripts/ckeditor/ckeditor.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/widgets/tooltip.js?1680607579" type="text/javascript"></script>

<h2>
	<span class="icon"><img alt="Comments" src="/images/famfamfam_silk/comments.png?1680607579" /> </span>	<span style="vertical-align: middle;">Comments</span>
	<span class="count_text" style="vertical-align: middle;">(1)</span>
</h2>

<ul class="sectionIcons">
  <li><span class="icon"><a href="http://www.myexperiment.org/workflows/3134/comments/timeline"><img alt="Timeline" src="/images/famfamfam_silk/timeline_marker.png?1680607579" /> View Timeline</a></span></li>
</ul>

<div class="commentsBox">
	
	
		
			<p><strong><a href="/session/new">Log in</a> to make a comment </strong></p>
		
		<a name="comments"></a>
		<ul id="comments" class="comments">
			<li>
	<a name='comment_701'></a>
  <table class="comment">
    <tr>
      <td class="avatar" rowspan="3">
        <div class="avatar" style="width: 66px">
  <a href="/users/23068"><img alt="3655?size=60x60" class="framed" src="/pictures/show/3655?size=60x60" title="Magnus Palmblad" /></a>
  
	<span style="display:block;margin-top:2px;text-align:center;">
		
			<img alt="Nl" src="/images/famfamfam_flags/nl.png?1680607579" style="vertical-align:middle; margin-right: 0.3em;" title="header=[] body=[&lt;b&gt;Location: &lt;/b&gt;Leiden, Netherlands] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		

    
			<a href="/users/23068" title="header=[] body=[Magnus Palmblad] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Magnus Palmblad</a>
		

    
	</span>
  
</div>

      </td>
      <td class="header" height="1.2em">
	      Monday 03 September 2012 10:47:29 (UTC)
	  </td>
    </tr>
	<tr>
	  <td class="content">        
    	<p>This is the local-only version of the workflow. The Trans-Proteomic Pipeline needs to be installed in the standard directories and rt 3.0 should be available where specified in the rt component.</p><p>A version with a Web service rt is also available.</p>
	  </td>
	</tr>
	<tr>
		<td style="text-align: right; padding-right: 1em;">
      
      
		</td>
	</tr>
  </table>
</li>

		</ul>
	
	
	<br/>
	
	

</div>


</div>

<!-- Other workflows that use a service this workflow users -->

<br />
<br />

<h2>
  <span class="icon"><img alt="Workflow" src="/images/redmond_studio/applications_16.png?1680607579" /> </span>  <span style="vertical-align: middle;">Other workflows that use similar services</span>
  <span class="count_text" style="vertical-align: middle;">(0)</span>
</h2>




    <p>There are no workflows in myExperiment that use similar services to this Workflow.</p>



<!-- OpenURL context object -->

<span class="Z3988" title="url_ver=Z39.88-2004&amp;url_ctx_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Actx&amp;ctx_ver=Z39.88-2004&amp;ctx_tim=2023-08-16T14%3A24%3A15%2B00%3A00&amp;ctx_id=&amp;ctx_enc=info%3Aofi%2Fenc%3AUTF-8&amp;rft.title=Retention+Time+Prediction+with+X%21Tandem&amp;rft.date=2012-09-03&amp;rft.au=Magnus+Palmblad&amp;rft.genre=unknown&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Adc"></span>
      </div>
    </div>




            <div id="ft">
          <div class="container">
  <div class="row">
    <div class="col-md-4 text-left">
      Funded by:
      <a href="http://biovel.eu">BioVeL</a>,
      <a href="http://www.epsrc.ac.uk/">EPSRC</a>,
      <a href="http://jisc.ac.uk/">JISC</a>,
      <a href="http://www.microsoft.com">Microsoft</a>

      <br/><br/>

      Affiliates:
      <a href="http://www.mygrid.org.uk">myGrid</a>,
      <a href="http://www.taverna.org.uk/">Taverna</a>,
      <a href="http://www.biocatalogue.org">BioCatalogue</a>
    </div>

    <div class="col-md-4 text-center">
      <a href="/about">About</a> |
      <a href="/privacy">Privacy</a> |
      <a href="http://wiki.myexperiment.org/index.php/Papers">Publications</a> |
      <a href="/feedback">Contact / Feedback</a> |
      <a href="http://lists.nongnu.org/mailman/listinfo/myexperiment-discuss">Mailing List</a>
    </div>

    <div class="col-md-4 text-right">
      Icons: <a href="http://www.famfamfam.com/lab/icons/silk/">Silk icon set 1.3</a><br/><br/>
      
    </div>

  </div>
  <div class="row">
    <div class="col-md-12 text-center">
      Copyright &#169; 2007 &ndash; 2018
      <a href="http://www.manchester.ac.uk/">The University of Manchester</a>
      and
      <a href="http://www.soton.ac.uk/">University of Southampton</a>
    </div>
  </div>
</div>

        </div>
    



<script type="text/javascript">
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
        (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
        m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-1498417-4', 'auto');
    ga('set', 'anonymizeIp', true);
    ga('send', 'pageview');
</script>




</body>
</html>

