<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
    "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <meta http-equiv="content-type" content="text/html;charset=UTF-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1">
  
  <meta name="google-site-verification" content="3hq0Cfhw5_-EaaNMum0ETX847sTl7jeQ3ZNAla4a6UQ" />
  
  <title>myExperiment &#45; Workflows &#45; Comparison of Peptide and Protein Fractionation Methods (Kate Mostovenko) [Taverna 2 Workflow]</title>
  <link rel="alternate" href="http://www.myexperiment.org/workflows/3486.rdf" type="application/rdf+xml" title="RDF+XML" />
  <link rel="alternate" href="http://www.myexperiment.org/workflows/3486.xml" type="application/xml" title="REST XML" />
  <link rel="shortcut icon" href="/favicon.ico" type="image/x-icon"/>
  <link href="/stylesheets/acts_as_taggable_stylesheet.css?1680607579" media="screen" rel="stylesheet" type="text/css" />
<link href="/stylesheets/star_rating.css?1680607579" media="screen" rel="stylesheet" type="text/css" />
<link href="/stylesheets/gadgets.css?1680607579" media="screen" rel="stylesheet" type="text/css" />
<link href="/stylesheets/misc_skinning.css?1680607579" media="screen" rel="stylesheet" type="text/css" />
<link href="/stylesheets/pagination.css?1680607579" media="screen" rel="stylesheet" type="text/css" />
<link href="/stylesheets/bootstrap.css?1680607579" media="screen" rel="stylesheet" type="text/css" />
<link href="/stylesheets/navbar.css?1680607579" media="screen" rel="stylesheet" type="text/css" />
<link href="/stylesheets/styles.css?1680607579" media="screen" rel="stylesheet" type="text/css" />
  <script src="/javascripts/jquery-1.11.1.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/jquery-noconflict.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/prototype-1.7.2-patched.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/effects.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/dragdrop.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/controls.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/boxover.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/bootstrap.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/tabs.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/folds.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/application.js?1680607579" type="text/javascript"></script>

  
  






  
  
    <script src='//www.google.com/recaptcha/api.js'></script>
  
</head>
<body>



<div class="navbar navbar-myexp" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      
          <a id="logo" href="/">
            <img alt="Logo" class="hidden-sm" src="/images/logo.png?1680607579" />
            <img alt="Logo_cropped" class="visible-sm" src="/images/logo_cropped.png?1680607579" />
          </a>
      
    </div>
    <div class="collapse navbar-collapse">
      <ul class="nav navbar-nav">
        
          <ul class="nav navbar-nav">
  
    <li >
      <a href="/home">Home</a>    </li>
  
    <li >
      <a href="/users">Users</a>    </li>
  
    <li >
      <a href="/groups">Groups</a>    </li>
  
    <li class="active">
      <a href="/workflows">Workflows</a>    </li>
  
    <li >
      <a href="/files">Files</a>    </li>
  
    <li >
      <a href="/packs">Packs</a>    </li>
  
</ul>

        
        
            <form action="/search" class="navbar-form navbar-left" id="search" method="get">
    <div class="input-group">
      <input class="form-control" id="query" name="query" placeholder="Search" type="text" />
      <span class="input-group-btn">
        
        <input class="btn btn-default" name="commit" type="submit" value="Go" />
        </span>
    </div>
</form>

        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
            <li><a href="/session/new?return_to=%2Fworkflows%2F3486.html">Log in</a></li>
    <li><a href="/users/new">Register</a></li>


      </ul>
    </div><!--/.nav-collapse -->
  </div>
  
</div>




    <div class="container" id="main">
      <div id="myexp_breadcrumbs_bar">
        <ol class="breadcrumb">
  <li><a href="/home">Home</a></li>
  


  <li><a href="/workflows">Workflows</a></li>



  
    <li>Comparison of Peptide and Protein Fractionation Methods</li>
  


  <div style="float: right"></div>
</ol>

      </div>
      <div id="myexp_content">
        


        



<h1 class="contribution_title">
  Comparison of Peptide and Protein Fractionation Methods
</h1>

<div class="contribution_aftertitle">
	<b>Created:</b> 2013-03-27 10:04:23
	
</div>
<ul class="sectionIcons">
  
      <li><span class="icon"><a href="http://www.myexperiment.org/workflows/3486/download/Comparison_of_Peptide_and_Protein_Fractionation_Methods-v1.t2flow?version=1"><img alt="Download Workflow (for version 1)" src="/images/redmond_studio/arrow-down_16.png?1680607579" /> Download Workflow</a></span></li>
      
  
      
</ul>


<div class="contribution_left_box">

  <div class="contribution_version_box">



    <div class="contribution_version_inner_box">

      
    <div class="contribution_description">
      <p>
This workflow was used to analyze the data in a manuscript  by Mostovenko et al. (2013, submitted), comparing peptide and protein fractionation methods. The workflow identifies proteins by X!Tandem database search and validates the results using PeptideProphet. Additional information such as pI and fraction number is extracted and plotted for IEF and SCX data. For each protein identified in SDS-PAGE derived data sequences are downloaded from UniProt and plotted against the fraction number. Rshell script produces the figures essentially as they appear in the manuscript.
</p>

<p>
Executing the workflow requires Rserve running and the Trans-Proteomic Pipeline (<a href="http://tools.proteomecenter.org/wiki/index.php?title=Software:TPP" rel="nofollow">http://tools.proteomecenter.org/wiki/index.php?title=Software:TPP</a>) installed with default settings and in the default location. The Rshell script contains the location where the figures will be generated. All other output files are stored in the input data folders by default. The version of X!Tandem called from this workflow is separate from the one installed with the Trans-Proteomic Pipeline (the location specified in the Tandem components).
</p>


    </div>


      <h3>
        <img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[This section shows the preview of this version of the Workflow (click on the image to expand)] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
        Preview
      </h3>

      

          <div class="contribution_preview" style="width: 100%;">
            <p style="text-align: center">
              <a href="/workflows/3486/versions/1/previews/full" onclick="window.open(this.href);return false;"><img alt="Medium" src="/workflows/3486/versions/1/previews/medium" /></a>
            </p>
            
                <p style="text-align: right; color: #666666;">
                  <a href="/workflows/3486/versions/1/previews/svg">Download as scalable diagram (SVG)</a>
                </p>
            

            
          </div>
      



      <h3>
        <img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[This section provides options for running this version of the Workflow] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
        Run
      </h3>

      
          
              <div style="margin: 0 0.5em;">
	<h4>
		Run this Workflow in the Taverna Workbench...
	</h4>
	
	<div class="option_box">
		
		<p style="text-align: center;">
			<b>Option 1:</b>
		</p>
		
		<p style="font-size: 93%; text-align: center;">
			Copy and paste this link into File > 'Open workflow location...'<br/>
			<a href="http://www.myexperiment.org/workflows/3486/download?version=1">http://www.myexperiment.org/workflows/3486/download?version=1</a><br/>
			<small>[ <a href="#" onclick="Effect.toggle(&quot;run_taverna_more_box&quot;,'blind',{duration:0.3});; return false;">More Info<img alt="Expand" src="/images/folds/unfold.png?1680607579" style="margin-left: 0.3em; vertical-align: middle;" /></a> ]</small>
		</p>
		
		<div id="run_taverna_more_box" style="display: none;">
			<p style="font-size: 85%; margin-top: 0.5em; text-align: center;">
				Taverna is available from <a href="http://taverna.sourceforge.net/" onclick="window.open(this.href);return false;">http://taverna.sourceforge.net/</a>
			</p>
			
			<p style="text-align: center; margin-top: 0.5em; font-size: 85%;">
				If you are having problems downloading it in Taverna, you may need to provide your username and password in the URL so that Taverna can access the Workflow: <br/>
				
					<b>Replace <font color="#990000">http://</font> in the link above with <font color="#990000">http://yourusername:yourpassword@</font></b>
				
			</p>
		</div>
	
	</div>
</div>


          
      

      <br/>

      <h3>
        <img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[This section shows the internal components of this workflow version] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
        Workflow Components
      </h3>

      
          
              
	    <!-- Authors -->
        <div class="fold">
      <div class="foldTitle">
        <img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[This is the author information extracted from the workflow version] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
        Authors (1)
      </div>
      <div class="foldContent" style="display: none;">
                  <table class="simple">
                          <tr>
                <td>Ekaterina Mostovenko (e.mostovenko@lumc.nl)
Jeroen de Bruin
Yassene Mohammed
Magnus Palmblad (n.m.palmblad@lumc.nl)</td>
              </tr>
            
          </table>
        
      </div>
    </div>
    
    <!-- Titles -->
        <div class="fold">
      <div class="foldTitle">
        <img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[These are the descriptive titles embedded within the workflow version] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
        Titles (1)
      </div>
      <div class="foldContent" style="display: none;">
                  <table class="simple">
                          <tr>
                <td>Comparison of Peptide and Protein Fractionation Methods</td>
              </tr>
            
          </table>
        
      </div>
    </div>

    <!-- Descriptions -->
        <div class="fold">
      <div class="foldTitle">
        <img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[These are the descriptions embedded within the workflow version] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
        Descriptions (1)
      </div>
      <div class="foldContent" style="display: none;">
                  <table class="simple">
                          <tr>
                <td>This workflow was used to analyze the data in a manuscript  by Mostovenko et al. (2013, submitted), comparing peptide and protein fractionation methods. The workflow identifies proteins by X!Tandem database search and validates the results using PeptideProphet. Additional information such as pI and fraction number is extracted and plotted for IEF and SCX data. For each protein identified in SDS-PAGE derived data sequences are downloaded from UniProt and plotted against the fraction number. Rshell script produces the figures essentially as they appear in the manuscript.

Executing the workflow requires Rserve running and the Trans-Proteomic Pipeline (http://tools.proteomecenter.org/wiki/index.php?title=Software:TPP) installed with default settings and in the default location. The Rshell script contains the location where the figures will be generated. All other output files are stored in the input data folders by default. The version of X!Tandem called from this workflow is separate from the one installed with the Trans-Proteomic Pipeline (the location specified in the Tandem components).</td>
              </tr>
            
          </table>
        
      </div>
    </div>
        
    <!-- Dependencies -->
        <div class="fold">
      <div class="foldTitle">
        <img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[These are the listed dependencies of the workflow] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
        Dependencies (0)
      </div>
      <div class="foldContent" style="display: none;">
        
          <p class="none_text">None</p>
        
      </div>
    </div>
        
    <!-- Sources -->
        <div class="fold">
      <div class="foldTitle">
        Inputs (9)
      </div>
      <div class="foldContent" style="display: none;">
                  <table class="simple">
            <tr>
              <th>Name</th>
              <th>Description</th>
            </tr>
          
                          <tr>
                <td><b>IEF_Dir</b></td>
                <td>
                  
                </td>
              </tr>
                          <tr>
                <td><b>IEF_Results_Dir</b></td>
                <td>
                  
                </td>
              </tr>
                          <tr>
                <td><b>SCX_Dir</b></td>
                <td>
                  
                </td>
              </tr>
                          <tr>
                <td><b>SCX_Results_Dir</b></td>
                <td>
                  
                </td>
              </tr>
                          <tr>
                <td><b>PAGE_Dir</b></td>
                <td>
                  
                </td>
              </tr>
                          <tr>
                <td><b>PAGE_Results_Dir</b></td>
                <td>
                  
                </td>
              </tr>
                          <tr>
                <td><b>Tandem_Param_File</b></td>
                <td>
                  
                </td>
              </tr>
                          <tr>
                <td><b>FASTA_File</b></td>
                <td>
                  
                </td>
              </tr>
                          <tr>
                <td><b>Prophet_Params</b></td>
                <td>
                  
                </td>
              </tr>
            
          </table>
        
      </div>
    </div>
      
    <!-- Processors -->
        <div class="fold">
      <div class="foldTitle">
        Processors (21)
      </div>
      <div class="foldContent" style="display: none;">
                  <table class="simple">
            <tr>
              <th>Name</th>
              <th>Type</th>
              <th>Description</th>
            </tr>
          
                          <tr>
                <td><b>Read_Text_File_PAGE</b></td>
                <td>localworker</td>
                <td>
                                    <div><h3>Script</h3><pre>BufferedReader getReader (String fileUrl, String encoding) throws IOException {
		InputStreamReader reader;
		try {
			if (encoding == null) {
				reader = new FileReader(fileUrl);
			} else {
				reader = new InputStreamReader(new FileInputStream(fileUrl),encoding); 
			}
		}
		catch (FileNotFoundException e) {
			// try a real URL instead
			URL url = new URL(fileUrl);
			if (encoding == null) {
				reader = new InputStreamReader (url.openStream());
			} else {
				reader = new InputStreamReader (url.openStream(), encoding);
			}
		}
		return new BufferedReader(reader);
	}



StringBuffer sb = new StringBuffer(4000);

if (encoding == void) {
	encoding = null;
}

BufferedReader in = getReader(fileurl, encoding);
String str;
String lineEnding = System.getProperty("line.separator");

while ((str = in.readLine()) != null) {
	sb.append(str);
	sb.append(lineEnding);
}
in.close();
filecontents = sb.toString();
  
</pre></div>                </td>
              </tr>
                          <tr>
                <td><b>Extract_Spectrum_Number_PAGE</b></td>
                <td>xpath</td>
                <td>
                                    <div><h3>Xpath Expression</h3><pre>/default:msms_pipeline_analysis/default:msms_run_summary/default:spectrum_query/@spectrum</pre></div>                </td>
              </tr>
                          <tr>
                <td><b>Extract_Protein_Names</b></td>
                <td>xpath</td>
                <td>
                                    <div><h3>Xpath Expression</h3><pre>/default:msms_pipeline_analysis/default:msms_run_summary/default:spectrum_query/default:search_result/default:search_hit/@protein</pre></div>                </td>
              </tr>
                          <tr>
                <td><b>Make_Well_Number_List_PAGE</b></td>
                <td>beanshell</td>
                <td>
                  The JavaBean to convert the spectrum information to the fraction number                  <div><h3>Script</h3><pre>int s = well.size();
String[] temp = new String[s];
String wellnum = new String();
int num1 = 0;
int num;
String num2 = new String();

for (i=0; i<well.size();i++){
	wellnum = well.get(i).charAt(31).toString();
	if (wellnum.equals("A")) num1 = 0;
	else num1 = 12;
	num2 = well.get(i).charAt(33).toString();
	num = num1 + Integer.parseInt(num2);
	temp[i] = num.toString();
}

out = temp;</pre></div>                </td>
              </tr>
                          <tr>
                <td><b>Get_Fasta</b></td>
                <td>beanshell</td>
                <td>
                  The JavaBean to extract protein sequences from the UniProt Database.                  <div><h3>Script</h3><pre>int s = protein.size();
String [] out1 = new String[s];
String [] out2 = new String[s];
String [] id = new String[3];
String [] seqsh = new String [2];
int MW = 0;

for (i=0; i<protein.size();i++){
	id = protein.get(i).split("\\|");
	out2[i] = id[2];
    if (id[1].length() == 8) {
        id[1] = id[1].substring(0,6);
    }
try {
	
URL url = new URL ("http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?rettype=fasta&db=protein&retmode=text&id=" + id[1]);

BufferedReader reader = new BufferedReader (new InputStreamReader(url.openStream()));
StringWriter writer = new StringWriter();

char[] buffer = new char[1024];
    while (true) {
        int r = reader.read(buffer);
        if (r <= 0) {
            break;
        }
        writer.write(buffer, 0, r);
    }
reader.close();
out1[i] = writer.toString();
}
catch (Exception e){
	out1[i] = "Sequence not found";
}
}

seq = out1;
ID = out2;</pre></div>                </td>
              </tr>
                          <tr>
                <td><b>Join_and_Insert_Tabs_PAGE</b></td>
                <td>beanshell</td>
                <td>
                                    <div><h3>Script</h3><pre>int s = in1.size();
String out = new String();

for (i=0; i<s;i++){
	out = out + in1.get(i) + "\t" + in2.get(i) + "\n";
}

out1 = out;</pre></div>                </td>
              </tr>
                          <tr>
                <td><b>Get_MW</b></td>
                <td>beanshell</td>
                <td>
                  This JavaBean component calculates protein molecular weights based on their sequences.                  <div><h3>Script</h3><pre>int s = seq.size();
String[] seqsh = new String[2];
String seqsh2 = new String();
double[] outMW = new double[s];

for (i=0; i<seq.size(); i++){
 double MW = 0;
 if (!seq.get(i).equals("Sequence not found")){
  seqsh = seq.get(i).split(">[^\n]+\n");
  seqsh2 = seqsh[1].replace("\n", "");
  seqsh2 = seqsh2.replace("[^A-Z]", "");
  String rep = new String();

  rep = seqsh2.replace("A", "71.04 ");
  rep = rep.replace("C", "103.01 ");
  rep = rep.replace("D", "115.03 ");
  rep = rep.replace("E", "129.04 ");
  rep = rep.replace("F", "147.07 ");
  rep = rep.replace("G", "57.02 ");
  rep = rep.replace("H", "137.06 ");
  rep = rep.replace("I", "113.08 ");
  rep = rep.replace("K", "128.09 ");
  rep = rep.replace("L", "113.08 ");
  rep = rep.replace("M", "131.04 ");
  rep = rep.replace("N", "114.04 ");
  rep = rep.replace("P", "97.05 ");
  rep = rep.replace("Q", "128.06 ");
  rep = rep.replace("R", "156.10 ");
  rep = rep.replace("S", "87.03 ");
  rep = rep.replace("T", "101.05 ");
  rep = rep.replace("V", "99.07 ");
  rep = rep.replace("W", "186.08 ");
  rep = rep.replace("Y", "163.06 ");
  rep = rep.replace("X", "0.0 ");
  rep = rep.replace("U", "168.96 ");

  seqnum = rep.split(" ", 0);

     for (int k=0; k<seqnum.length-1; k++){
	if(seqnum[k].length() > 0 && !seqnum[k].equals("") ){
  	MW = MW + Double.parseDouble(seqnum[k]);
	}
	else MW = MW;
     }
 } 
 else MW = 0;
 outMW[i] = MW/1000;
}
out = outMW;
</pre></div>                </td>
              </tr>
                          <tr>
                <td><b>MW_Plot</b></td>
                <td>rshell</td>
                <td>
                  The Rshell script to plot the calculated protein molecular weight against its localization on the gel (fraction number).                  <div><h3>Script</h3><pre>MW <- read.table(MassList, sep="\t");
l <- nrow(MW);
MW.m <- as.matrix(MW, nrow=l, ncol=2);
y <- max(MW.m[,2]);
ax <- c(1:24);
png("MW_plot.png");
plot (MW.m[,1],MW.m[,2], xlim = c(1,24),ylim=NULL, col=1,xlab="Fraction Number", ylab="Calculated Mass (kDa)",pch=10, cex=1, cex.lab=2, mar=c(5,20,5,5))
axis (1, at=ax)
dev.off()
</pre></div><div><h3>R Server</h3>localhost:6311</div>                </td>
              </tr>
                          <tr>
                <td><b>Database_Search</b></td>
                <td>workflow</td>
                <td>
                  For each dataset this nested workflow converts raw data to generic .mzXML format and passes it on to X!Tandem for search. Resulting files are converted to .pepXML and the probabilities are assigned to peptide-spectrum matches with PeptideProphet.                                  </td>
              </tr>
                          <tr>
                <td><b>Read_Text_File_IEF</b></td>
                <td>localworker</td>
                <td>
                                    <div><h3>Script</h3><pre>BufferedReader getReader (String fileUrl, String encoding) throws IOException {
		InputStreamReader reader;
		try {
			if (encoding == null) {
				reader = new FileReader(fileUrl);
			} else {
				reader = new InputStreamReader(new FileInputStream(fileUrl),encoding); 
			}
		}
		catch (FileNotFoundException e) {
			// try a real URL instead
			URL url = new URL(fileUrl);
			if (encoding == null) {
				reader = new InputStreamReader (url.openStream());
			} else {
				reader = new InputStreamReader (url.openStream(), encoding);
			}
		}
		return new BufferedReader(reader);
	}



StringBuffer sb = new StringBuffer(4000);

if (encoding == void) {
	encoding = null;
}

BufferedReader in = getReader(fileurl, encoding);
String str;
String lineEnding = System.getProperty("line.separator");

while ((str = in.readLine()) != null) {
	sb.append(str);
	sb.append(lineEnding);
}
in.close();
filecontents = sb.toString();
  
</pre></div>                </td>
              </tr>
                          <tr>
                <td><b>Extract_Parameters_IEF</b></td>
                <td>xpath</td>
                <td>
                                    <div><h3>Xpath Expression</h3><pre>/default:msms_pipeline_analysis/default:msms_run_summary/default:spectrum_query/default:search_result/default:search_hit/default:analysis_result/default:peptideprophet_result/default:search_score_summary/default:parameter[5]/@value</pre></div>                </td>
              </tr>
                          <tr>
                <td><b>Join_and_Insert_Tabs_IEF</b></td>
                <td>beanshell</td>
                <td>
                                    <div><h3>Script</h3><pre>int s = well.size();
String out = new String();

for (i=0; i<s;i++){
	out = out + well.get(i) + "\t" + pI.get(i) + "\n";
}

out1 = out;</pre></div>                </td>
              </tr>
                          <tr>
                <td><b>Extract_Spectrum_Number_IEF</b></td>
                <td>xpath</td>
                <td>
                                    <div><h3>Xpath Expression</h3><pre>/default:msms_pipeline_analysis/default:msms_run_summary/default:spectrum_query/@spectrum</pre></div>                </td>
              </tr>
                          <tr>
                <td><b>Make_Well_Number_List_IEF</b></td>
                <td>beanshell</td>
                <td>
                  The JavaBean to convert the spectrum information to the fraction number                  <div><h3>Script</h3><pre>int s = well.size();
String[] temp = new String[s];
String wellnum = new String();
int num1 = 0;
int num;
String num2 = new String();

for (i=0; i<well.size();i++){
	wellnum = well.get(i).charAt(31).toString();
	if (wellnum.equals("A")) num1 = 0;
	else num1 = 12;
	num2 = well.get(i).charAt(33).toString();
	num = num1 + Integer.parseInt(num2);
	temp[i] = num.toString();
}

out = temp;</pre></div>                </td>
              </tr>
                          <tr>
                <td><b>pI_Plot_IEF</b></td>
                <td>rshell</td>
                <td>
                  This Rshell uses the extracted from pepXML files pI and spectrum information to generate the pI distribution plot for the IEF fractionation derived data.                  <div><h3>Script</h3><pre>pI <- read.table(pIList, sep="\t");
l <- nrow(pI);
pI.m <- as.matrix(pI, nrow=l, ncol=2);
y <- max(pI.m[,2]);
ax <- c(1:24);
png("pI_Plot_IEF.png");
plot (pI.m[,1],pI.m[,2], xlim = c(1,24),ylim=NULL, col=1,xlab="Fraction Number", ylab="pI",pch=10, cex=1, cex.lab=2, mar=c(5,20,5,5))
axis (1, at=ax)
dev.off()
</pre></div><div><h3>R Server</h3>localhost:6311</div>                </td>
              </tr>
                          <tr>
                <td><b>pI_Plot_SCX</b></td>
                <td>rshell</td>
                <td>
                  This Rshell uses the extracted from pepXML files pI and spectrum information to generate the pI distribution plot for the SCX fractionation derived data.                  <div><h3>Script</h3><pre>pI <- read.table(pIList, sep="\t");
l <- nrow(pI);
pI.m <- as.matrix(pI, nrow=l, ncol=2);
y <- max(pI.m[,2]);
ax <- c(1:24);
png("pI_plot.png");
plot (pI.m[,1],pI.m[,2], xlim = c(1,24),ylim=NULL, col=1,xlab="Fraction Number", ylab="Calculated Mass (kDa)",pch=10, cex=1, cex.lab=2, mar=c(5,20,5,5))
axis (1, at=ax)
dev.off()
</pre></div><div><h3>R Server</h3>localhost:6311</div>                </td>
              </tr>
                          <tr>
                <td><b>Read_Text_File_SCX</b></td>
                <td>localworker</td>
                <td>
                                    <div><h3>Script</h3><pre>BufferedReader getReader (String fileUrl, String encoding) throws IOException {
		InputStreamReader reader;
		try {
			if (encoding == null) {
				reader = new FileReader(fileUrl);
			} else {
				reader = new InputStreamReader(new FileInputStream(fileUrl),encoding); 
			}
		}
		catch (FileNotFoundException e) {
			// try a real URL instead
			URL url = new URL(fileUrl);
			if (encoding == null) {
				reader = new InputStreamReader (url.openStream());
			} else {
				reader = new InputStreamReader (url.openStream(), encoding);
			}
		}
		return new BufferedReader(reader);
	}



StringBuffer sb = new StringBuffer(4000);

if (encoding == void) {
	encoding = null;
}

BufferedReader in = getReader(fileurl, encoding);
String str;
String lineEnding = System.getProperty("line.separator");

while ((str = in.readLine()) != null) {
	sb.append(str);
	sb.append(lineEnding);
}
in.close();
filecontents = sb.toString();
  
</pre></div>                </td>
              </tr>
                          <tr>
                <td><b>Extract_Parameters_SCX</b></td>
                <td>xpath</td>
                <td>
                                    <div><h3>Xpath Expression</h3><pre>/default:msms_pipeline_analysis/default:msms_run_summary/default:spectrum_query/default:search_result/default:search_hit/default:analysis_result/default:peptideprophet_result/default:search_score_summary/default:parameter[5]/@value</pre></div>                </td>
              </tr>
                          <tr>
                <td><b>Extract_Spectrum_Number_SCX</b></td>
                <td>xpath</td>
                <td>
                                    <div><h3>Xpath Expression</h3><pre>/default:msms_pipeline_analysis/default:msms_run_summary/default:spectrum_query/@spectrum</pre></div>                </td>
              </tr>
                          <tr>
                <td><b>Make_Well_Number_List_SCX</b></td>
                <td>beanshell</td>
                <td>
                  The JavaBean to convert the spectrum information to the fraction number                  <div><h3>Script</h3><pre>int s = well.size();
String[] temp = new String[s];
String wellnum = new String();
int num1 = 0;
int num;
String num2 = new String();

for (i=0; i<well.size();i++){
	wellnum = well.get(i).charAt(31).toString();
	if (wellnum.equals("A")) num1 = 0;
	else num1 = 12;
	num2 = well.get(i).charAt(33).toString();
	num = num1 + Integer.parseInt(num2);
	temp[i] = num.toString();
}

out = temp;</pre></div>                </td>
              </tr>
                          <tr>
                <td><b>Join_and_Insert_Tabs_SCX</b></td>
                <td>beanshell</td>
                <td>
                                    <div><h3>Script</h3><pre>int s = well.size();
String out = new String();

for (i=0; i<s;i++){
	out = out + well.get(i) + "\t" + pI.get(i) + "\n";
}

out1 = out;</pre></div>                </td>
              </tr>
            
          </table>
        
      </div>
    </div>
    
    <!-- Beanshells -->
        <div class="fold">
      <div class="foldTitle">
        Beanshells (17)
      </div>
      <div class="foldContent" style="display: none;">
                  <table class="simple">
            <tr>
              <th>Name</th>
              <th>Description</th>
              <th>Inputs</th>
              <th>Outputs</th>
            </tr>
          
                          <tr>
                <td><b>Make_Well_Number_List_PAGE</b></td>
                <td>The JavaBean to convert the spectrum information to the fraction number</td>
                <td>
                  
                    
                      well                      
                    
                  
                </td>
                <td>
                  
                    
                      out                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>Get_Fasta</b></td>
                <td>The JavaBean to extract protein sequences from the UniProt Database.</td>
                <td>
                  
                    
                      protein                      
                    
                  
                </td>
                <td>
                  
                    
                      seq                      <hr />
                    
                      ID                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>Join_and_Insert_Tabs_PAGE</b></td>
                <td></td>
                <td>
                  
                    
                      in1                      <hr />
                    
                      in2                      
                    
                  
                </td>
                <td>
                  
                    
                      out1                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>Get_MW</b></td>
                <td>This JavaBean component calculates protein molecular weights based on their sequences.</td>
                <td>
                  
                    
                      seq                      
                    
                  
                </td>
                <td>
                  
                    
                      out                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>Join_and_Insert_Tabs_IEF</b></td>
                <td></td>
                <td>
                  
                    
                      well                      <hr />
                    
                      pI                      
                    
                  
                </td>
                <td>
                  
                    
                      out1                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>Make_Well_Number_List_IEF</b></td>
                <td>The JavaBean to convert the spectrum information to the fraction number</td>
                <td>
                  
                    
                      well                      
                    
                  
                </td>
                <td>
                  
                    
                      out                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>Make_Well_Number_List_SCX</b></td>
                <td>The JavaBean to convert the spectrum information to the fraction number</td>
                <td>
                  
                    
                      well                      
                    
                  
                </td>
                <td>
                  
                    
                      out                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>Join_and_Insert_Tabs_SCX</b></td>
                <td></td>
                <td>
                  
                    
                      well                      <hr />
                    
                      pI                      
                    
                  
                </td>
                <td>
                  
                    
                      out1                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>Tandem_SCX</b></td>
                <td></td>
                <td>
                  
                    
                      mzxml_file                      <hr />
                    
                      fasta_file                      <hr />
                    
                      parameter_file                      
                    
                  
                </td>
                <td>
                  
                    
                      tandem_file                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>Tandem_PAGE</b></td>
                <td></td>
                <td>
                  
                    
                      mzxml_file                      <hr />
                    
                      fasta_file                      <hr />
                    
                      parameter_file                      
                    
                  
                </td>
                <td>
                  
                    
                      tandem_file                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>Tandem_IEF</b></td>
                <td></td>
                <td>
                  
                    
                      mzxml_file                      <hr />
                    
                      fasta_file                      <hr />
                    
                      parameter_file                      
                    
                  
                </td>
                <td>
                  
                    
                      tandem_file                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>SCX_CompassXport</b></td>
                <td></td>
                <td>
                  
                    
                      raw_data_dir                      <hr />
                    
                      raw_flag                      <hr />
                    
                      result_dir                      
                    
                  
                </td>
                <td>
                  
                    
                      output_file                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>PAGE_CompassXport</b></td>
                <td></td>
                <td>
                  
                    
                      raw_data_dir                      <hr />
                    
                      raw_flag                      <hr />
                    
                      result_dir                      
                    
                  
                </td>
                <td>
                  
                    
                      output_file                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>IEF_CompassXport</b></td>
                <td></td>
                <td>
                  
                    
                      raw_data_dir                      <hr />
                    
                      raw_flag                      <hr />
                    
                      result_dir                      
                    
                  
                </td>
                <td>
                  
                    
                      output_file                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>PeptideProphet_PAGE</b></td>
                <td></td>
                <td>
                  
                    
                      input_files                      <hr />
                    
                      arguments                      
                    
                  
                </td>
                <td>
                  
                    
                      output_file                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>PeptideProphet_IEF</b></td>
                <td></td>
                <td>
                  
                    
                      input_files                      <hr />
                    
                      arguments                      
                    
                  
                </td>
                <td>
                  
                    
                      output_file                      
                    
                  
                </td>
              </tr>
                          <tr>
                <td><b>PeptideProphet_SCX</b></td>
                <td></td>
                <td>
                  
                    
                      input_files                      <hr />
                    
                      arguments                      
                    
                  
                </td>
                <td>
                  
                    
                      output_file                      
                    
                  
                </td>
              </tr>
            
          </table>
        
      </div>
    </div>
    
    <!-- Sinks -->
        <div class="fold">
      <div class="foldTitle">
        Outputs (3)
      </div>
      <div class="foldContent" style="display: none;">
                  <table class="simple">
            <tr>
              <th>Name</th>
              <th>Description</th>
            </tr>
          
                          <tr>
                <td><b>MW_Plot</b></td>
                <td>
                  
                </td>
              </tr>
                          <tr>
                <td><b>pI_Plot_IEF</b></td>
                <td>
                  
                </td>
              </tr>
                          <tr>
                <td><b>pI_Plot_SCX</b></td>
                <td>
                  
                </td>
              </tr>
            
          </table>
        
      </div>
    </div>
    
    <!-- Datalinks -->
        <div class="fold">
      <div class="foldTitle">
        Datalinks (35)
      </div>
      <div class="foldContent" style="display: none;">
                  <table class="simple">
            <tr>
              <th>Source</th>
              <th>Sink</th>
            </tr>
          
                          <tr>
                <td>Database_Search:PAGE_pepXML_Output</td>
                <td>Read_Text_File_PAGE:fileurl</td>
              </tr>
                          <tr>
                <td>Read_Text_File_PAGE:filecontents</td>
                <td>Extract_Spectrum_Number_PAGE:xml_text</td>
              </tr>
                          <tr>
                <td>Read_Text_File_PAGE:filecontents</td>
                <td>Extract_Protein_Names:xml_text</td>
              </tr>
                          <tr>
                <td>Extract_Spectrum_Number_PAGE:nodelist</td>
                <td>Make_Well_Number_List_PAGE:well</td>
              </tr>
                          <tr>
                <td>Extract_Protein_Names:nodelist</td>
                <td>Get_Fasta:protein</td>
              </tr>
                          <tr>
                <td>Make_Well_Number_List_PAGE:out</td>
                <td>Join_and_Insert_Tabs_PAGE:in1</td>
              </tr>
                          <tr>
                <td>Get_MW:out</td>
                <td>Join_and_Insert_Tabs_PAGE:in2</td>
              </tr>
                          <tr>
                <td>Get_Fasta:seq</td>
                <td>Get_MW:seq</td>
              </tr>
                          <tr>
                <td>Join_and_Insert_Tabs_PAGE:out1</td>
                <td>MW_Plot:MassList</td>
              </tr>
                          <tr>
                <td>Prophet_Params</td>
                <td>Database_Search:Prophet_Params</td>
              </tr>
                          <tr>
                <td>FASTA_File</td>
                <td>Database_Search:FASTA_File</td>
              </tr>
                          <tr>
                <td>Tandem_Param_File</td>
                <td>Database_Search:Tandem_Param_File</td>
              </tr>
                          <tr>
                <td>PAGE_Results_Dir</td>
                <td>Database_Search:PAGE_Results_Dir</td>
              </tr>
                          <tr>
                <td>PAGE_Dir</td>
                <td>Database_Search:PAGE_Dir</td>
              </tr>
                          <tr>
                <td>SCX_Results_Dir</td>
                <td>Database_Search:SCX_Results_Dir</td>
              </tr>
                          <tr>
                <td>SCX_Dir</td>
                <td>Database_Search:SCX_Dir</td>
              </tr>
                          <tr>
                <td>IEF_Dir</td>
                <td>Database_Search:IEF_Dir</td>
              </tr>
                          <tr>
                <td>IEF_Results_Dir</td>
                <td>Database_Search:IEF_Results_Dir</td>
              </tr>
                          <tr>
                <td>Database_Search:IEF_pepXML_Output</td>
                <td>Read_Text_File_IEF:fileurl</td>
              </tr>
                          <tr>
                <td>Read_Text_File_IEF:filecontents</td>
                <td>Extract_Parameters_IEF:xml_text</td>
              </tr>
                          <tr>
                <td>Make_Well_Number_List_IEF:out</td>
                <td>Join_and_Insert_Tabs_IEF:well</td>
              </tr>
                          <tr>
                <td>Extract_Parameters_IEF:nodelist</td>
                <td>Join_and_Insert_Tabs_IEF:pI</td>
              </tr>
                          <tr>
                <td>Read_Text_File_IEF:filecontents</td>
                <td>Extract_Spectrum_Number_IEF:xml_text</td>
              </tr>
                          <tr>
                <td>Extract_Spectrum_Number_IEF:nodelist</td>
                <td>Make_Well_Number_List_IEF:well</td>
              </tr>
                          <tr>
                <td>Join_and_Insert_Tabs_IEF:out1</td>
                <td>pI_Plot_IEF:pIList</td>
              </tr>
                          <tr>
                <td>Join_and_Insert_Tabs_SCX:out1</td>
                <td>pI_Plot_SCX:pIList</td>
              </tr>
                          <tr>
                <td>Database_Search:SCX_pepXML_Output</td>
                <td>Read_Text_File_SCX:fileurl</td>
              </tr>
                          <tr>
                <td>Read_Text_File_SCX:filecontents</td>
                <td>Extract_Parameters_SCX:xml_text</td>
              </tr>
                          <tr>
                <td>Read_Text_File_SCX:filecontents</td>
                <td>Extract_Spectrum_Number_SCX:xml_text</td>
              </tr>
                          <tr>
                <td>Extract_Spectrum_Number_SCX:nodelist</td>
                <td>Make_Well_Number_List_SCX:well</td>
              </tr>
                          <tr>
                <td>Make_Well_Number_List_SCX:out</td>
                <td>Join_and_Insert_Tabs_SCX:well</td>
              </tr>
                          <tr>
                <td>Extract_Parameters_SCX:nodelist</td>
                <td>Join_and_Insert_Tabs_SCX:pI</td>
              </tr>
                          <tr>
                <td>MW_Plot:MW_plot</td>
                <td>MW_Plot</td>
              </tr>
                          <tr>
                <td>pI_Plot_IEF:pI_Plot_IEF</td>
                <td>pI_Plot_IEF</td>
              </tr>
                          <tr>
                <td>pI_Plot_SCX:pI_plot</td>
                <td>pI_Plot_SCX</td>
              </tr>
            
          </table>
        
      </div>
    </div>
    
    <!-- Coordinations -->
        <div class="fold">
      <div class="foldTitle">
        Coordinations (0)
      </div>
      <div class="foldContent" style="display: none;">
        
          <p class="none_text">None</p>
        
      </div>
    </div>
  
    

          
      
    </div>

  </div>

</div>

<div class="contribution_right_box">
  <div class="contribution_section_box" style= "font-size: 100%; padding: 0.7em 0.9em; font-weight: bold;">
    <p><img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[The type of workflow system that this Workflow is designed for.] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" /> Workflow Type</p>
    <p><a href="/content_types/2">Taverna 2</a></p>
  </div>

  <div class="contribution_section_box">
	<p class="heading">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[The Uploader is the person who uploaded/created this Workflow on myExperiment, and who manages it on myExperiment.] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Uploader
	</p>
	<p>
		<center>
			<div class="avatar" style="width: 66px">
  <a href="/users/57863"><img alt="Avatar" class="framed" height="60" src="/images/avatar.png?1680607579" title="Kate Mostovenko" width="60" /></a>
  
	<span style="display:block;margin-top:2px;text-align:center;">
		
			
		

    
			<a href="/users/57863" title="header=[] body=[Kate Mostovenko] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Kate Mostovenko</a>
		

    
	</span>
  
</div>

		</center>
	</p>
</div>


  

  

<div class="contribution_section_box">
	<p class="heading">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[This specifies what rights users have when they download and use this Workflow.] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		License
		<a name="license"></a>
	</p>

  

    
    
    
    
    


    

    <!--

    <rdf:RDF xmlns="http://creativecommons.org/ns#"
        xmlns:dc="http://purl.org/dc/elements/1.1/"
        xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#">
    <Work rdf:about="">
       <dc:title>Comparison of Peptide and Protein Fractionation Methods</dc:title>
       <dc:date>2013-03-27 10:04:23 UTC</dc:date>
       <dc:description>&lt;p&gt;
This workflow was used to analyze the data in a manuscript  by Mostovenko et al. (2013, submitted), comparing peptide and protein fractionation methods. The workflow identifies proteins by X!Tandem database search and validates the results using PeptideProphet. Additional information such as pI and fraction number is extracted and plotted for IEF and SCX data. For each protein identified in SDS-PAGE derived data sequences are downloaded from UniProt and plotted against the fraction number. Rshell script produces the figures essentially as they appear in the manuscript.
&lt;/p&gt;

&lt;p&gt;
Executing the workflow requires Rserve running and the Trans-Proteomic Pipeline (&lt;a href=&quot;http://tools.proteomecenter.org/wiki/index.php?title=Software:TPP&quot; rel=&quot;nofollow&quot;&gt;http://tools.proteomecenter.org/wiki/index.php?title=Software:TPP&lt;/a&gt;) installed with default settings and in the default location. The Rshell script contains the location where the figures will be generated. All other output files are stored in the input data folders by default. The version of X!Tandem called from this workflow is separate from the one installed with the Trans-Proteomic Pipeline (the location specified in the Tandem components).
&lt;/p&gt;

</dc:description>
       <dc:creator><Agent>
          <dc:title><a href="/users/57863" title="header=[] body=[Kate Mostovenko] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Kate Mostovenko</a></dc:title>
       </Agent></dc:creator>
       <dc:rights><Agent>
          <dc:title><a href="/users/57863" title="header=[] body=[Kate Mostovenko] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Kate Mostovenko</a></dc:title>
       </Agent></dc:rights>
       <dc:type rdf:resource="http://purl.org/dc/dcmitype/Data" />
       <dc:source rdf:resource="http://www.myexperiment.org/workflows/3486.html"/>
       <license rdf:resource="http://creativecommons.org/licenses/by-sa/3.0/" />
    </Work>

    <License rdf:about="http://creativecommons.org/licenses/by-sa/3.0/">
      <permits rdf:resource="http://creativecommons.org/ns#Reproduction" />  <permits rdf:resource="http://creativecommons.org/ns#Distribution" />  <requires rdf:resource="http://creativecommons.org/ns#Notice" />  <requires rdf:resource="http://creativecommons.org/ns#Attribution" />  <permits rdf:resource="http://creativecommons.org/ns#DerivativeWorks" />  <requires rdf:resource="http://creativecommons.org/ns#ShareAlike" />
    </License>

    </rdf:RDF>

    -->

  

  <div class="contribution_currentlicense">
    <p>
      
        All versions of this Workflow are
      
      
        licensed under: <br/><br/><a rel="Copyright" href="/licenses/2" title="Creative Commons Attribution-Share Alike 3.0 Unported License"><img src="/images/by-sa.png" /></a>
      
    </p>
  </div>
	
</div>


  <script type="text/javascript">
    function showVersion(form) {
        var url = $("resource_versions").value;
        location.href = url;
        form.submit
    }
</script>
<style>
  .version_selection {
      font-size:smaller;
      text-align: center;
  }
</style>

<div class="contribution_section_box">
  <p class="heading">
    <img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[This box shows version 1 for this entry] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />    <span>Version 1 </span>
    <span class="count_text">(of 1)</span>
  </p>
  
</div>

  <div class="contribution_section_box">
	
	<p class="heading" style="margin-bottom: 0; padding-bottom: 0;">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[Which people/groups get the credit for this Workflow?] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Credits <span class="count_text">(3)</span>
		<a name="credits"></a>
	</p>
	<span style="font-size: 85%; color: #666666; display: block; padding: 0.3em 0.5em 0 0.5em; margin: 0; line-height: 1em;">(People/Groups)</span>
	
		<div style="font-size: 85%;">
			<ul class="items">
				
					<li>
						
							
								<span class="icon"><img alt="User" src="/images/famfamfam_silk/user.png?1680607579" /> </span> 
							
							<a href="/users/57863" title="header=[] body=[Kate Mostovenko] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Kate Mostovenko</a>
						
					</li>
				
					<li>
						
							
								<span class="icon"><img alt="User" src="/images/famfamfam_silk/user.png?1680607579" /> </span> 
							
							<a href="/users/17146" title="header=[] body=[Yassene] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Yassene</a>
						
					</li>
				
					<li>
						
							
								<span class="icon"><img alt="User" src="/images/famfamfam_silk/user.png?1680607579" /> </span> 
							
							<a href="/users/23068" title="header=[] body=[Magnus Palmblad] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Magnus Palmblad</a>
						
					</li>
				
			</ul>
		</div>
	
	
	
	<p class="heading" style="margin-bottom: 0; padding-bottom: 0;">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[What other Workflows/Files is this one based on?] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Attributions <span class="count_text">(1)</span>
		<a name="attributions"></a>
	</p>
	<span style="font-size: 85%; color: #666666; display: block; padding: 0.3em 0.5em 0 0.5em; margin: 0; line-height: 1em;">(Workflows/Files)</span>
	
		<div style="font-size: 85%;">
			<p>
				<ul class="items">
					
						<li>
							
								
									
										<span class="icon"><img alt="Workflow" src="/images/redmond_studio/applications_16.png?1680607579" /> </span> 
									 
									<a href="/workflows/3057">Plasma Precipitation Analysis</a>
								
						  
						</li>
					
				</ul>
			</p>
		</div>
	
	
	
	
</div>


  


  <div class="contribution_section_box">
	<p class="heading">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[The tags for this Workflow entry.&lt;br/&gt;Note(1): if this Workflow has many versions, the tags apply to ALL versions in this entry.&lt;br/&gt;Note(2): the size of the tags show how popular they are on myExperiment.] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Tags <span id="tags_box_header_tag_count_span" class="count_text">(7)</span>
		<a name="tags"></a>
	</p>

	<script type="text/javascript">
  function showAllTags() {
    Element.show('all_tags');
	  Element.hide('user_tags');
	  return false;
  }
  
  function showUsersTags() {
  	Element.hide('all_tags');
	  Element.show('user_tags');
	  return false;
  }
	
	function flip_tags_mode()
	{
		if( $('toggle_tags_mode').checked )
		  showUsersTags();
	  else
		  showAllTags();
	}
</script>

<div id="tags_inner_box">
	
		<div class="box_currentuser_specific" style="padding: 0.3em 0.2em; margin: 0.6em 0.4em;">
		  <label for="toggle_tags_mode" style="font-size: 85%; font-weight: bold;"><input id="toggle_tags_mode" name="toggle_tags_mode" onclick="flip_tags_mode();" style="margin-right: 0.3em;" type="checkbox" value="1" />Uploader tags</label>
		</div>
		<div id="all_tags">
			
				<div class="hTagcloud">
  <ul class="popularity">
    <li><a href='/tags/3730' class='popular'>molecular weight</a></li>
<li> | </li>
    <li><a href='/tags/373' class='popular'>peptide</a></li>
<li> | </li>
    <li><a href='/tags/3729' class='popular'>pi</a></li>
<li> | </li>
    <li><a href='/tags/2151' class='v-popular'>plot</a></li>
<li> | </li>
    <li><a href='/tags/402' class='vvvv-popular'>protein</a></li>
<li> | </li>
    <li><a href='/tags/405' class='v-popular'>proteomics</a></li>
<li> | </li>
    <li><a href='/tags/451' class='v-popular'>search</a></li>
  </ul>
</div>

			
		</div>
		<div id="user_tags"  style="display: none;">
			
				<div class="hTagcloud">
  <ul class="popularity">
    <li><a href='/tags/3730' class='popular'>molecular weight</a></li>
<li> | </li>
    <li><a href='/tags/373' class='popular'>peptide</a></li>
<li> | </li>
    <li><a href='/tags/3729' class='popular'>pi</a></li>
<li> | </li>
    <li><a href='/tags/2151' class='v-popular'>plot</a></li>
<li> | </li>
    <li><a href='/tags/402' class='vvvv-popular'>protein</a></li>
<li> | </li>
    <li><a href='/tags/405' class='v-popular'>proteomics</a></li>
<li> | </li>
    <li><a href='/tags/451' class='v-popular'>search</a></li>
  </ul>
</div>

			
		</div>
	
</div>
	
	
	
	
		<p><small><b><a href="/session/new">Log in</a></b> to add Tags</small></p>
	
</div>


  
<div class="contribution_section_box">
	<p class="heading">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[All the Groups that this item has been shared with] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Shared with Groups <span class="count_text">(1)</span>
	</p>
	
		<div style="font-size: 85%;">
			<ul class="items">
				
					<li>
						<span class="icon"><img alt="Network-member" src="/images/famfamfam_silk/group.png?1680607579" /> </span>
						<a href="/groups/1020">Proteomics</a>
					</li>
				
			</ul>
		</div>
	
</div>

  <div class="contribution_section_box" id="in_packs_box">
  
	<p class="heading">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[All the Packs that this item (or versions of this item [[if available]]) is referred to in] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Featured In Packs  <span class="count_text">(0)</span>
		<a name="featured_in_packs"></a>
	</p>
	
		<p class="none_text">None</p>
	
	
	
		<p><small><b><a href="/session/new">Log in</a></b> to add to one of your Packs</small></p>
	
</div>


  <div class="contribution_section_box">
	
	<p class="heading" style="margin-bottom: 0; padding-bottom: 0;">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[Shows files and workflows that have attributed this Workflow.] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Attributed By <span class="count_text">(0)</span>
	  <a name="attributed_by"></a>
	</p>
	<span style="font-size: 85%; color: #666666; display: block; padding: 0.3em 0.5em 0 0.5em; margin: 0; line-height: 1em;">(Workflows/Files)</span>
	
		
		<p class="none_text">None</p>
	
	
</div>


  
<div class="contribution_section_box">
	<p class="heading">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[The users who have added this Workflow to their favourites] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Favourited By <font class="count_text">(0)</font>
		<a name="favourited_by"></a>
	</p>
	
	
		<p class="none_text">No one</p>
	
	
	
</div>

  

<div class="contribution_section_box">
	<p class="heading" style="margin-bottom: 0.3em;">
		<img alt="Information" src="/images/famfamfam_silk/information.png?1680607579" style="vertical-align:middle;" title="header=[] body=[Viewing / download counts are grand totals for this Workflow.&lt;br/&gt;Click &#39;see breakdown&#39; to view details.] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]" />
		Statistics
	</p>
	
	<div class="stats_box"> 
		<p style="font-size: 85%;">
  1412 viewings
</p>

	<p style="font-size: 85%;">
	  962 downloads
	</p>

<p style="margin-top: 0.2em;">
	<small>[ <a href="/workflows/3486/statistics">see breakdown</a> ]</small>
</p>
		
		<div class="fold" style="padding-top: 0.6em;">
			<div class="foldTitle">More</div>
			<div class="foldContent" style="display: none; text-align: left;">
				
					<p>
						<a href="#credits">3 credits</a> <!-- won't work for type=='group'; needs anchor called '#group_creditations' -->
					</p>
					
					<p>
						<a href="#attributions">1 attribution</a>
					</p>
				
				
				
					<p>
			      <a href="#tags">7 tags</a>
			    </p>
				
				
				
					<p>
			      <a href="#favourited_by">0 favourites</a>
			    </p>
				

				
					<p>
						<a href="#citations">0 citations</a>
			    </p>
				
				
				
					<p>
			      <a href="#reviews">0 reviews</a>
			    </p>
				
				
				
			    <p>
			    	<a href="#comments">0 comments</a>
			    </p>
				
			</div>
		</div>
		
		
		
		
	</div>
</div>

</div>

<div class="clearer">&nbsp;</div>

<!-- BEGIN tabs -->

<br/>

<div id="tabsContainer" class="tabsContainer"></div>

<div class="tabContainer">
  <div class="tabTitle">Citations (0)</div>
  <div class="tabContent">

    <a name="citations"></a>
    
      <p><i>None</i></p>
    
		
		<br/>

    

  </div>
</div>


<div class="tabContainer">
  <div class="tabTitle">Version History</div>
  <div class="tabContent">
    <a name="version_history"></a>
    <p>In chronological order:</p>

<ol class="version_history">
  
    <li>
      <div class="title">
        <a href="/workflows/3486/versions/1">Comparison of Peptide and Protein Fractionation Methods</a>
      </div>
      <div class="metadata">

        <p>
          Created
          
            by <a href="/users/57863" title="header=[] body=[Kate Mostovenko] cssheader=[boxoverTooltipHeader] cssbody=[boxoverTooltipBody] delay=[200]">Kate Mostovenko</a>
          
          on Wednesday 27 March 2013 10:04:23 (UTC)
        </p>

        

        
      </div>
    </li>
  
</ol>


  </div>
</div>



<!-- END tabs -->

<br/>
<br/>

<div id="ratingsBox">
  <h2>
	<span class="icon"><img alt="Reviews" src="/images/famfamfam_silk/report.png?1680607579" /> </span>	<span style="vertical-align: middle;"><a href="/workflows/3486/reviews">Reviews</a></span>
	<span class="count_text" style="vertical-align: middle;">(0)</span>
</h2>

<div class="reviewsBox">

	
		<p><i>No reviews yet</i></p>
		<a name="reviews"></a>
		<p style="margin-top: 0.6em;">
			<strong><a href="/workflows/3486/reviews/new">Be the first to review!</a></strong>
		</p>
	
	
</div>

</div>

<br/>
<br/>

<div id="commentsBox">
  <script src="/javascripts/ckeditor/ckeditor.js?1680607579" type="text/javascript"></script>
<script src="/javascripts/widgets/tooltip.js?1680607579" type="text/javascript"></script>

<h2>
	<span class="icon"><img alt="Comments" src="/images/famfamfam_silk/comments.png?1680607579" /> </span>	<span style="vertical-align: middle;">Comments</span>
	<span class="count_text" style="vertical-align: middle;">(0)</span>
</h2>

<ul class="sectionIcons">
  <li><span class="icon"><a href="http://www.myexperiment.org/workflows/3486/comments/timeline"><img alt="Timeline" src="/images/famfamfam_silk/timeline_marker.png?1680607579" /> View Timeline</a></span></li>
</ul>

<div class="commentsBox">
	
	
		<p><i>No comments yet</i></p>
		<a name="comments"></a>
		
			<p style="margin-top: 0.6em;"><strong><a href="/session/new">Log in</a> to make a comment </strong></p>
		
	
	
	<br/>
	
	

</div>


</div>

<!-- Other workflows that use a service this workflow users -->

<br />
<br />

<h2>
  <span class="icon"><img alt="Workflow" src="/images/redmond_studio/applications_16.png?1680607579" /> </span>  <span style="vertical-align: middle;">Other workflows that use similar services</span>
  <span class="count_text" style="vertical-align: middle;">(0)</span>
</h2>




    <p>There are no workflows in myExperiment that use similar services to this Workflow.</p>



<!-- OpenURL context object -->

<span class="Z3988" title="url_ver=Z39.88-2004&amp;url_ctx_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Actx&amp;ctx_ver=Z39.88-2004&amp;ctx_tim=2023-08-16T15%3A44%3A16%2B00%3A00&amp;ctx_id=&amp;ctx_enc=info%3Aofi%2Fenc%3AUTF-8&amp;rft.title=Comparison+of+Peptide+and+Protein+Fractionation+Methods&amp;rft.date=2013-03-27&amp;rft.au=Kate+Mostovenko&amp;rft.genre=unknown&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Adc"></span>
      </div>
    </div>




            <div id="ft">
          <div class="container">
  <div class="row">
    <div class="col-md-4 text-left">
      Funded by:
      <a href="http://biovel.eu">BioVeL</a>,
      <a href="http://www.epsrc.ac.uk/">EPSRC</a>,
      <a href="http://jisc.ac.uk/">JISC</a>,
      <a href="http://www.microsoft.com">Microsoft</a>

      <br/><br/>

      Affiliates:
      <a href="http://www.mygrid.org.uk">myGrid</a>,
      <a href="http://www.taverna.org.uk/">Taverna</a>,
      <a href="http://www.biocatalogue.org">BioCatalogue</a>
    </div>

    <div class="col-md-4 text-center">
      <a href="/about">About</a> |
      <a href="/privacy">Privacy</a> |
      <a href="http://wiki.myexperiment.org/index.php/Papers">Publications</a> |
      <a href="/feedback">Contact / Feedback</a> |
      <a href="http://lists.nongnu.org/mailman/listinfo/myexperiment-discuss">Mailing List</a>
    </div>

    <div class="col-md-4 text-right">
      Icons: <a href="http://www.famfamfam.com/lab/icons/silk/">Silk icon set 1.3</a><br/><br/>
      
    </div>

  </div>
  <div class="row">
    <div class="col-md-12 text-center">
      Copyright &#169; 2007 &ndash; 2018
      <a href="http://www.manchester.ac.uk/">The University of Manchester</a>
      and
      <a href="http://www.soton.ac.uk/">University of Southampton</a>
    </div>
  </div>
</div>

        </div>
    



<script type="text/javascript">
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
        (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
        m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-1498417-4', 'auto');
    ga('set', 'anonymizeIp', true);
    ga('send', 'pageview');
</script>




</body>
</html>

